## 1. Run this script to genrate all of the intermediate files needed for the plots.

$ bash ~/Scripts/Wraper_scripts/33_Rebuild_sit_plots_v4.sh /group/soranzo/manuel.tardaguila/Rebuild_sit_plots/test/

  The PCHi-C data is publicly available and comes from:

  "Lineage-Specific Genome Architecture Links Enhancers and Non-coding Disease Variants to Target Gene Promoters" Javierre et al, Cell 2016


  The ATAC data is publicly available: https://github.com/caleblareau/singlecell_bloodtraits/tree/master/data/bulk/ATAC and comes from:

  "Interrogation of human hematopoiesis at single-cell and single-variant resolution" Ulirsch et al, Nature Genetics 2019


## 2. To plot the situation plot (GWAS + DE model + ATAC + PCHi-C)

------> Jupyter notebook: Figure_4_and_Figure_S4_situation_plots.ipynb

## 3. To plot the Alternative Transcript Usage (ATU) plots:

------> Jupyter notebook: Figure_4_and_Figure_S4_ATU_plots.ipynb

## 4. To plot the Alternative Transcript Usage (ATU) plots for the different haplotypes:

------> Jupyter notebook: Figure_4_and_Figure_S4_ATU_haplotypes_plots.ipynb

## 5. MPRA snippets:

------> Jupyter notebook: Figure_4_and_Figure_S4_MPRA_snippet_plots.ipynb

------> copy the dependencies to Zenodo