{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5ba6e839-d30f-48b7-bade-2bebc44d281d",
   "metadata": {},
   "source": [
    "# Load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "898e4e86-7d07-4ed9-8353-945efc378c08",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:02.030507Z",
     "iopub.status.busy": "2025-07-06T21:41:02.028895Z",
     "iopub.status.idle": "2025-07-06T21:41:28.350832Z",
     "shell.execute_reply": "2025-07-06T21:41:28.349591Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/home/manuel.tardaguila/conda_envs/general_purpose/lib/R/library'"
      ],
      "text/latex": [
       "'/home/manuel.tardaguila/conda\\_envs/general\\_purpose/lib/R/library'"
      ],
      "text/markdown": [
       "'/home/manuel.tardaguila/conda_envs/general_purpose/lib/R/library'"
      ],
      "text/plain": [
       "[1] \"/home/manuel.tardaguila/conda_envs/general_purpose/lib/R/library\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/latex": [
       "'/home/manuel.tardaguila/R/x86\\_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/markdown": [
       "'/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/plain": [
       "[1] \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    ".libPaths()\n",
    "\n",
    "assign(\".lib.loc\", \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\", envir = environment(.libPaths))\n",
    ".libPaths()\n",
    "\n",
    "suppressMessages(library(\"plyr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"data.table\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"withr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggplot2\", lib.loc = \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"farver\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"labeling\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"optparse\", lib.loc = \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"dplyr\", lib.loc = \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"withr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"backports\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"broom\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"rstudioapi\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"tzdb\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"svglite\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggeasy\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"tidyverse\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"BiocGenerics\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"S4Vectors\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"IRanges\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"GenomeInfoDb\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"GenomicRanges\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"Biobase\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"AnnotationDbi\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"GO.db\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"org.Hs.eg.db\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggrepel\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"RColorBrewer\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"svglite\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"cowplot\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"Matrix\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"rtracklayer\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"Biostrings\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggrepel\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggarchery\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggnewscale\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"splitstackshape\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"viridis\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8af0ed16-9a2b-441f-91b6-553a182e9163",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:28.373231Z",
     "iopub.status.busy": "2025-07-06T21:41:28.352695Z",
     "iopub.status.idle": "2025-07-06T21:41:28.380844Z",
     "shell.execute_reply": "2025-07-06T21:41:28.379795Z"
    }
   },
   "outputs": [],
   "source": [
    "DEBUG<-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "10a98b7d-21f2-4f70-a3ee-0b4bbc59090b",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:28.383612Z",
     "iopub.status.busy": "2025-07-06T21:41:28.382654Z",
     "iopub.status.idle": "2025-07-06T21:41:28.390316Z",
     "shell.execute_reply": "2025-07-06T21:41:28.389295Z"
    }
   },
   "outputs": [],
   "source": [
    "build_files_from_pipelines_dir<-'/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/test/Build_files/'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6c920f4-552e-48fb-87f1-2981231f94c4",
   "metadata": {},
   "source": [
    "# Read input files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "96bfc149-eadb-4df5-9596-053d1586d4dd",
   "metadata": {},
   "source": [
    "## haplotypes file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3c36d8b2-5f46-44d8-8a07-0192ce591cb7",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:28.393202Z",
     "iopub.status.busy": "2025-07-06T21:41:28.392173Z",
     "iopub.status.idle": "2025-07-06T21:41:29.113034Z",
     "shell.execute_reply": "2025-07-06T21:41:29.111916Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rescue_haplotypes_0\n",
      "'data.frame':\t165051 obs. of  26 variables:\n",
      " $ VAR                           : chr  \"chr9_135920196_C_T\" \"chr9_135920196_C_T\" \"chr9_135920196_C_T\" \"chr9_135920196_C_T\" ...\n",
      " $ Proxy_VAR                     : chr  \"chr9_135864513_C_T\" \"chr9_135864513_C_T\" \"chr9_135864513_C_T\" \"chr9_135864513_C_T\" ...\n",
      " $ comparison                    : chr  \"HOM_REF__HET|HOM_REF\" \"HOM_REF__HET|HOM_REF\" \"HOM_REF__HET|HOM_REF\" \"HOM_REF__HET|HOM_REF\" ...\n",
      " $ ensembl_gene_id               : chr  \"ENSG00000000003\" \"ENSG00000000419\" \"ENSG00000000457\" \"ENSG00000000460\" ...\n",
      " $ HGNC                          : chr  \"TSPAN6\" \"DPM1\" \"SCYL3\" \"C1orf112\" ...\n",
      " $ CIS_gene_minuslogpvalue       : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ CIS_Beta                      : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ n_breakdown_string            : chr  NA NA NA NA ...\n",
      " $ n_SIG_CIS                     : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ n_tested_CIS                  : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Perc_SIG_CIS                  : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Block_PCHiC_minuslogpvalue    : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Block_PCHiC_Beta              : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Block_PCHiC_n_breakdown_string: chr  NA NA NA NA ...\n",
      " $ n_SIG_Block                   : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ n_tested_Block                : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Perc_SIG_Block                : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Genome_wide_minuslogpvalue    : num  0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ Genome_wide_Beta              : num  -0.1023 0.1286 -0.0172 -0.166 -0.1872 ...\n",
      " $ Genome_wide_n_breakdown_string: chr  \"HOM_REF__2861;HET|HOM_REF__13\" \"HOM_REF__2861;HET|HOM_REF__13\" \"HOM_REF__2861;HET|HOM_REF__13\" \"HOM_REF__2861;HET|HOM_REF__13\" ...\n",
      " $ n_SIG_GW                      : int  17 17 17 17 17 17 17 17 17 17 ...\n",
      " $ n_tested_GW                   : int  18339 18339 18339 18339 18339 18339 18339 18339 18339 18339 ...\n",
      " $ Perc_SIG_GW                   : num  0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1 ...\n",
      " $ CIS_Beta_Z_score              : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Block_PCHiC_Beta_Z_score      : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Genome_wide_Beta_Z_score      : num  -0.000524 0.000388 -0.000188 -0.000776 -0.000859 ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#### Read rescue_haplotypes----\n",
    "  \n",
    "rescue_haplotypes<-as.data.frame(fread(file=\"/group/soranzo/manuel.tardaguila/INTERVAL_results/chr9_135920196_C_T_ALL_BY_ALL_DE_LM_FPKM_results_Haplotypes.tsv\", sep=\"\\t\", header=T), stringsAsFactors=F)\n",
    "\n",
    "\n",
    "\n",
    "cat(\"rescue_haplotypes_0\\n\")\n",
    "cat(str(rescue_haplotypes))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18476377-f46d-46d0-9076-b5beef15bd15",
   "metadata": {},
   "source": [
    "## dataframe for sit plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "01f95fe1-050d-4f34-be2b-fd6fc0ab6ba2",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:29.115796Z",
     "iopub.status.busy": "2025-07-06T21:41:29.114855Z",
     "iopub.status.idle": "2025-07-06T21:41:29.159451Z",
     "shell.execute_reply": "2025-07-06T21:41:29.158431Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_to_print_SIT_plots_0\n",
      "'data.frame':\t23 obs. of  15 variables:\n",
      " $ VAR        : chr  \"chr22_28761148_C_T\" \"chr22_28761148_C_T\" \"chr1_92925654_G_C\" \"chr1_198680015_G_A\" ...\n",
      " $ rs         : chr  \"rs62237617\" \"rs62237617\" \"rs150649461\" \"rs12733073\" ...\n",
      " $ AS_ID      : chr  \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" ...\n",
      " $ Genes      : chr  \"ENSG00000183765;ENSG00000100154;ENSG00000159873;ENSG00000100209\" \"ENSG00000183765;ENSG00000100154;ENSG00000159873;ENSG00000100209\" \"ENSG00000162676;ENSG00000067208\" \"ENSG00000081237\" ...\n",
      " $ ATAC_cells : chr  \"CD4;CD8;NK\" \"Mega\" \"CMP;GMP_A\" NA ...\n",
      " $ oeID       : chr  \"432044\" \"432044\" NA \"49883\" ...\n",
      " $ PCHiC_cells: chr  \"tCD4;tCD8\" \"MK\" NA \"tCD4\" ...\n",
      " $ Height     : num  7 7 6 3 6 3 2 3.5 4 6 ...\n",
      " $ Width      : int  4 4 4 4 4 4 4 4 4 4 ...\n",
      " $ weight_A   : num  0.1 0.1 0.3 0.1 0.3 0.1 0.1 0.1 0.1 0.3 ...\n",
      " $ weight_B   : num  0.4 0.4 0.4 0.2 0.4 0.2 0.2 0.18 0.1 0.4 ...\n",
      " $ weight_C   : num  0.8 0.8 0.6 0 0.6 0 0 0 0 0.6 ...\n",
      " $ weight_D   : num  0.5 0.5 0.3 0.2 0.4 0.2 0 0.2 0.2 0.4 ...\n",
      " $ RNA_source : chr  \"Whole blood\" \"Whole blood\" \"Whole blood\" \"Whole blood\" ...\n",
      " $ beta_span  : chr  \"-0.25;-0.1;0;0.1;0.25\" \"-0.25;-0.1;0;0.1;0.25\" \"-0.5;-0.4;-0.3;-0.2;-0.1;0;0.1;0.2;0.4;0.5;0.3\" \"-0.1;-0.2;0;0.1;0.2\" ...\n",
      "\n",
      " chr [1:22] \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" \"pct__44\" ...\n",
      "\n",
      "Monocytes naive T-CD4 Cells Whole blood\n",
      "1 2 20\n"
     ]
    }
   ],
   "source": [
    " dataframe_to_print_SIT_plots<-as.data.frame(fread(file=\"/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/dataframe_to_print_SIT_plots_3.txt\", sep=\"\\t\", header=T), stringsAsFactors=F)\n",
    "  \n",
    "colnames(dataframe_to_print_SIT_plots)[which(colnames(dataframe_to_print_SIT_plots) == 'AS')]<-'AS_ID'\n",
    "\n",
    "dataframe_to_print_SIT_plots$beta_span<-gsub(\"\\\"\",\"\",dataframe_to_print_SIT_plots$beta_span)\n",
    "dataframe_to_print_SIT_plots$beta_span<-gsub(\"\\'\",\"\",dataframe_to_print_SIT_plots$beta_span)\n",
    "\n",
    "\n",
    "cat(\"dataframe_to_print_SIT_plots_0\\n\")\n",
    "cat(str(dataframe_to_print_SIT_plots))\n",
    "cat(\"\\n\")\n",
    "cat(str(unique(dataframe_to_print_SIT_plots$AS_ID)))\n",
    "cat(\"\\n\")\n",
    "cat(sprintf(as.character(names(summary(as.factor(dataframe_to_print_SIT_plots$RNA_source))))))\n",
    "cat(\"\\n\")\n",
    "cat(sprintf(as.character(summary(as.factor(dataframe_to_print_SIT_plots$RNA_source)))))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5694cf86-65cb-48db-8345-884ab0838b16",
   "metadata": {},
   "source": [
    "### Process the dataframe for sitplots"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a97f8446-5606-45c8-baf5-eafd0332db27",
   "metadata": {},
   "source": [
    "#### csplit long ENSG ----"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a5d4f193-b6c2-40e4-9714-1591fcd4f4e2",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:29.162051Z",
     "iopub.status.busy": "2025-07-06T21:41:29.161170Z",
     "iopub.status.idle": "2025-07-06T21:41:29.271494Z",
     "shell.execute_reply": "2025-07-06T21:41:29.270447Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_subset_ENSG_0\n",
      "'data.frame':\t22 obs. of  2 variables:\n",
      " $ AS_ID: chr  \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" ...\n",
      " $ Genes: chr  \"ENSG00000183765;ENSG00000100154;ENSG00000159873;ENSG00000100209\" \"ENSG00000183765;ENSG00000100154;ENSG00000159873;ENSG00000100209\" \"ENSG00000162676;ENSG00000067208\" \"ENSG00000081237\" ...\n",
      "\n",
      " chr [1:22] \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" \"pct__44\" ...\n",
      "\n",
      "dataframe_subset_ENSG_NO_NA_0\n",
      "'data.frame':\t22 obs. of  2 variables:\n",
      " $ AS_ID: chr  \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" ...\n",
      " $ Genes: chr  \"ENSG00000183765;ENSG00000100154;ENSG00000159873;ENSG00000100209\" \"ENSG00000183765;ENSG00000100154;ENSG00000159873;ENSG00000100209\" \"ENSG00000162676;ENSG00000067208\" \"ENSG00000081237\" ...\n",
      "\n",
      " chr [1:22] \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" \"pct__44\" ...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in type.convert.default(unlist(x, use.names = FALSE)):\n",
      "“'as.is' should be specified by the caller; using TRUE”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_subset_ENSG_NO_NA_1\n",
      "'data.frame':\t43 obs. of  2 variables:\n",
      " $ AS_ID          : chr  \"lymph__304\" \"lymph__304\" \"lymph__304\" \"lymph__304\" ...\n",
      " $ ensembl_gene_id: chr  \"ENSG00000183765\" \"ENSG00000100154\" \"ENSG00000159873\" \"ENSG00000100209\" ...\n",
      "\n",
      " chr [1:22] \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" \"pct__44\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "  \n",
    "  \n",
    "  \n",
    "  dataframe_subset_ENSG<-unique(dataframe_to_print_SIT_plots[,c(which(colnames(dataframe_to_print_SIT_plots) == 'AS_ID'),\n",
    "                                                                 which(colnames(dataframe_to_print_SIT_plots) == 'Genes'))])\n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"dataframe_subset_ENSG_0\\n\")\n",
    "    cat(str(dataframe_subset_ENSG))\n",
    "    cat(\"\\n\")\n",
    "    cat(str(unique(dataframe_subset_ENSG$AS_ID)))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "  \n",
    "  dataframe_subset_ENSG_NO_NA<-dataframe_subset_ENSG[!is.na(dataframe_subset_ENSG$Genes),]\n",
    "  \n",
    "  cat(\"dataframe_subset_ENSG_NO_NA_0\\n\")\n",
    "  cat(str(dataframe_subset_ENSG_NO_NA))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(dataframe_subset_ENSG_NO_NA$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  dataframe_subset_ENSG_NO_NA<-unique(as.data.frame(cSplit(dataframe_subset_ENSG_NO_NA,sep = ';', direction = \"long\",\n",
    "                                                            splitCols = \"Genes\"),stringsAsFactors=F))\n",
    "  \n",
    "  colnames(dataframe_subset_ENSG_NO_NA)[which(colnames(dataframe_subset_ENSG_NO_NA) == 'Genes')]<-'ensembl_gene_id'\n",
    "  \n",
    "  \n",
    " \n",
    "  cat(\"dataframe_subset_ENSG_NO_NA_1\\n\")\n",
    "  cat(str(dataframe_subset_ENSG_NO_NA))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(dataframe_subset_ENSG_NO_NA$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  \n",
    " \n",
    "  \n",
    "  \n",
    "  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a10d164-1a3a-45e8-9d47-e93043754311",
   "metadata": {},
   "source": [
    "#### csplit long PCHiC ----"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "abbc9cc2-6562-4b59-ae89-d5a6a5c8898d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:29.274328Z",
     "iopub.status.busy": "2025-07-06T21:41:29.273358Z",
     "iopub.status.idle": "2025-07-06T21:41:29.334393Z",
     "shell.execute_reply": "2025-07-06T21:41:29.333130Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_subset_PCHiC_0\n",
      "'data.frame':\t22 obs. of  3 variables:\n",
      " $ AS_ID      : chr  \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" ...\n",
      " $ oeID       : chr  \"432044\" \"432044\" NA \"49883\" ...\n",
      " $ PCHiC_cells: chr  \"tCD4;tCD8\" \"MK\" NA \"tCD4\" ...\n",
      "\n",
      " chr [1:22] \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" \"pct__44\" ...\n",
      "\n",
      "dataframe_subset_PCHiC_NO_NA_0\n",
      "'data.frame':\t14 obs. of  3 variables:\n",
      " $ AS_ID      : chr  \"lymph__304\" \"pct__350\" \"lymph__38\" \"pct__44\" ...\n",
      " $ oeID       : chr  \"432044\" \"432044\" \"49883\" \"50866\" ...\n",
      " $ PCHiC_cells: chr  \"tCD4;tCD8\" \"MK\" \"tCD4\" \"MK\" ...\n",
      "\n",
      " chr [1:14] \"lymph__304\" \"pct__350\" \"lymph__38\" \"pct__44\" \"ret__209\" ...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in type.convert.default(unlist(x, use.names = FALSE)):\n",
      "“'as.is' should be specified by the caller; using TRUE”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_subset_PCHiC_NO_NA_1\n",
      "'data.frame':\t18 obs. of  3 variables:\n",
      " $ AS_ID      : chr  \"lymph__304\" \"pct__350\" \"lymph__38\" \"pct__44\" ...\n",
      " $ oeID       : chr  \"432044\" \"432044\" \"49883\" \"50866\" ...\n",
      " $ PCHiC_cells: chr  \"tCD4;tCD8\" \"MK\" \"tCD4\" \"MK\" ...\n",
      "\n",
      " chr [1:14] \"lymph__304\" \"pct__350\" \"lymph__38\" \"pct__44\" \"ret__209\" ...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in type.convert.default(unlist(x, use.names = FALSE)):\n",
      "“'as.is' should be specified by the caller; using TRUE”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_subset_PCHiC_NO_NA_2\n",
      "'data.frame':\t22 obs. of  3 variables:\n",
      " $ AS_ID    : chr  \"lymph__304\" \"lymph__304\" \"pct__350\" \"lymph__38\" ...\n",
      " $ oeID     : chr  \"432044\" \"432044\" \"432044\" \"49883\" ...\n",
      " $ Cell_Type: chr  \"tCD4\" \"tCD8\" \"MK\" \"tCD4\" ...\n",
      "\n",
      " chr [1:14] \"lymph__304\" \"pct__350\" \"lymph__38\" \"pct__44\" \"ret__209\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "  \n",
    " \n",
    "  dataframe_subset_PCHiC<-unique(dataframe_to_print_SIT_plots[,c(which(colnames(dataframe_to_print_SIT_plots) == 'AS_ID'),\n",
    "                                                                 which(colnames(dataframe_to_print_SIT_plots) == 'oeID'),\n",
    "                                                                 which(colnames(dataframe_to_print_SIT_plots) == 'PCHiC_cells'))])\n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"dataframe_subset_PCHiC_0\\n\")\n",
    "    cat(str(dataframe_subset_PCHiC))\n",
    "    cat(\"\\n\")\n",
    "    cat(str(unique(dataframe_subset_PCHiC$AS_ID)))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "  \n",
    "  dataframe_subset_PCHiC_NO_NA<-dataframe_subset_PCHiC[!is.na(dataframe_subset_PCHiC$oeID),]\n",
    "  \n",
    "  cat(\"dataframe_subset_PCHiC_NO_NA_0\\n\")\n",
    "  cat(str(dataframe_subset_PCHiC_NO_NA))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(dataframe_subset_PCHiC_NO_NA$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  \n",
    "  \n",
    " \n",
    "  dataframe_subset_PCHiC_NO_NA<-unique(as.data.frame(cSplit(dataframe_subset_PCHiC_NO_NA,sep = ';', direction = \"long\",\n",
    "                                                  splitCols = \"oeID\"),stringsAsFactors=F))\n",
    " \n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"dataframe_subset_PCHiC_NO_NA_1\\n\")\n",
    "    cat(str(dataframe_subset_PCHiC_NO_NA))\n",
    "    cat(\"\\n\")\n",
    "    cat(str(unique(dataframe_subset_PCHiC_NO_NA$AS_ID)))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "    \n",
    "  dataframe_subset_PCHiC_NO_NA<-unique(as.data.frame(cSplit(dataframe_subset_PCHiC_NO_NA,sep = ';', direction = \"long\",\n",
    "                                                            splitCols = \"PCHiC_cells\"),stringsAsFactors=F))\n",
    "  \n",
    "  colnames(dataframe_subset_PCHiC_NO_NA)[which(colnames(dataframe_subset_PCHiC_NO_NA) == 'PCHiC_cells')]<-'Cell_Type'\n",
    "  \n",
    "  \n",
    "  cat(\"dataframe_subset_PCHiC_NO_NA_2\\n\")\n",
    "  cat(str(dataframe_subset_PCHiC_NO_NA))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(dataframe_subset_PCHiC_NO_NA$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  \n",
    "  \n",
    "  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39292165-5e90-456f-a4f7-fc981b497e91",
   "metadata": {},
   "source": [
    "#### csplit long ATAC ----"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7b5af7c6-ca4e-45f6-8f54-4b3c9669bb9a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:29.337671Z",
     "iopub.status.busy": "2025-07-06T21:41:29.336530Z",
     "iopub.status.idle": "2025-07-06T21:41:29.388168Z",
     "shell.execute_reply": "2025-07-06T21:41:29.386862Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_subset_ATAC_0\n",
      "'data.frame':\t22 obs. of  2 variables:\n",
      " $ AS_ID     : chr  \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" ...\n",
      " $ ATAC_cells: chr  \"CD4;CD8;NK\" \"Mega\" \"CMP;GMP_A\" NA ...\n",
      "\n",
      " chr [1:22] \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" \"pct__44\" ...\n",
      "\n",
      "dataframe_subset_ATAC_NO_NA_0\n",
      "'data.frame':\t11 obs. of  2 variables:\n",
      " $ AS_ID     : chr  \"lymph__304\" \"pct__350\" \"mono__27\" \"pct__44\" ...\n",
      " $ ATAC_cells: chr  \"CD4;CD8;NK\" \"Mega\" \"CMP;GMP_A\" \"HSC;Mega;GMP_A\" ...\n",
      "\n",
      " chr [1:11] \"lymph__304\" \"pct__350\" \"mono__27\" \"pct__44\" \"mono__325\" ...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in type.convert.default(unlist(x, use.names = FALSE)):\n",
      "“'as.is' should be specified by the caller; using TRUE”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_subset_ATAC_NO_NA_1\n",
      "'data.frame':\t30 obs. of  2 variables:\n",
      " $ AS_ID    : chr  \"lymph__304\" \"lymph__304\" \"lymph__304\" \"pct__350\" ...\n",
      " $ Cell_Type: chr  \"CD4\" \"CD8\" \"NK\" \"Mega\" ...\n",
      "\n",
      " chr [1:11] \"lymph__304\" \"pct__350\" \"mono__27\" \"pct__44\" \"mono__325\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "  \n",
    "  dataframe_subset_ATAC<-unique(dataframe_to_print_SIT_plots[,c(which(colnames(dataframe_to_print_SIT_plots) == 'AS_ID'),\n",
    "                                                                 which(colnames(dataframe_to_print_SIT_plots) == 'ATAC_cells'))])\n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"dataframe_subset_ATAC_0\\n\")\n",
    "    cat(str(dataframe_subset_ATAC))\n",
    "    cat(\"\\n\")\n",
    "    cat(str(unique(dataframe_subset_ATAC$AS_ID)))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "  \n",
    "  dataframe_subset_ATAC_NO_NA<-dataframe_subset_ATAC[!is.na(dataframe_subset_ATAC$ATAC_cells),]\n",
    "  \n",
    "  cat(\"dataframe_subset_ATAC_NO_NA_0\\n\")\n",
    "  cat(str(dataframe_subset_ATAC_NO_NA))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(dataframe_subset_ATAC_NO_NA$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  \n",
    "  \n",
    "  \n",
    "  dataframe_subset_ATAC_NO_NA<-unique(as.data.frame(cSplit(dataframe_subset_ATAC_NO_NA,sep = ';', direction = \"long\",\n",
    "                                                            splitCols = \"ATAC_cells\"),stringsAsFactors=F))\n",
    "  \n",
    "  colnames(dataframe_subset_ATAC_NO_NA)[which(colnames(dataframe_subset_ATAC_NO_NA) == 'ATAC_cells')]<-'Cell_Type'\n",
    "  \n",
    "  \n",
    "  cat(\"dataframe_subset_ATAC_NO_NA_1\\n\")\n",
    "  cat(str(dataframe_subset_ATAC_NO_NA))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(dataframe_subset_ATAC_NO_NA$AS_ID)))\n",
    "  cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0872780-00f8-44e4-be2b-b2c66284e5cd",
   "metadata": {},
   "source": [
    "## ATACseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "75728dda-a38d-4b2f-a89a-b098bc3d88bc",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:29.391418Z",
     "iopub.status.busy": "2025-07-06T21:41:29.390303Z",
     "iopub.status.idle": "2025-07-06T21:41:30.702141Z",
     "shell.execute_reply": "2025-07-06T21:41:30.700993Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATAC_tracks_0\n",
      "'data.frame':\t65806 obs. of  22 variables:\n",
      " $ chr  : Ord.factor w/ 19 levels \"chr1\"<\"chr2\"<..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ start: int  201868605 201872852 201879084 201885669 201887430 201894176 201906989 201910440 201914312 201915441 ...\n",
      " $ end  : int  201869105 201873352 201879584 201886169 201887930 201894676 201907489 201910940 201914812 201915941 ...\n",
      " $ B    : num  20 36 24 20 19 9 11 27 14 51 ...\n",
      " $ CD4  : num  13 4 14 15 12 9 6 11 14 37 ...\n",
      " $ CD8  : num  9 8 10 14 6 8 5 9 17 34 ...\n",
      " $ CLP  : num  9 3 3 3 8 7 5 7 7 23 ...\n",
      " $ CMP  : num  11 10 17 47 5 36 34 4 27 150 ...\n",
      " $ Ery  : num  23 18 16 9 11 9 1 23 11 29 ...\n",
      " $ GMP_A: num  3 6 5 11 17 5 8 5 11 18 ...\n",
      " $ GMP_B: num  2 3 4 6 5 6 12 3 8 20 ...\n",
      " $ GMP_C: num  1 0 1 3 2 3 10 0 5 25 ...\n",
      " $ HSC  : num  4 6 6 25 7 7 42 3 15 158 ...\n",
      " $ LMPP : num  2 2 1 8 4 13 13 2 16 44 ...\n",
      " $ mDC  : num  1 5 4 2 6 3 10 1 1 11 ...\n",
      " $ Mega : num  12 14 11 22 10 8 8 14 18 148 ...\n",
      " $ MEP  : num  11 12 17 26 18 12 22 9 59 147 ...\n",
      " $ Mono : num  9 8 12 12 8 8 10 7 7 14 ...\n",
      " $ MPP  : num  4 5 2 34 5 7 31 8 29 155 ...\n",
      " $ NK   : num  11 9 12 18 8 5 11 16 14 51 ...\n",
      " $ pDC  : num  5 5 2 2 1 0 2 1 3 6 ...\n",
      " $ AS_ID: chr  \"baso_p__14\" \"baso_p__14\" \"baso_p__14\" \"baso_p__14\" ...\n",
      "\n",
      " chr [1:333] \"baso_p__14\" \"pct__44\" \"mscv__45\" \"plt__49\" \"lymph_p__27\" ...\n",
      "\n",
      "ATAC_tracks_subset_0\n",
      "'data.frame':\t1855 obs. of  22 variables:\n",
      " $ chr  : Ord.factor w/ 19 levels \"chr1\"<\"chr2\"<..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ start: int  201868605 201872852 201879084 201885669 201887430 201894176 201906989 201910440 201914312 201915441 ...\n",
      " $ end  : int  201869105 201873352 201879584 201886169 201887930 201894676 201907489 201910940 201914812 201915941 ...\n",
      " $ B    : num  20 36 24 20 19 9 11 27 14 51 ...\n",
      " $ CD4  : num  13 4 14 15 12 9 6 11 14 37 ...\n",
      " $ CD8  : num  9 8 10 14 6 8 5 9 17 34 ...\n",
      " $ CLP  : num  9 3 3 3 8 7 5 7 7 23 ...\n",
      " $ CMP  : num  11 10 17 47 5 36 34 4 27 150 ...\n",
      " $ Ery  : num  23 18 16 9 11 9 1 23 11 29 ...\n",
      " $ GMP_A: num  3 6 5 11 17 5 8 5 11 18 ...\n",
      " $ GMP_B: num  2 3 4 6 5 6 12 3 8 20 ...\n",
      " $ GMP_C: num  1 0 1 3 2 3 10 0 5 25 ...\n",
      " $ HSC  : num  4 6 6 25 7 7 42 3 15 158 ...\n",
      " $ LMPP : num  2 2 1 8 4 13 13 2 16 44 ...\n",
      " $ mDC  : num  1 5 4 2 6 3 10 1 1 11 ...\n",
      " $ Mega : num  12 14 11 22 10 8 8 14 18 148 ...\n",
      " $ MEP  : num  11 12 17 26 18 12 22 9 59 147 ...\n",
      " $ Mono : num  9 8 12 12 8 8 10 7 7 14 ...\n",
      " $ MPP  : num  4 5 2 34 5 7 31 8 29 155 ...\n",
      " $ NK   : num  11 9 12 18 8 5 11 16 14 51 ...\n",
      " $ pDC  : num  5 5 2 2 1 0 2 1 3 6 ...\n",
      " $ AS_ID: chr  \"pct__44\" \"pct__44\" \"pct__44\" \"pct__44\" ...\n",
      "\n",
      " chr [1:11] \"pct__44\" \"mono__27\" \"eo__306\" \"lymph__328\" \"mono__325\" ...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(ATAC_tracks_subset, id.vars = c(\"chr\", \"start\", \"end\", \"AS_ID\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(ATAC_tracks_subset). In the next version, this warning will become an error.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATAC_tracks_subset.m_0\n",
      "'data.frame':\t33390 obs. of  6 variables:\n",
      " $ chr      : Ord.factor w/ 19 levels \"chr1\"<\"chr2\"<..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ start    : int  201868605 201872852 201879084 201885669 201887430 201894176 201906989 201910440 201914312 201915441 ...\n",
      " $ end      : int  201869105 201873352 201879584 201886169 201887930 201894676 201907489 201910940 201914812 201915941 ...\n",
      " $ AS_ID    : chr  \"pct__44\" \"pct__44\" \"pct__44\" \"pct__44\" ...\n",
      " $ Cell_Type: Factor w/ 18 levels \"B\",\"CD4\",\"CD8\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ counts   : num  20 36 24 20 19 9 11 27 14 51 ...\n",
      "\n",
      " chr [1:11] \"pct__44\" \"mono__27\" \"eo__306\" \"lymph__328\" \"mono__325\" ...\n",
      "\n",
      "B CD4 CD8 CLP CMP Ery GMP_A GMP_B GMP_C HSC LMPP mDC Mega MEP Mono MPP NK pDC\n",
      "1855 1855 1855 1855 1855 1855 1855 1855 1855 1855 1855 1855 1855 1855 1855 1855 1855 1855\n",
      "Min. 1st Qu. Median Mean 3rd Qu. Max.\n",
      "0 5 12 41.4061395627433 29 2564\n",
      "ATAC_tracks_subset.m_1\n",
      "'data.frame':\t4914 obs. of  6 variables:\n",
      " $ AS_ID    : chr  \"eo__172\" \"eo__172\" \"eo__172\" \"eo__172\" ...\n",
      " $ Cell_Type: chr  \"CD4\" \"CD4\" \"CD4\" \"CD4\" ...\n",
      " $ chr      : Ord.factor w/ 19 levels \"chr1\"<\"chr2\"<..: 15 15 15 15 15 15 15 15 15 15 ...\n",
      " $ start    : int  38507785 38509121 38527420 38528873 38484647 38486440 38529470 38506870 38518835 38599792 ...\n",
      " $ end      : int  38508285 38509621 38527920 38529373 38485147 38486940 38529970 38507370 38519335 38600292 ...\n",
      " $ counts   : num  16 7 53 5 2 34 5 13 214 143 ...\n",
      "\n",
      " chr [1:11] \"eo__172\" \"eo__306\" \"lymph__189\" \"lymph__304\" \"lymph__328\" ...\n",
      "\n",
      "CD4 CD8 CLP CMP GMP_A HSC Mega MEP Mono MPP NK\n",
      "674 359 315 821 485 528 212 177 521 463 359\n",
      "Min. 1st Qu. Median Mean 3rd Qu. Max.\n",
      "0 8 16 57.9521774521775 41 2564\n"
     ]
    }
   ],
   "source": [
    "#### READ ATAC_tracks ----\n",
    "\n",
    "setwd(build_files_from_pipelines_dir)\n",
    "\n",
    "  ATAC_tracks<-readRDS(file=\"Intersected_ATAC.rds\")\n",
    "  \n",
    "  cat(\"ATAC_tracks_0\\n\")\n",
    "  cat(str(ATAC_tracks))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(ATAC_tracks$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  ATAC_tracks_subset<-ATAC_tracks[which(ATAC_tracks$AS_ID%in%dataframe_subset_ATAC_NO_NA$AS_ID),]\n",
    "  \n",
    "  cat(\"ATAC_tracks_subset_0\\n\")\n",
    "  cat(str(ATAC_tracks_subset))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(ATAC_tracks_subset$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  ATAC_tracks_subset.m<-melt(ATAC_tracks_subset, \n",
    "                          id.vars=c('chr','start','end','AS_ID'), variable.name=\"Cell_Type\", value.name=\"counts\")\n",
    "  \n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"ATAC_tracks_subset.m_0\\n\")\n",
    "    cat(str(ATAC_tracks_subset.m))\n",
    "    cat(\"\\n\")\n",
    "    cat(str(unique(ATAC_tracks_subset.m$AS_ID)))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(names(summary(as.factor(ATAC_tracks_subset.m$Cell_Type))))))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(summary(as.factor(ATAC_tracks_subset.m$Cell_Type)))))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(names(summary(ATAC_tracks_subset.m$counts)))))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(summary(ATAC_tracks_subset.m$counts))))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "  \n",
    "  ATAC_tracks_subset.m<-merge(dataframe_subset_ATAC_NO_NA,\n",
    "                           ATAC_tracks_subset.m,\n",
    "                           by=c(\"AS_ID\",\"Cell_Type\"))\n",
    "  \n",
    "  cat(\"ATAC_tracks_subset.m_1\\n\")\n",
    "  cat(str(ATAC_tracks_subset.m))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(ATAC_tracks_subset.m$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(names(summary(as.factor(ATAC_tracks_subset.m$Cell_Type))))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(summary(as.factor(ATAC_tracks_subset.m$Cell_Type)))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(names(summary(ATAC_tracks_subset.m$counts)))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(summary(ATAC_tracks_subset.m$counts))))\n",
    "  cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a522ae8-a3fe-4b34-bbb9-d4fbd9c955b6",
   "metadata": {},
   "source": [
    "## PCHi-C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ec2c7d97-9c62-4f4c-b7fd-5882124ca26e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:30.704913Z",
     "iopub.status.busy": "2025-07-06T21:41:30.703980Z",
     "iopub.status.idle": "2025-07-06T21:41:31.056295Z",
     "shell.execute_reply": "2025-07-06T21:41:31.055207Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PCHiC_HITS_0\n",
      "'data.frame':\t39498 obs. of  29 variables:\n",
      " $ oeID           : chr  \"148237\" \"148237\" \"148237\" \"148246\" ...\n",
      " $ baitID         : chr  \"148316\" \"148316\" \"148316\" \"148316\" ...\n",
      " $ AS_ID          : chr  \"pct__121\" \"pdw__95\" \"plt__122\" \"pct__121\" ...\n",
      " $ oeChr          : chr  \"12\" \"12\" \"12\" \"12\" ...\n",
      " $ oeStart        : int  22473153 22473153 22473153 22526224 22526224 22526224 22598980 22598980 22598980 22610592 ...\n",
      " $ oeEnd          : int  22488539 22488539 22488539 22527749 22527749 22527749 22599135 22599135 22599135 22611259 ...\n",
      " $ baitChr        : chr  \"12\" \"12\" \"12\" \"12\" ...\n",
      " $ baitStart      : int  22777870 22777870 22777870 22777870 22777870 22777870 22777870 22777870 22777870 22777870 ...\n",
      " $ baitEnd        : int  22780005 22780005 22780005 22780005 22780005 22780005 22780005 22780005 22780005 22780005 ...\n",
      " $ Mon            : num  5.05 5.05 5.05 5.48 5.48 ...\n",
      " $ Mac0           : num  1.05 1.05 1.05 3.02 3.02 ...\n",
      " $ Mac1           : num  1.53 1.53 1.53 2.82 2.82 ...\n",
      " $ Mac2           : num  0.573 0.573 0.573 3.282 3.282 ...\n",
      " $ Neu            : num  0.4789 0.4789 0.4789 0.0271 0.0271 ...\n",
      " $ MK             : num  2.35 2.35 2.35 1.98 1.98 ...\n",
      " $ EP             : num  0.314 0.314 0.314 0.246 0.246 ...\n",
      " $ Ery            : num  2.79 2.79 2.79 2.34 2.34 ...\n",
      " $ FoeT           : num  0.707 0.707 0.707 0.583 0.583 ...\n",
      " $ nCD4           : num  1.44 1.44 1.44 1.07 1.07 ...\n",
      " $ tCD4           : num  3.17 3.17 3.17 2.12 2.12 ...\n",
      " $ aCD4           : num  2.79 2.79 2.79 1.74 1.74 ...\n",
      " $ naCD4          : num  2.28 2.28 2.28 1.06 1.06 ...\n",
      " $ nCD8           : num  2.4 2.4 2.4 1.09 1.09 ...\n",
      " $ tCD8           : num  2.72 2.72 2.72 1.19 1.19 ...\n",
      " $ nB             : num  1.02 1.02 1.02 0.77 0.77 ...\n",
      " $ tB             : num  0.564 0.564 0.564 0.707 0.707 ...\n",
      " $ baitName       : chr  \"ETNK1\" \"ETNK1\" \"ETNK1\" \"ETNK1\" ...\n",
      " $ baitSYMBOL     : chr  \"ETNK1\" \"ETNK1\" \"ETNK1\" \"ETNK1\" ...\n",
      " $ ensembl_gene_id: chr  \"ENSG00000139163\" \"ENSG00000139163\" \"ENSG00000139163\" \"ENSG00000139163\" ...\n",
      "\n",
      " chr [1:327] \"pct__121\" \"pdw__95\" \"plt__122\" \"mrv__107\" \"hlr_p__91\" ...\n",
      "\n",
      "PCHiC_HITS_subset_0\n",
      "'data.frame':\t1904 obs. of  29 variables:\n",
      " $ oeID           : chr  \"287176\" \"287176\" \"287176\" \"287177\" ...\n",
      " $ baitID         : chr  \"287183\" \"287212\" \"287228\" \"287189\" ...\n",
      " $ AS_ID          : chr  \"lymph__189\" \"lymph__189\" \"lymph__189\" \"lymph__189\" ...\n",
      " $ oeChr          : chr  \"17\" \"17\" \"17\" \"17\" ...\n",
      " $ oeStart        : int  38460840 38460840 38460840 38466233 38466233 38466434 38472354 38472354 38473747 38473747 ...\n",
      " $ oeEnd          : int  38466232 38466232 38466232 38466433 38466433 38472353 38473746 38473746 38490412 38490412 ...\n",
      " $ baitChr        : chr  \"17\" \"17\" \"17\" \"17\" ...\n",
      " $ baitStart      : int  38517587 38655914 38719900 38567144 38719900 38719900 38460840 38719900 38460840 38719900 ...\n",
      " $ baitEnd        : int  38524866 38661896 38726969 38574423 38726969 38726969 38466232 38726969 38466232 38726969 ...\n",
      " $ Mon            : num  11.837 1.483 1.215 5.642 0.696 ...\n",
      " $ Mac0           : num  10.072 1.958 0.819 2.15 0.73 ...\n",
      " $ Mac1           : num  6.256 0.663 0.759 2.971 0.633 ...\n",
      " $ Mac2           : num  9.648 1.104 0.468 1.389 0.12 ...\n",
      " $ Neu            : num  10.235 0.302 0.522 3.07 0.23 ...\n",
      " $ MK             : num  8.65 1.59 0.63 3.71 1.4 ...\n",
      " $ EP             : num  6.89 1.98 1.44 5.74 4.5 ...\n",
      " $ Ery            : num  11.967 1.437 0.761 2.372 2.477 ...\n",
      " $ FoeT           : num  5.73 2.41 3.63 3.42 4.68 ...\n",
      " $ nCD4           : num  6.45 5.72 4.12 2.56 5 ...\n",
      " $ tCD4           : num  8.15 5.64 3.64 1.95 5.05 ...\n",
      " $ aCD4           : num  9.6 4.22 4.38 4.05 5.21 ...\n",
      " $ naCD4          : num  7.02 4.03 2.63 3.08 4.15 ...\n",
      " $ nCD8           : num  7.89 5.09 4.27 2.43 5.96 ...\n",
      " $ tCD8           : num  9.2 5.16 6.2 4.27 8.55 ...\n",
      " $ nB             : num  9.67 3.66 3.32 1.96 4.36 ...\n",
      " $ tB             : num  7.53 2.32 3.14 2.11 6.67 ...\n",
      " $ baitName       : chr  \"CTD-2267D19.4;GJD3\" \"TNS4\" \"CCR7\" \"TOP2A\" ...\n",
      " $ baitSYMBOL     : chr  \"CTD-2267D19.4;GJD3\" \"TNS4\" \"CCR7\" \"TOP2A\" ...\n",
      " $ ensembl_gene_id: chr  \"ENSG00000264655;ENSG00000183153\" \"ENSG00000131746\" \"ENSG00000126353\" \"ENSG00000131747\" ...\n",
      "\n",
      " chr [1:13] \"lymph__189\" \"eo__172\" \"rbc__248\" \"lymph__218\" \"ret__209\" ...\n",
      "\n",
      "scale_df_0\n",
      "'data.frame':\t1904 obs. of  26 variables:\n",
      " $ AS_ID     : chr  \"lymph__189\" \"lymph__189\" \"lymph__189\" \"lymph__189\" ...\n",
      " $ oeStart   : int  38460840 38460840 38460840 38466233 38466233 38466434 38472354 38472354 38473747 38473747 ...\n",
      " $ oeEnd     : int  38466232 38466232 38466232 38466433 38466433 38472353 38473746 38473746 38490412 38490412 ...\n",
      " $ oeID      : chr  \"287176\" \"287176\" \"287176\" \"287177\" ...\n",
      " $ baitChr   : chr  \"17\" \"17\" \"17\" \"17\" ...\n",
      " $ baitStart : int  38517587 38655914 38719900 38567144 38719900 38719900 38460840 38719900 38460840 38719900 ...\n",
      " $ baitEnd   : int  38524866 38661896 38726969 38574423 38726969 38726969 38466232 38726969 38466232 38726969 ...\n",
      " $ baitID    : chr  \"287183\" \"287212\" \"287228\" \"287189\" ...\n",
      " $ baitSYMBOL: chr  \"CTD-2267D19.4;GJD3\" \"TNS4\" \"CCR7\" \"TOP2A\" ...\n",
      " $ Mon       : num  11.837 1.483 1.215 5.642 0.696 ...\n",
      " $ Mac0      : num  10.072 1.958 0.819 2.15 0.73 ...\n",
      " $ Mac1      : num  6.256 0.663 0.759 2.971 0.633 ...\n",
      " $ Mac2      : num  9.648 1.104 0.468 1.389 0.12 ...\n",
      " $ Neu       : num  10.235 0.302 0.522 3.07 0.23 ...\n",
      " $ MK        : num  8.65 1.59 0.63 3.71 1.4 ...\n",
      " $ EP        : num  6.89 1.98 1.44 5.74 4.5 ...\n",
      " $ Ery       : num  11.967 1.437 0.761 2.372 2.477 ...\n",
      " $ FoeT      : num  5.73 2.41 3.63 3.42 4.68 ...\n",
      " $ nCD4      : num  6.45 5.72 4.12 2.56 5 ...\n",
      " $ tCD4      : num  8.15 5.64 3.64 1.95 5.05 ...\n",
      " $ aCD4      : num  9.6 4.22 4.38 4.05 5.21 ...\n",
      " $ naCD4     : num  7.02 4.03 2.63 3.08 4.15 ...\n",
      " $ nCD8      : num  7.89 5.09 4.27 2.43 5.96 ...\n",
      " $ tCD8      : num  9.2 5.16 6.2 4.27 8.55 ...\n",
      " $ nB        : num  9.67 3.66 3.32 1.96 4.36 ...\n",
      " $ tB        : num  7.53 2.32 3.14 2.11 6.67 ...\n",
      "\n",
      " chr [1:13] \"lymph__189\" \"eo__172\" \"rbc__248\" \"lymph__218\" \"ret__209\" ...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(scale_df, id.vars = c(\"AS_ID\", \"oeStart\", \"oeEnd\", \"oeID\", :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(scale_df). In the next version, this warning will become an error.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scale_df.m_0\n",
      "'data.frame':\t32368 obs. of  11 variables:\n",
      " $ AS_ID       : chr  \"lymph__189\" \"lymph__189\" \"lymph__189\" \"lymph__189\" ...\n",
      " $ oeStart     : int  38460840 38460840 38460840 38466233 38466233 38466434 38472354 38472354 38473747 38473747 ...\n",
      " $ oeEnd       : int  38466232 38466232 38466232 38466433 38466433 38472353 38473746 38473746 38490412 38490412 ...\n",
      " $ oeID        : chr  \"287176\" \"287176\" \"287176\" \"287177\" ...\n",
      " $ baitChr     : chr  \"17\" \"17\" \"17\" \"17\" ...\n",
      " $ baitStart   : int  38517587 38655914 38719900 38567144 38719900 38719900 38460840 38719900 38460840 38719900 ...\n",
      " $ baitEnd     : int  38524866 38661896 38726969 38574423 38726969 38726969 38466232 38726969 38466232 38726969 ...\n",
      " $ baitID      : chr  \"287183\" \"287212\" \"287228\" \"287189\" ...\n",
      " $ baitSYMBOL  : chr  \"CTD-2267D19.4;GJD3\" \"TNS4\" \"CCR7\" \"TOP2A\" ...\n",
      " $ Cell_Type   : Factor w/ 17 levels \"Mon\",\"Mac0\",\"Mac1\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ ChicagoScore: num  11.837 1.483 1.215 5.642 0.696 ...\n",
      "\n",
      " chr [1:13] \"lymph__189\" \"eo__172\" \"rbc__248\" \"lymph__218\" \"ret__209\" ...\n",
      "\n",
      "Mon Mac0 Mac1 Mac2 Neu MK EP Ery FoeT nCD4 tCD4 aCD4 naCD4 nCD8 tCD8 nB tB\n",
      "1904 1904 1904 1904 1904 1904 1904 1904 1904 1904 1904 1904 1904 1904 1904 1904 1904\n",
      "Min. 1st Qu. Median Mean 3rd Qu. Max.\n",
      "0 1.8025439525496 3.46117593158413 4.90781585284249 5.95046790739808 154.824612862939\n"
     ]
    }
   ],
   "source": [
    " #### READ PCHiC_HITS ----\n",
    "\n",
    "setwd(build_files_from_pipelines_dir)\n",
    "\n",
    "  PCHiC_HITS<-readRDS(file=\"PCHiC_AS_POST_ENSEMBL.rds\")\n",
    "  \n",
    "  cat(\"PCHiC_HITS_0\\n\")\n",
    "  cat(str(PCHiC_HITS))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(PCHiC_HITS$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  \n",
    "  PCHiC_HITS_subset<-PCHiC_HITS[which(PCHiC_HITS$AS_ID%in%dataframe_subset_PCHiC_NO_NA$AS_ID),]\n",
    "  \n",
    "  cat(\"PCHiC_HITS_subset_0\\n\")\n",
    "  cat(str(PCHiC_HITS_subset))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(PCHiC_HITS_subset$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  indx.scale.CS<-c(which(colnames(PCHiC_HITS_subset) == 'AS_ID'),\n",
    "                   which(colnames(PCHiC_HITS_subset) == 'oeStart'),which(colnames(PCHiC_HITS_subset) == 'oeEnd'),which(colnames(PCHiC_HITS_subset) == 'oeID'),\n",
    "              which(colnames(PCHiC_HITS_subset) == 'baitChr'),which(colnames(PCHiC_HITS_subset) == 'baitStart'),which(colnames(PCHiC_HITS_subset) == 'baitEnd'),which(colnames(PCHiC_HITS_subset) == 'baitID'),\n",
    "              which(colnames(PCHiC_HITS_subset) == 'baitSYMBOL'),\n",
    "              which(colnames(PCHiC_HITS_subset) == 'Mon'),which(colnames(PCHiC_HITS_subset) == 'Mac0'),which(colnames(PCHiC_HITS_subset) == 'Mac1'),\n",
    "              which(colnames(PCHiC_HITS_subset) == 'Mac2'),which(colnames(PCHiC_HITS_subset) == 'Neu'),which(colnames(PCHiC_HITS_subset) == 'MK'),\n",
    "              which(colnames(PCHiC_HITS_subset) == 'EP'),which(colnames(PCHiC_HITS_subset) == 'Ery'),which(colnames(PCHiC_HITS_subset) == 'FoeT'),\n",
    "              which(colnames(PCHiC_HITS_subset) == 'nCD4'),which(colnames(PCHiC_HITS_subset) == 'tCD4'),which(colnames(PCHiC_HITS_subset) == 'aCD4'),\n",
    "              which(colnames(PCHiC_HITS_subset) == 'naCD4'),which(colnames(PCHiC_HITS_subset) == 'nCD8'),which(colnames(PCHiC_HITS_subset) == 'tCD8'),\n",
    "              which(colnames(PCHiC_HITS_subset) == 'nB'),which(colnames(PCHiC_HITS_subset) == 'tB'))\n",
    "  \n",
    "  \n",
    "  \n",
    "  scale_df<-unique(PCHiC_HITS_subset[,indx.scale.CS])\n",
    "  \n",
    "\n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    " \n",
    "    cat(\"scale_df_0\\n\")\n",
    "    cat(str(scale_df))\n",
    "    cat(\"\\n\")\n",
    "    cat(str(unique(scale_df$AS_ID)))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "\n",
    "  \n",
    "  scale_df.m<-melt(scale_df, id.vars=c(\"AS_ID\",\"oeStart\",\"oeEnd\",\"oeID\",\"baitChr\",\"baitStart\",\"baitEnd\",\"baitID\",\"baitSYMBOL\"),\n",
    "                 variable.name=\"Cell_Type\", value.name=\"ChicagoScore\")\n",
    "  \n",
    "  \n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"scale_df.m_0\\n\")\n",
    "    cat(str(scale_df.m))\n",
    "    cat(\"\\n\")\n",
    "    cat(str(unique(scale_df.m$AS_ID)))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(names(summary(as.factor(scale_df.m$Cell_Type))))))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(summary(as.factor(scale_df.m$Cell_Type)))))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(names(summary(scale_df.m$ChicagoScore)))))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(summary(scale_df.m$ChicagoScore))))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "  \n",
    "  \n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "34bc1c63-c30e-463a-9b6e-3672f98760a8",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.059038Z",
     "iopub.status.busy": "2025-07-06T21:41:31.058099Z",
     "iopub.status.idle": "2025-07-06T21:41:31.115280Z",
     "shell.execute_reply": "2025-07-06T21:41:31.114177Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scale_df.m_subset_0\n",
      "'data.frame':\t45 obs. of  11 variables:\n",
      " $ AS_ID       : chr  \"baso__92\" \"baso__92\" \"baso__92\" \"baso__92\" ...\n",
      " $ Cell_Type   : chr  \"Mon\" \"Mon\" \"Mon\" \"Mon\" ...\n",
      " $ oeID        : chr  \"397014\" \"397014\" \"397014\" \"397016\" ...\n",
      " $ oeStart     : int  219020056 219020056 219020056 219025053 219025053 219025053 219025053 219134749 219134749 219134749 ...\n",
      " $ oeEnd       : int  219022996 219022996 219022996 219027246 219027246 219027246 219027246 219135860 219135860 219135860 ...\n",
      " $ baitChr     : chr  \"2\" \"2\" \"2\" \"2\" ...\n",
      " $ baitStart   : int  219134749 219261830 218988939 219134749 219261830 219119543 218988939 219221459 219261830 219242191 ...\n",
      " $ baitEnd     : int  219135860 219266623 218991185 219135860 219266623 219131509 218991185 219242190 219266623 219248639 ...\n",
      " $ baitID      : chr  \"397048\" \"397070\" \"397003\" \"397048\" ...\n",
      " $ baitSYMBOL  : chr  \"AAMP;PNKD\" \"CTDSP1;RP11-378A13.2\" \"CXCR2\" \"AAMP;PNKD\" ...\n",
      " $ ChicagoScore: num  1.258 0.527 0.347 2.359 1.288 ...\n",
      "\n",
      " chr [1:11] \"baso__92\" \"eo__172\" \"lymph__189\" \"lymph__304\" \"lymph__328\" ...\n",
      "\n",
      "Mac1 MK Mon nB nCD4 Neu tCD4 tCD8\n",
      "2 5 17 1 2 10 5 3\n",
      "Min. 1st Qu. Median Mean 3rd Qu. Max.\n",
      "0 3.04681570746066 6.01351132400557 9.66336816403606 9.24093522700596 50.4447918167562\n"
     ]
    }
   ],
   "source": [
    "scale_df.m_subset<-merge(dataframe_subset_PCHiC_NO_NA,\n",
    "                           scale_df.m,\n",
    "                           by=c(\"AS_ID\",\"Cell_Type\",\"oeID\"))\n",
    "  \n",
    "  cat(\"scale_df.m_subset_0\\n\")\n",
    "  cat(str(scale_df.m_subset))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(scale_df.m_subset$AS_ID)))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(names(summary(as.factor(scale_df.m_subset$Cell_Type))))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(summary(as.factor(scale_df.m_subset$Cell_Type)))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(names(summary(scale_df.m_subset$ChicagoScore)))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(summary(scale_df.m_subset$ChicagoScore))))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d05497f5-b6a2-467b-89b1-d25985bfdd57",
   "metadata": {},
   "source": [
    "### graph parameters PCHi-C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "dd9afd0f-c41f-4e34-aaec-de1a1be487ea",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.118054Z",
     "iopub.status.busy": "2025-07-06T21:41:31.117118Z",
     "iopub.status.idle": "2025-07-06T21:41:31.237452Z",
     "shell.execute_reply": "2025-07-06T21:41:31.236346Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "breaks_CS\n",
      "0 5 50.4447918167562\n",
      "labels_CS\n",
      "0 5 50\n",
      "check_0\n",
      "'data.frame':\t4692 obs. of  11 variables:\n",
      " $ AS_ID       : chr  \"lymph__218\" \"lymph__218\" \"lymph__218\" \"lymph__218\" ...\n",
      " $ oeStart     : int  15335031 15356751 15371168 15371168 15371168 15382101 15382101 15394213 15395401 15395401 ...\n",
      " $ oeEnd       : int  15345796 15357037 15382100 15382100 15382100 15394212 15394212 15395400 15402689 15402689 ...\n",
      " $ oeID        : chr  \"322682\" \"322686\" \"322689\" \"322689\" ...\n",
      " $ baitChr     : chr  \"19\" \"19\" \"19\" \"19\" ...\n",
      " $ baitStart   : int  15489199 15516830 15440064 15489199 15516830 15440064 15489199 15440064 15440064 15489199 ...\n",
      " $ baitEnd     : int  15497116 15530487 15464246 15497116 15530487 15464246 15497116 15464246 15464246 15497116 ...\n",
      " $ baitID      : chr  \"322706\" \"322713\" \"322702\" \"322706\" ...\n",
      " $ baitSYMBOL  : chr  \"AKAP8\" \"AKAP8L\" \"BRD4\" \"AKAP8\" ...\n",
      " $ Cell_Type   : Factor w/ 17 levels \"Mon\",\"Mac0\",\"Mac1\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ ChicagoScore: num  2.11 2.44 2.61 2.54 1.48 ...\n",
      "\n",
      " chr [1:2] \"lymph__218\" \"ret__209\"\n",
      "\n",
      "Mon Mac0 Mac1 Mac2 Neu MK EP Ery FoeT nCD4 tCD4 aCD4 naCD4 nCD8 tCD8 nB tB\n",
      "276 276 276 276 276 276 276 276 276 276 276 276 276 276 276 276 276\n",
      "Min. 1st Qu. Median Mean 3rd Qu. Max.\n",
      "0 2.49919889074113 3.98758799776727 5.44319326392044 6.36042047564976 46.2465806093139\n",
      "vector_colors_0\n",
      " chr [1:6] \"#08306B\" \"#2171B5\" \"#6BAED6\" \"#C6DBEF\" \"#DF65B0\" \"#CE1256\"\n",
      "\n"
     ]
    }
   ],
   "source": [
    "breaks_CS<-unique(sort(unique(c(0,5,max(scale_df.m_subset$ChicagoScore[!is.na(scale_df.m_subset$ChicagoScore)])))))\n",
    "  labels_CS<-as.character(round(breaks_CS,0))\n",
    "  \n",
    " \n",
    "  cat(\"breaks_CS\\n\")\n",
    "  cat(sprintf(as.character(breaks_CS)))\n",
    "  cat(\"\\n\")\n",
    "  cat(\"labels_CS\\n\")\n",
    "  cat(sprintf(as.character(labels_CS)))\n",
    "  cat(\"\\n\")\n",
    " \n",
    "  \n",
    "  \n",
    "  check<-scale_df.m[-which(scale_df.m$AS_ID%in%scale_df.m_subset$AS_ID),]\n",
    "  \n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"check_0\\n\")\n",
    "    cat(str(check))\n",
    "    cat(\"\\n\")\n",
    "    cat(str(unique(check$AS_ID)))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(names(summary(as.factor(check$Cell_Type))))))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(summary(as.factor(check$Cell_Type)))))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(names(summary(check$ChicagoScore)))))\n",
    "    cat(\"\\n\")\n",
    "    cat(sprintf(as.character(summary(check$ChicagoScore))))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "  # setwd(out)\n",
    "  # \n",
    "  # write.table(check,file='test.tsv', sep=\"\\t\", quote=F, row.names = F)\n",
    "  # \n",
    "  # \n",
    "  # ####################################################################\n",
    "  # quit(status = 1)\n",
    "  \n",
    "  #### vector_colors ----\n",
    "  \n",
    "  vector_colors_ALL_but_index<-brewer.pal(9, \"Blues\")[c(9,7,5,3)]\n",
    "  vector_colors_index<-brewer.pal(9, \"PuRd\")[c(5,7)]\n",
    "  \n",
    "  vector_colors<-c(vector_colors_ALL_but_index,vector_colors_index)\n",
    "  \n",
    "  cat(\"vector_colors_0\\n\")\n",
    "  cat(str(vector_colors))\n",
    "  cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f8c9882c-2c5c-4bce-910d-a92cb093186c",
   "metadata": {},
   "source": [
    "## Variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "720a8b0b-8f59-4415-b78e-b2093299ecde",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.240603Z",
     "iopub.status.busy": "2025-07-06T21:41:31.239453Z",
     "iopub.status.idle": "2025-07-06T21:41:31.313661Z",
     "shell.execute_reply": "2025-07-06T21:41:31.312347Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "variants_file_0\n",
      "'data.frame':\t1164 obs. of  28 variables:\n",
      " $ VAR                   : chr  \"chr22_50949811_T_C\" \"chr22_50949811_T_C\" \"chr22_50967912_C_T\" \"chr22_50949811_T_C\" ...\n",
      " $ rs                    : chr  \"rs543594419\" \"rs543594419\" \"rs74624637\" \"rs543594419\" ...\n",
      " $ variant_id            : int  16348658 16348658 16348988 16348658 16348916 16348988 16349035 16348658 16348988 16223665 ...\n",
      " $ chr                   : Ord.factor w/ 25 levels \"chr1\"<\"chr2\"<..: 22 22 22 22 22 22 22 22 22 22 ...\n",
      " $ pos37                 : int  50949811 50949811 50967912 50949811 50963540 50967912 50971266 50949811 50967912 39341663 ...\n",
      " $ ref                   : chr  \"T\" \"T\" \"C\" \"T\" ...\n",
      " $ alt                   : chr  \"C\" \"C\" \"T\" \"C\" ...\n",
      " $ block_no              : int  310 323 323 339 339 339 339 387 387 262 ...\n",
      " $ phenotype             : chr  \"mrv\" \"mch\" \"mch\" \"mcv\" ...\n",
      " $ ncondind              : int  1 1 1 3 3 3 3 1 1 2 ...\n",
      " $ credset_size          : int  4 8 8 10 10 10 10 5 5 20 ...\n",
      " $ finemap_beta          : num  -0.1809 -0.1098 -0.1072 -0.1391 0.0803 ...\n",
      " $ finemap_se            : num  0.00815 0.00791 0.00776 0.00783 0.00207 ...\n",
      " $ finemap_z             : num  -22.2 -13.9 -13.8 -17.8 38.8 ...\n",
      " $ finemap_prob          : num  0.996 0.267 0.101 0.761 0.399 ...\n",
      " $ finemap_log10bf       : chr  \"6.14315\" \"3.31446\" \"2.80525\" \"4.25462\" ...\n",
      " $ info                  : num  0.94 0.94 0.971 0.94 1 ...\n",
      " $ is_cond_ind           : int  1 1 0 1 1 0 1 1 0 1 ...\n",
      " $ chromstates           : chr  \"E1C2:E1C3:E1C5:E1C6:E1C7:E1C8:E1C9:E1C10:E1C11:E1C12:E1C14:E1C15:E1C16:E1C17:E1C18:E1C19:E2C20:E1C21:E1C22:E1C2\"| __truncated__ \"E1C2:E1C3:E1C5:E1C6:E1C7:E1C8:E1C9:E1C10:E1C11:E1C12:E1C14:E1C15:E1C16:E1C17:E1C18:E1C19:E2C20:E1C21:E1C22:E1C2\"| __truncated__ \"E8C1:E11C6:E11C7:E11C8:E11C9:E11C10:E10C12:E10C13:E10C14:E11C17:E10C18:E10C20:E10C21:E10C22:E10C23:E10C24:E1C26\"| __truncated__ \"E1C2:E1C3:E1C5:E1C6:E1C7:E1C8:E1C9:E1C10:E1C11:E1C12:E1C14:E1C15:E1C16:E1C17:E1C18:E1C19:E2C20:E1C21:E1C22:E1C2\"| __truncated__ ...\n",
      " $ AS_ID                 : chr  \"mrv__310\" \"mch__323\" \"mch__323\" \"mcv__339\" ...\n",
      " $ phenotype_DEF         : Ord.factor w/ 29 levels \"PLT#\"<\"MPV\"<\"PDW\"<..: 17 8 8 6 6 6 6 18 18 11 ...\n",
      " $ pos                   : int  50949811 50949811 50967912 50949811 50963540 50967912 50971266 50949811 50967912 39341663 ...\n",
      " $ maf_origin            : num  0.0184 0.0184 0.0185 0.0184 0.4279 ...\n",
      " $ Variant_classification: Ord.factor w/ 5 levels \"Tier 1\"<\"Tier 2\"<..: 5 5 3 5 1 3 1 5 3 1 ...\n",
      " $ Paper_group           : chr  NA NA \"Rare non-coding lowPP\" NA ...\n",
      " $ min                   : num  -0.189 -0.1177 -0.115 -0.147 0.0782 ...\n",
      " $ max                   : num  -0.1727 -0.1019 -0.0995 -0.1313 0.0824 ...\n",
      " $ DUMMY                 : Ord.factor w/ 470 levels \"rs543594419\"<..: 1 1 2 1 3 2 4 1 2 5 ...\n",
      "\n",
      " chr [1:470] \"chr22_50949811_T_C\" \"chr22_50967912_C_T\" \"chr22_50963540_G_C\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#### READ variants_file ----\n",
    "\n",
    "setwd(build_files_from_pipelines_dir)\n",
    "\n",
    "variants_file<-readRDS(file=\"AS_selector.rds\")\n",
    "\n",
    "cat(\"variants_file_0\\n\")\n",
    "cat(str(variants_file))\n",
    "cat(\"\\n\")\n",
    "cat(str(unique(variants_file$VAR)))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6990934d-60af-457f-b9ee-689835142a5b",
   "metadata": {},
   "source": [
    "## GWAS Blocks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8800a5cb-ed5d-4a4c-b6e8-4fed4611aa80",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.316924Z",
     "iopub.status.busy": "2025-07-06T21:41:31.315810Z",
     "iopub.status.idle": "2025-07-06T21:41:31.349912Z",
     "shell.execute_reply": "2025-07-06T21:41:31.348638Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BLOCK_POST_0\n",
      "'data.frame':\t333 obs. of  4 variables:\n",
      " $ chr  : Ord.factor w/ 19 levels \"chr1\"<\"chr2\"<..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ start: int  201865580 201865580 201865580 201951500 198348663 198372447 198413661 198430015 158132664 158132664 ...\n",
      " $ end  : int  202561834 202561834 202561834 202561834 198930015 198930015 198930015 198930015 158888810 158888810 ...\n",
      " $ AS_ID: chr  \"baso_p__14\" \"pct__44\" \"mscv__45\" \"plt__49\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    " #### READ GWAS block sizes ----\n",
    "\n",
    "setwd(build_files_from_pipelines_dir)\n",
    "\n",
    "BLOCK_POST<-readRDS(file=\"Boundaries_AS_POST_genes_POST_ATAC.rds\")\n",
    "\n",
    "\n",
    "\n",
    "cat(\"BLOCK_POST_0\\n\")\n",
    "cat(str(BLOCK_POST))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "461df99b-56fd-4d98-a378-eb9a2ccf1ac6",
   "metadata": {},
   "source": [
    "## genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c5033d3d-cdeb-4570-8517-676984780db4",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.353159Z",
     "iopub.status.busy": "2025-07-06T21:41:31.352010Z",
     "iopub.status.idle": "2025-07-06T21:41:31.438214Z",
     "shell.execute_reply": "2025-07-06T21:41:31.436872Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene_tracks_0\n",
      "'data.frame':\t7313 obs. of  7 variables:\n",
      " $ chr            : Ord.factor w/ 19 levels \"chr1\"<\"chr2\"<..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ start          : int  201865580 201924619 201951500 201977073 202091986 202102532 202116141 202137179 202163029 202300785 ...\n",
      " $ end            : int  201915715 201939789 201975275 201986316 202102720 202113869 202130716 202158583 202288909 202311108 ...\n",
      " $ ensembl_gene_id: chr  \"ENSG00000163431\" \"ENSG00000134375\" \"ENSG00000176393\" \"ENSG00000163435\" ...\n",
      " $ strand         : chr  \"-\" \"+\" \"+\" \"+\" ...\n",
      " $ hgnc           : chr  \"LMOD1\" \"TIMM17A\" \"RNPEP\" \"ELF3\" ...\n",
      " $ AS_ID          : chr  \"baso_p__14\" \"baso_p__14\" \"baso_p__14\" \"baso_p__14\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    " #### READ gene_tracks ----\n",
    "\n",
    "setwd(build_files_from_pipelines_dir)\n",
    "\n",
    "gene_tracks<-readRDS(file=\"Intersected_genes.rds\")\n",
    "\n",
    "cat(\"gene_tracks_0\\n\")\n",
    "cat(str(gene_tracks))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bf35f6a8-3091-4746-87c6-602a744b420a",
   "metadata": {},
   "source": [
    "## DE and ATU results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f7e7f09c-a3d5-4da9-b059-5f02793d0113",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.441456Z",
     "iopub.status.busy": "2025-07-06T21:41:31.440336Z",
     "iopub.status.idle": "2025-07-06T21:41:31.564237Z",
     "shell.execute_reply": "2025-07-06T21:41:31.562999Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Table_S7_0\n",
      "'data.frame':\t4544 obs. of  12 variables:\n",
      " $ VAR                              : chr  \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" ...\n",
      " $ HGNC                             : chr  \"DPH5\" \"DPH5\" \"DPH5\" \"EXTL2\" ...\n",
      " $ ensembl_gene_id                  : chr  \"ENSG00000117543\" \"ENSG00000117543\" \"ENSG00000117543\" \"ENSG00000162694\" ...\n",
      " $ rs                               : Ord.factor w/ 88 levels \"rs115820925\"<..: 6 6 6 6 6 6 6 6 6 6 ...\n",
      " $ RNASeq_source                    : Ord.factor w/ 4 levels \"Whole blood\"<..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ Analysis                         : Ord.factor w/ 2 levels \"DE\"<\"ATU\": 1 2 2 1 2 2 1 1 1 1 ...\n",
      " $ transcript_id                    : chr  NA \"ENST00000427040\" \"ENST00000370109\" NA ...\n",
      " $ Beta                             : num  -0.00182 0 0.11236 -0.28817 0.3839 ...\n",
      " $ Beta_Z_score                     : num  -0.0232 -0.038 0.588 -0.0823 2.1008 ...\n",
      " $ adjusted_pval                    : num  1 NA 0.708 0.871 0.708 ...\n",
      " $ adjusted_minus_logpval           : num  0 NA 0.15 0.06 0.15 NA 0 0.1 0 0.01 ...\n",
      " $ regulated_gene_in_Gene_body_CLASS: Ord.factor w/ 2 levels \"Variant_outside_gene_body_of_regulated_gene\"<..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "\n",
      " chr [1:88] \"chr1_101744646_C_G\" \"chr1_158613314_G_A\" \"chr1_158638260_G_A\" ...\n",
      "\n",
      " chr [1:906] \"ENSG00000117543\" \"ENSG00000162694\" \"ENSG00000235795\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    " #### READ Table_S7 ----\n",
    "  \n",
    "  Table_S7<-readRDS(file=\"/group/soranzo/manuel.tardaguila/Paper_bits/FIX_TABLES/Provisional_Tables/Table_S7_Provisional.rds\")\n",
    "  \n",
    "  cat(\"Table_S7_0\\n\")\n",
    "  cat(str(Table_S7))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S7$VAR)))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S7$ensembl_gene_id)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  Table_S7$Analysis[which(Table_S7$Analysis == 'DTU')]<-'ATU'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "607d9065-d45a-498c-9536-ba245167e844",
   "metadata": {},
   "source": [
    "# General path graphs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "a3781acc-b3cd-4c77-9c00-3d76eb65895a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.567498Z",
     "iopub.status.busy": "2025-07-06T21:41:31.566341Z",
     "iopub.status.idle": "2025-07-06T21:41:31.579365Z",
     "shell.execute_reply": "2025-07-06T21:41:31.578089Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_graphs<-paste('/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/test/','Graphs_selected','/',sep='')\n",
    "  \n",
    "  if (file.exists(path_graphs)){\n",
    "    \n",
    "    \n",
    "  }else{\n",
    "    \n",
    "    dir.create(file.path(path_graphs))\n",
    "    \n",
    "  }#path_graphs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1d2b1ff-8020-4598-af79-b0b17fd649ee",
   "metadata": {},
   "source": [
    "# LOOP to print graphs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "131a952c-fb4a-4efa-a555-650dcbfbbb94",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.582635Z",
     "iopub.status.busy": "2025-07-06T21:41:31.581493Z",
     "iopub.status.idle": "2025-07-06T21:41:31.599353Z",
     "shell.execute_reply": "2025-07-06T21:41:31.598064Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VARS_0\n",
      " chr [1:20] \"chr22_28761148_C_T\" \"chr1_92925654_G_C\" \"chr1_198680015_G_A\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "VARS<-unique(dataframe_to_print_SIT_plots$VAR)\n",
    "\n",
    "\n",
    "\n",
    "cat(\"VARS_0\\n\")\n",
    "cat(str(VARS))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "6c097343-cc7c-4eed-ba68-288fa877db9d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.602583Z",
     "iopub.status.busy": "2025-07-06T21:41:31.601439Z",
     "iopub.status.idle": "2025-07-06T21:41:31.610285Z",
     "shell.execute_reply": "2025-07-06T21:41:31.609125Z"
    }
   },
   "outputs": [],
   "source": [
    "START<-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c51d6328-8f1a-4cf1-82d1-1b2c1d5903b7",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.613580Z",
     "iopub.status.busy": "2025-07-06T21:41:31.612450Z",
     "iopub.status.idle": "2025-07-06T21:41:31.620348Z",
     "shell.execute_reply": "2025-07-06T21:41:31.619237Z"
    }
   },
   "outputs": [],
   "source": [
    "#START<-which(VARS == 'chr7_101499930_G_A')\n",
    "#START<-which(VARS == 'chr9_135864513_C_T')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e0b75116-d26a-4577-a811-5f210709206a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-06T21:41:31.624986Z",
     "iopub.status.busy": "2025-07-06T21:41:31.623886Z",
     "iopub.status.idle": "2025-07-06T21:42:09.412350Z",
     "shell.execute_reply": "2025-07-06T21:42:09.410967Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t1\trs62237617\tchr22_28761148_C_T\t1\tlymph__304\tchr22\t28374004\t29185283\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t1\trs62237617\tchr22_28761148_C_T\t2\tpct__350\tchr22\t28374004\t29076293\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t2\trs150649461\tchr1_92925654_G_C\t1\tmono__27\tchr1\t92675654\t93258159\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t3\trs12733073\tchr1_198680015_G_A\t1\tlymph__38\tchr1\t198372447\t198930015\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t4\trs147176673\tchr1_202129205_G_A\t1\tpct__44\tchr1\t201865580\t202561834\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t5\trs150221884\tchr2_24091099_C_T\t1\tret__209\tchr2\t23607896\t24583743\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t6\trs183347927\tchr2_74920648_G_A\t1\tmpv__248\tchr2\t74652963\t75170648\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message:\n",
      "“ggrepel: 6 unlabeled data points (too many overlaps). Consider increasing max.overlaps”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t7\trs114050631\tchr2_219020958_C_T\t1\tneut__229\tchr2\t218664512\t219433421\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t7\trs114050631\tchr2_219020958_C_T\t2\tbaso__92\tchr2\t218664512\t219433421\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t8\trs182089148\tchr3_46354444_C_T\t1\tmono__325\tchr3\t45864808\t46694405\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t9\trs35592432\tchr3_71355240_G_C\t1\tlymph__328\tchr3\t71003722\t71676419\tnaive T-CD4 Cells\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t10\trs191451841\tchr3_128317978_C_T\t1\teo__306\tchr3\t127871867\t128573829\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t11\trs112097551\tchr3_128322617_G_A\t1\teo__306\tchr3\t127871867\t128573829\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t12\trs572476245\tchr3_184091102_T_G\t1\tplt__411\tchr3\t183833140\t184402546\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message:\n",
      "“ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t13\trs139141690\tchr7_101499930_G_A\t1\tmpv__423\tchr7\t101247358\t101964212\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t14\trs149489081\tchr8_41589736_T_G\t1\tret__395\tchr8\t41262581\t41914119\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t15\trs187715179\tchr9_135920196_C_T\t1\tmpv__462\tchr9\t135894816\t136395932\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message:\n",
      "“ggrepel: 8 unlabeled data points (too many overlaps). Consider increasing max.overlaps”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t16\trs112401631\tchr17_38764524_T_A\t1\tlymph__189\tchr17\t38459270\t39019534\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t16\trs112401631\tchr17_38764524_T_A\t2\teo__172\tchr17\t38465179\t39019534\tMonocytes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t17\trs17758695\tchr18_60920854_C_T\t1\trbc__248\tchr18\t60667873\t61172318\tnaive T-CD4 Cells\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t18\trs569022307\tchr19_15653669_T_C\t1\tlymph__218\tchr19\t15315646\t15908339\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t19\trs543594419\tchr22_50949811_T_C\t1\tmcv__339\tchr22\t50639063\t51241033\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message:\n",
      "“ggrepel: 8 unlabeled data points (too many overlaps). Consider increasing max.overlaps”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t20\trs150813342\tchr9_135864513_C_T\t1\tmpv__461\tchr9\t135600965\t136115920\tWhole blood\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving 6.67 x 6.67 in image\n",
      "\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n",
      "Warning message in as_grob.default(plot):\n",
      "“Cannot convert object of class logical into a grob.”\n"
     ]
    }
   ],
   "source": [
    " DEBUG<-0\n",
    "  \n",
    "  for(i in START:length(VARS))\n",
    "  {\n",
    "    VAR_sel<-VARS[i]\n",
    "      \n",
    "    dataframe_to_print_SIT_plots_sel<-dataframe_to_print_SIT_plots[which(dataframe_to_print_SIT_plots$VAR == VAR_sel),]\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"dataframe_to_print_SIT_plots_sel_0\\n\")\n",
    "      cat(str(dataframe_to_print_SIT_plots_sel))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "   \n",
    "    \n",
    "    AS_ID_array<-unique(dataframe_to_print_SIT_plots_sel$AS_ID)\n",
    "    \n",
    "    # AS_ID_array<-'mpv__423'\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"AS_ID_array_0\\n\")\n",
    "      cat(str(AS_ID_array))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    rs_sel<-unique(dataframe_to_print_SIT_plots_sel$rs)\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"rs_sel_0\\n\")\n",
    "      cat(str(rs_sel))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "\n",
    "    path_graphs_rs<-paste(path_graphs,paste(rs_sel,VAR_sel, sep='__'),'/',sep='')\n",
    "\n",
    "    if (file.exists(path_graphs_rs)){\n",
    "\n",
    "\n",
    "    }else{\n",
    "\n",
    "      dir.create(file.path(path_graphs_rs))\n",
    "\n",
    "    }#path_graphs_rs\n",
    "\n",
    "\n",
    "       for(k in 1:length(AS_ID_array))\n",
    "        {\n",
    "          AS_ID_sel<-AS_ID_array[k]\n",
    "\n",
    "           dataframe_to_print_SIT_plots_sel_AS_sel<-dataframe_to_print_SIT_plots[which(dataframe_to_print_SIT_plots$VAR == VAR_sel &\n",
    "                                                                                    dataframe_to_print_SIT_plots$AS_ID == AS_ID_sel),]\n",
    "      \n",
    "      \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"dataframe_to_print_SIT_plots_sel_AS_sel_0\\n\")\n",
    "                cat(str(dataframe_to_print_SIT_plots_sel_AS_sel))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "              \n",
    "              source_sel<-unique(dataframe_to_print_SIT_plots_sel_AS_sel$RNA_source)\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"source_sel_0\\n\")\n",
    "                cat(str(source_sel))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "             \n",
    "              \n",
    "              BLOCK_POST_sel<-BLOCK_POST[which(BLOCK_POST$AS_ID == AS_ID_sel),]\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"BLOCK_POST_sel_0\\n\")\n",
    "                cat(str(BLOCK_POST_sel))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "              \n",
    "              chr_sel<-unique(BLOCK_POST_sel$chr)\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"chr_sel_0\\n\")\n",
    "                cat(str(chr_sel))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "              \n",
    "              start_sel<-unique(BLOCK_POST_sel$start)\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"start_sel_0\\n\")\n",
    "                cat(str(start_sel))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "              \n",
    "              end_sel<-unique(BLOCK_POST_sel$end)\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"end_sel_0\\n\")\n",
    "                cat(str(end_sel))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "              \n",
    "              cat(\"---------------------------------------------------------------------------------->\\t\")\n",
    "              cat(sprintf(as.character(i)))\n",
    "              cat(\"\\t\")\n",
    "              cat(sprintf(as.character(rs_sel)))\n",
    "              cat(\"\\t\")\n",
    "              cat(sprintf(as.character(VAR_sel)))\n",
    "              cat(\"\\t\")\n",
    "              cat(sprintf(as.character(k)))\n",
    "              cat(\"\\t\")\n",
    "              cat(sprintf(as.character(AS_ID_sel)))\n",
    "              cat(\"\\t\")\n",
    "              cat(sprintf(as.character(chr_sel)))\n",
    "              cat(\"\\t\")\n",
    "              cat(sprintf(as.character(start_sel)))\n",
    "              cat(\"\\t\")\n",
    "              cat(sprintf(as.character(end_sel)))\n",
    "              cat(\"\\t\")\n",
    "              cat(sprintf(as.character(source_sel)))\n",
    "              cat(\"\\n\")\n",
    "\n",
    "\n",
    "               path_graphs_rs_AS<-paste(path_graphs_rs,AS_ID_sel,'/',sep='')\n",
    "\n",
    "                if (file.exists(path_graphs_rs_AS)){\n",
    "            \n",
    "            \n",
    "                }else{\n",
    "            \n",
    "                  dir.create(file.path(path_graphs_rs_AS))\n",
    "            \n",
    "                }#path_graphs_rs\n",
    "\n",
    "\n",
    "               variants_file_sel<-unique(variants_file[which(variants_file$AS_ID == AS_ID_sel),])\n",
    "      \n",
    "                  variants_file_sel$is_cond_ind<-factor(variants_file_sel$is_cond_ind,\n",
    "                                                        levels=c(\"0\",\"1\"),\n",
    "                                                        ordered=T)\n",
    "                  \n",
    "                  if(DEBUG == 1)\n",
    "                  {\n",
    "                    cat(\"variants_file_sel_0\\n\")\n",
    "                    cat(str(variants_file_sel))\n",
    "                    cat(\"\\n\")\n",
    "                    cat(str(unique(variants_file_sel$VAR)))\n",
    "                    cat(\"\\n\")\n",
    "                  }\n",
    "\n",
    "                 #### PLOT Effect size -----------------------------------------------------------------------------------\n",
    "      \n",
    "     \n",
    "      \n",
    "                          phenotype_sel<-unique(variants_file_sel$phenotype)\n",
    "                          \n",
    "                          if(DEBUG == 1)\n",
    "                          {\n",
    "                            cat(\"phenotype_sel_0\\n\")\n",
    "                            cat(str(phenotype_sel))\n",
    "                            cat(\"\\n\")\n",
    "                          }\n",
    "                          \n",
    "                          phenotype_DEF_sel<-unique(variants_file_sel$phenotype_DEF)\n",
    "                          \n",
    "                          if(DEBUG == 1)\n",
    "                          {\n",
    "                            cat(\"phenotype_DEF_sel_0\\n\")\n",
    "                            cat(str(phenotype_DEF_sel))\n",
    "                            cat(\"\\n\")\n",
    "                          }\n",
    "                          \n",
    "                          \n",
    "                          indx_min<-which(colnames(variants_file_sel) == 'min')\n",
    "                          indx_max<-which(colnames(variants_file_sel) == 'max')\n",
    "                          \n",
    "                          \n",
    "                          \n",
    "                          vector_SE<-c(variants_file_sel[,indx_min][!is.na(variants_file_sel[,indx_min])],\n",
    "                                       variants_file_sel[,indx_max][!is.na(variants_file_sel[,indx_max])])\n",
    "                          \n",
    "                          \n",
    "                          summary_vector_SE<-summary(vector_SE)\n",
    "                          \n",
    "                          \n",
    "                          \n",
    "                          if(DEBUG == 1)\n",
    "                          {\n",
    "                            cat(\"vector_SE\\n\")\n",
    "                            cat(sprintf(as.character(vector_SE)))\n",
    "                            cat(\"\\n\")\n",
    "                            cat(\"summary_vector_SE\\n\")\n",
    "                            cat(sprintf(as.character(names(summary_vector_SE))))\n",
    "                            cat(\"\\n\")\n",
    "                            cat(sprintf(as.character(summary_vector_SE)))\n",
    "                            cat(\"\\n\")\n",
    "                          }\n",
    "                          \n",
    "                          max_value<-summary_vector_SE[6]\n",
    "                          min_value<-summary_vector_SE[1]\n",
    "                          \n",
    "                       \n",
    "                          beta_span<-sort(as.numeric(unlist(strsplit(dataframe_to_print_SIT_plots_sel_AS_sel$beta_span, split=\";\"))))\n",
    "                          \n",
    "                          if(DEBUG == 1)\n",
    "                          {\n",
    "                            cat(\"beta_span\\n\")\n",
    "                            cat(sprintf(as.character(beta_span)))\n",
    "                            cat(\"\\n\")\n",
    "                          }\n",
    "                             \n",
    "                       \n",
    "                          #breaks_Beta<-unique(sort(unique(c(0,min_value,max_value,seq(min_value,max_value, by=step)))))\n",
    "                          #breaks_Beta<-unique(sort(unique(c(0,-0.1,0.1,min_value,max_value))))\n",
    "                         # breaks_Beta<-unique(sort(unique(c(0,-0.1,0.1,-0.2,0.2,0.3,-0.3,min_value,max_value))))\n",
    "                          breaks_Beta<-unique(sort(unique(beta_span)))\n",
    "                          labels_Beta<-as.character(round(breaks_Beta,2))\n",
    "                          \n",
    "                          limit_min<-min(min_value,beta_span)\n",
    "                          limit_max<-max(max_value,beta_span)\n",
    "                          \n",
    "                          if(DEBUG == 1)\n",
    "                          {\n",
    "                          \n",
    "                            cat(\"breaks_Beta\\n\")\n",
    "                            cat(sprintf(as.character(breaks_Beta)))\n",
    "                            cat(\"\\n\")\n",
    "                            cat(\"labels_Beta\\n\")\n",
    "                            cat(sprintf(as.character(labels_Beta)))\n",
    "                            cat(\"\\n\")\n",
    "                          }\n",
    "                          \n",
    "                    \n",
    "                          # geom_segment(data=variants_file_sel,\n",
    "                          #              aes(y=min,\n",
    "                          #                  yend=max,\n",
    "                          #                  x=pos37,\n",
    "                          #                  xend=pos37),\n",
    "                          #              color=\"black\", size=2)+\n",
    "                            \n",
    "                          \n",
    "                          finemap_beta_dot_plot<-ggplot(data=variants_file_sel,\n",
    "                                          aes(x=pos37))+\n",
    "                            geom_segment(aes(y=breaks_Beta[1],\n",
    "                                         yend=finemap_beta,\n",
    "                                         x=pos37,\n",
    "                                         xend=pos37), size=0.5,color=\"gray\",linetype=\"dashed\")+\n",
    "                            scale_x_continuous(name=NULL,breaks=c(start_sel,end_sel),labels=as.character(c(start_sel,end_sel)),\n",
    "                                               limits=c(start_sel-1,end_sel+1))+\n",
    "                            geom_rect(data=variants_file_sel,\n",
    "                                      aes(ymin=min,\n",
    "                                          ymax=max,\n",
    "                                          xmin=pos37-10,\n",
    "                                          xmax=pos37+10),\n",
    "                                      color=\"black\")+\n",
    "                            geom_point(data=variants_file_sel,\n",
    "                                       aes(x=pos37,\n",
    "                                           y=finemap_beta,\n",
    "                                           fill=Variant_classification,\n",
    "                                           shape=is_cond_ind),\n",
    "                                       size=1.5, stroke=0.5)+\n",
    "                            scale_shape_manual(values=c(21,23), drop=F)+\n",
    "                            scale_fill_manual(values=vector_colors, drop=F)+\n",
    "                            scale_y_continuous(name=paste(\"beta (s.d.)\",phenotype_DEF_sel,sep=' '),breaks=breaks_Beta,labels=labels_Beta,\n",
    "                                               limits=c(limit_min,limit_max))+\n",
    "                            theme_classic()+\n",
    "                            theme(plot.title=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.title.y=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.title.x=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.text.y=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.text.x=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.line.x = element_line(size = 0.2),\n",
    "                                  axis.ticks.x = element_line(size = 0.2),\n",
    "                                  axis.line.y = element_line(size = 0.2),\n",
    "                                  axis.ticks.y = element_line(size = 0.2))+\n",
    "                            theme(legend.title = element_blank(),\n",
    "                                  legend.text = element_text(size=6),\n",
    "                                  legend.position=\"hidden\")+\n",
    "                            geom_hline(yintercept=0, color=\"black\",size=0.25)+\n",
    "                            ggeasy::easy_center_title()\n",
    "                          \n",
    "                          finemap_beta_dot_plot<-finemap_beta_dot_plot+\n",
    "                            geom_text_repel(data=variants_file_sel,\n",
    "                                          aes(x=pos37,\n",
    "                                              y=finemap_beta,\n",
    "                                              label=rs),\n",
    "                                          box.padding = 1,\n",
    "                                          color='black',\n",
    "                                          max.overlaps = Inf,\n",
    "                                          show.legend = FALSE,\n",
    "                                          size=2)\n",
    "                          \n",
    "                          if(DEBUG == 1)\n",
    "                          {\n",
    "                            cat(\"Effect size plot\\n\")\n",
    "                            \n",
    "                          }\n",
    "                          \n",
    "                          \n",
    "                          finemap_beta_dot_plot2<-ggplot(data=variants_file_sel,\n",
    "                                                         aes(x=pos37))+\n",
    "                            scale_x_continuous(name=NULL,breaks=c(start_sel,end_sel),labels=as.character(c(start_sel,end_sel)),\n",
    "                                               limits=c(start_sel-1,end_sel+1))+\n",
    "                            geom_point(data=variants_file_sel,\n",
    "                                       aes(x=pos37,\n",
    "                                           y=finemap_beta,\n",
    "                                           color=Variant_classification,\n",
    "                                           fill=Variant_classification,\n",
    "                                           shape=is_cond_ind),\n",
    "                                       size=1.5, stroke=0.5)+\n",
    "                            scale_shape_manual(values=c(21,23), drop=F)+\n",
    "                            scale_color_manual(values=vector_colors, drop=F)+\n",
    "                            scale_fill_manual(values=vector_colors, drop=F)+\n",
    "                            scale_y_continuous(name=paste(\"beta (s.d.)\",phenotype_DEF_sel,sep=' '),breaks=breaks_Beta,labels=labels_Beta,\n",
    "                                               limits=c(limit_min,limit_max))+\n",
    "                            geom_hline(yintercept=0, color=\"black\",size=0.25)+\n",
    "                            geom_text_repel(data=variants_file_sel,\n",
    "                                            aes(x=pos37,\n",
    "                                                y=finemap_beta,\n",
    "                                                label=rs),\n",
    "                                            box.padding = 1,\n",
    "                                            color='black',\n",
    "                                            max.overlaps = Inf,\n",
    "                                            show.legend = FALSE,\n",
    "                                            size=2)+\n",
    "                            theme_classic()+\n",
    "                            theme(plot.title=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.title.y=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.title.x=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.text.y=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.text.x=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                                  axis.line.x = element_line(size = 0.2),\n",
    "                                  axis.ticks.x = element_line(size = 0.2),\n",
    "                                  axis.line.y = element_line(size = 0.2),\n",
    "                                  axis.ticks.y = element_line(size = 0.2))+\n",
    "                            theme(legend.title = element_text(size=6),\n",
    "                                  legend.text = element_text(size=6),\n",
    "                                  legend.key.size = unit(0.5, 'cm'), #change legend key size\n",
    "                                  legend.key.height = unit(0.5, 'cm'), #change legend key height\n",
    "                                  legend.key.width = unit(0.5, 'cm'), #change legend key width\n",
    "                                  legend.position=\"right\")+\n",
    "                            ggeasy::easy_center_title()\n",
    "                    \n",
    "                          if(DEBUG == 1)\n",
    "                          {\n",
    "                            cat(\"Effect size plot 2\\n\")\n",
    "                            \n",
    "                          }\n",
    "                          \n",
    "                          \n",
    "                      \n",
    "                          setwd(path_graphs_rs_AS)\n",
    "                          \n",
    "                          svgname<-paste('Finemap_Beta_COORD_order_2',\".svg\",sep='')\n",
    "                          makesvg = TRUE\n",
    "                          \n",
    "                          if (makesvg == TRUE)\n",
    "                          {\n",
    "                            ggsave(svgname, plot= finemap_beta_dot_plot2,\n",
    "                                   device=\"svg\")\n",
    "                          }\n",
    "                          \n",
    "                          if(DEBUG == 1)\n",
    "                          {\n",
    "                            cat(\"Finemap_Beta_COORD_order plot 2\\n\")\n",
    "                            \n",
    "                          }\n",
    "\n",
    "           ############### ---------------------------------------------------------------------------------------\n",
    "\n",
    "   \n",
    "     \n",
    "      \n",
    "          gene_tracks_sel<-gene_tracks[which(gene_tracks$AS_ID == AS_ID_sel),]\n",
    "          \n",
    "          if(DEBUG == 1)\n",
    "          {\n",
    "            cat(\"gene_tracks_sel_0\\n\")\n",
    "            cat(str(gene_tracks_sel))\n",
    "            cat(\"\\n\")\n",
    "          }\n",
    "      \n",
    "              if(dim(gene_tracks_sel)[1] >0)\n",
    "              {\n",
    "                finemap_beta_dot_plot<-finemap_beta_dot_plot+\n",
    "                  theme(axis.text.x=element_blank(),\n",
    "                        axis.line.x = element_blank(),\n",
    "                        axis.ticks.x = element_blank())\n",
    "                \n",
    "                \n",
    "                \n",
    "                Table_S7_DE_sel<-Table_S7[which(Table_S7$VAR == VAR_sel &\n",
    "                                               Table_S7$Analysis == 'DE'),]\n",
    "                \n",
    "                \n",
    "                if(dim(Table_S7_DE_sel)[1] >0)\n",
    "                {\n",
    "                  \n",
    "                }else{\n",
    "                  rescue_haplotypes_sel<-rescue_haplotypes[which(rescue_haplotypes$Proxy_VAR == VAR_sel &\n",
    "                                                                   rescue_haplotypes$ensembl_gene_id%in%gene_tracks_sel$ensembl_gene_id &\n",
    "                                                                   rescue_haplotypes$comparison == 'HOM_REF__HET|HET'),]\n",
    "                  \n",
    "                  if(DEBUG == 1)\n",
    "                  {\n",
    "                    cat(\"rescue_haplotypes_sel_0\\n\")\n",
    "                    cat(str(rescue_haplotypes_sel))\n",
    "                    cat(\"\\n\")\n",
    "                    cat(str(unique(rescue_haplotypes_sel$ensembl_gene_id)))\n",
    "                    cat(\"\\n\")\n",
    "                    cat(str(unique(rescue_haplotypes_sel$VAR)))\n",
    "                    cat(\"\\n\")\n",
    "                    cat(str(unique(rescue_haplotypes_sel$Proxy_VAR)))\n",
    "                    cat(\"\\n\")\n",
    "                    cat(str(unique(rescue_haplotypes_sel$comparison)))\n",
    "                    cat(\"\\n\")\n",
    "                  }\n",
    "                  \n",
    "                  my_DF<- data.frame(matrix(vector(), dim(rescue_haplotypes_sel)[1], dim(Table_S7)[2],\n",
    "                                            dimnames=list(c(),\n",
    "                                                          colnames(Table_S7))),  stringsAsFactors=F)\n",
    "                  \n",
    "                  \n",
    "                  if(DEBUG == 1)\n",
    "                  {\n",
    "                    cat(\"my_DF_0\\n\")\n",
    "                    cat(str(my_DF))\n",
    "                    cat(\"\\n\")\n",
    "                    \n",
    "                  }\n",
    "                  \n",
    "                  my_DF$ensembl_gene_id<-rescue_haplotypes_sel$ensembl_gene_id\n",
    "                  my_DF$RNASeq_source<-'Whole blood'\n",
    "                  my_DF$Analysis<-'DE'\n",
    "                  my_DF$HGNC<-rescue_haplotypes_sel$HGNC\n",
    "                  my_DF$Beta<-rescue_haplotypes_sel$Block_PCHiC_Beta\n",
    "                  my_DF$Beta_Z_score<-rescue_haplotypes_sel$Block_PCHiC_Beta_Z_score\n",
    "                  my_DF$adjusted_pval<-rescue_haplotypes_sel$Block_PCHiC_minuslogpvalue\n",
    "                  my_DF$adjusted_minus_logpval<-rescue_haplotypes_sel$Block_PCHiC_minuslogpvalue\n",
    "                  \n",
    "                  \n",
    "                  if(DEBUG == 1)\n",
    "                  {\n",
    "                    cat(\"my_DF_1\\n\")\n",
    "                    cat(str(my_DF))\n",
    "                    cat(\"\\n\")\n",
    "                    \n",
    "                  }\n",
    "                    \n",
    "                  Table_S7_DE_sel<-my_DF\n",
    "                  \n",
    "               \n",
    "                }#dim(Table_S7_DE_sel)[1] >0\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"Table_S7_DE_sel_0\\n\")\n",
    "                  cat(str(Table_S7_DE_sel))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                Table_S7_DE_sel<-Table_S7_DE_sel[which(Table_S7_DE_sel$RNASeq_source == source_sel),]\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"Table_S7_DE_sel_0.5\\n\")\n",
    "                  cat(str(Table_S7_DE_sel))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "               \n",
    "                \n",
    "                Table_S7_DE_sel$Beta_Z_score<-round(Table_S7_DE_sel$Beta_Z_score,2)\n",
    "                \n",
    "                A<-round(summary(Table_S7_DE_sel$Beta_Z_score[!is.na(Table_S7_DE_sel$Beta_Z_score)]),4)\n",
    "                \n",
    "                max_Beta_Z_score<-max(A)\n",
    "                min_Beta_Z_score<-min(A)\n",
    "                \n",
    "                abs_max_Beta_Z_score<-abs(max_Beta_Z_score)\n",
    "                abs_min_Beta_Z_score<-abs(min_Beta_Z_score)\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"max_Beta_Z_score_0\\n\")\n",
    "                  cat(str(max_Beta_Z_score))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                  cat(\"min_Beta_Z_score_0\\n\")\n",
    "                  cat(str(min_Beta_Z_score))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                  cat(\"abs_max_Beta_Z_score_0\\n\")\n",
    "                  cat(str(abs_max_Beta_Z_score))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                  cat(\"abs_min_Beta_Z_score_0\\n\")\n",
    "                  cat(str(abs_min_Beta_Z_score))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                if(abs_max_Beta_Z_score > abs_min_Beta_Z_score)\n",
    "                {\n",
    "                  step<-abs(max_Beta_Z_score - -1*max_Beta_Z_score)/4\n",
    "                  breaks.EXP<-sort(unique(c(0,seq(from= -1*abs_max_Beta_Z_score, to=abs_max_Beta_Z_score,by=step))))\n",
    "                  breaks.EXP<-sort(unique(c(0,-1*abs_max_Beta_Z_score, abs_max_Beta_Z_score)))\n",
    "                  labels.EXP<-as.character(round(breaks.EXP,2))\n",
    "                  \n",
    "                }else{\n",
    "                  \n",
    "                  step<-abs(min_Beta_Z_score - -1*min_Beta_Z_score)/4\n",
    "                  breaks.EXP<-sort(unique(c(0,seq(from= -1*abs_min_Beta_Z_score, to=abs_min_Beta_Z_score,by=step))))\n",
    "                  breaks.EXP<-sort(unique(c(0,-1*abs_min_Beta_Z_score,abs_min_Beta_Z_score,by=step)))\n",
    "                  labels.EXP<-as.character(round(breaks.EXP,2))\n",
    "                  \n",
    "                }#abs_max_Beta_Z_score > abs_min_Beta_Z_score\n",
    "               \n",
    "                \n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"labels.EXP:\\t\")\n",
    "                  cat(sprintf(as.character(labels.EXP)))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                  \n",
    "                  \n",
    "                  # quit(status = 1)\n",
    "                }\n",
    "                \n",
    "                \n",
    "             \n",
    "                \n",
    "                \n",
    "                \n",
    "                indx.negative.strand<-which(gene_tracks_sel$strand == '-')\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"indx.negative.strand_0\\n\")\n",
    "                  cat(str(indx.negative.strand))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "            \n",
    "                \n",
    "                indx.positive.strand<-which(gene_tracks_sel$strand == '+')\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"indx.positive.strand_0\\n\")\n",
    "                  cat(str(indx.positive.strand))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "            \n",
    "                \n",
    "                gene_tracks_sel$COORD<-NA\n",
    "                \n",
    "                gene_tracks_sel$COORD[indx.positive.strand]<-1\n",
    "                gene_tracks_sel$COORD[indx.negative.strand]<-0\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"gene_tracks_sel_1\\n\")\n",
    "                  cat(str(gene_tracks_sel))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                gene_tracks_sel$definitive_start<-NA\n",
    "                gene_tracks_sel$definitive_end<-NA\n",
    "                \n",
    "                gene_tracks_sel$definitive_start[indx.positive.strand]<-gene_tracks_sel$start[indx.positive.strand]\n",
    "                gene_tracks_sel$definitive_end[indx.positive.strand]<-gene_tracks_sel$end[indx.positive.strand]\n",
    "                \n",
    "                \n",
    "                gene_tracks_sel$definitive_start[indx.negative.strand]<-gene_tracks_sel$end[indx.negative.strand]\n",
    "                gene_tracks_sel$definitive_end[indx.negative.strand]<-gene_tracks_sel$start[indx.negative.strand]\n",
    "                \n",
    "                gene_tracks_sel$ymin<-gene_tracks_sel$COORD\n",
    "                gene_tracks_sel$ymax<-gene_tracks_sel$COORD+0.2\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"gene_tracks_sel_2\\n\")\n",
    "                  cat(str(gene_tracks_sel))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                  cat(\"ymin_0\\n\")\n",
    "                  cat(sprintf(as.character(names(summary(gene_tracks_sel$ymin)))))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(summary(gene_tracks_sel$ymin))))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                  cat(\"ymax_0\\n\")\n",
    "                  cat(sprintf(as.character(names(summary(gene_tracks_sel$ymax)))))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(summary(gene_tracks_sel$ymax))))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                \n",
    "                \n",
    "                \n",
    "                \n",
    "                \n",
    "        \n",
    "                random_vec <- round(runif(n=dim(gene_tracks_sel)[1], min=-0.25, max=0.25),2)\n",
    "                \n",
    "                gene_tracks_sel$ymin<-gene_tracks_sel$ymin+random_vec\n",
    "                gene_tracks_sel$ymax<-gene_tracks_sel$ymax+random_vec\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  # cat(\"jitter_pos_0\\n\")\n",
    "                  # cat(str(jitter_pos))\n",
    "                  # cat(\"\\n\")\n",
    "                  \n",
    "                  cat(\"random_vec_0\\n\")\n",
    "                  cat(str(random_vec))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                  cat(\"ymin_1\\n\")\n",
    "                  cat(sprintf(as.character(names(summary(gene_tracks_sel$ymin)))))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(summary(gene_tracks_sel$ymin))))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                  cat(\"ymax_1\\n\")\n",
    "                  cat(sprintf(as.character(names(summary(gene_tracks_sel$ymax)))))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(summary(gene_tracks_sel$ymax))))\n",
    "                  cat(\"\\n\")\n",
    "                 \n",
    "                  \n",
    "                }\n",
    "                \n",
    "                Table_S7_DE_sel<-merge(Table_S7_DE_sel,\n",
    "                                       gene_tracks_sel,\n",
    "                                       by='ensembl_gene_id',\n",
    "                                       all.y=T)\n",
    "                \n",
    "                Table_S7_DE_sel$SIG<-NA\n",
    "                \n",
    "                Table_S7_DE_sel$SIG[which(Table_S7_DE_sel$adjusted_minus_logpval >= 1.3)]<-'YES'\n",
    "                Table_S7_DE_sel$SIG[which(Table_S7_DE_sel$adjusted_minus_logpval < 1.3)]<-'NO'\n",
    "                \n",
    "                Table_S7_DE_sel$SIG<-factor(Table_S7_DE_sel$SIG,\n",
    "                                            levels=c('YES','NO'),\n",
    "                                            ordered=T)\n",
    "                \n",
    "                \n",
    "                \n",
    "                \n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"Table_S7_DE_sel_1\\n\")\n",
    "                  cat(str(Table_S7_DE_sel))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                # geom_point(data=variants_file_sel,\n",
    "                #            aes(x=pos37,\n",
    "                #                y=1.5,\n",
    "                #                fill=Variant_classification,\n",
    "                #                shape=is_cond_ind),\n",
    "                #            size=1.5, stroke=0.5)+\n",
    "                #   scale_shape_manual(values=c(21,23), drop=F)+\n",
    "                #   scale_fill_manual(values=vector_colors, drop=F)+\n",
    "                #   new_scale(\"fill\")+\n",
    "                \n",
    "                # geom_rect(data=gene_tracks_sel,\n",
    "                #           aes(ymin=ymin,\n",
    "                #               ymax=ymax,\n",
    "                #               xmin=definitive_start,\n",
    "                #               xmax=definitive_end), fill=\"gray\", color='white',size=0.2)+\n",
    "                \n",
    "                \n",
    "                graph_genes<-ggplot()+\n",
    "                  scale_x_continuous(name=NULL,breaks=c(start_sel,end_sel),\n",
    "                                     labels=as.character(c(start_sel,end_sel)),\n",
    "                                     limits=c(start_sel-1,end_sel+1))+\n",
    "                  scale_y_continuous(name=NULL, \n",
    "                                     breaks=seq(0,1,by=1),\n",
    "                                     labels=as.character(c('gDNA -','gDNA +')), \n",
    "                                     limits=c(-0.5,1.5))+\n",
    "                  geom_segment(data=variants_file_sel,\n",
    "                               aes(y=-0.5,\n",
    "                                   yend=1.5,\n",
    "                                   x=pos37,\n",
    "                                   xend=pos37), size=0.5,color=\"gray\",linetype=\"dashed\")+\n",
    "                  geom_rect(data=Table_S7_DE_sel,\n",
    "                            aes(ymin=ymin,\n",
    "                                ymax=ymax,\n",
    "                                xmin=definitive_start,\n",
    "                                xmax=definitive_end,\n",
    "                                fill=Beta_Z_score), color=NA,size=0.2)+\n",
    "                  scale_fill_gradient2(\n",
    "                    midpoint = 0,\n",
    "                    low = \"blue\",\n",
    "                    mid = \"white\",\n",
    "                    high = \"red\",\n",
    "                    breaks=breaks.EXP,labels=labels.EXP,\n",
    "                    limits=c(breaks.EXP[1],breaks.EXP[length(breaks.EXP)]),\n",
    "                    name=paste('Beta',\"Z-score\",sep=\"\\n\"),na.value = \"gray\")+\n",
    "                  new_scale(\"fill\")+\n",
    "                  geom_rect(data=Table_S7_DE_sel[is.na(Table_S7_DE_sel$SIG),],\n",
    "                            aes(ymin=ymin,\n",
    "                                ymax=ymax,\n",
    "                                xmin=definitive_start,\n",
    "                                xmax=definitive_end), fill= NA, color='black',size=0.05)+\n",
    "                  geom_rect(data=Table_S7_DE_sel[which(Table_S7_DE_sel$SIG == 'NO'),],\n",
    "                            aes(ymin=ymin,\n",
    "                                ymax=ymax,\n",
    "                                xmin=definitive_start,\n",
    "                                xmax=definitive_end), fill= NA, color='black',size=0.05)+\n",
    "                  geom_rect(data=Table_S7_DE_sel[which(Table_S7_DE_sel$SIG == 'YES'),],\n",
    "                            aes(ymin=ymin,\n",
    "                                ymax=ymax,\n",
    "                                xmin=definitive_start,\n",
    "                                xmax=definitive_end), fill= NA, color='black',size=0.25)+\n",
    "                  geom_hline(yintercept=0.5, color=\"black\",size=0.25)+\n",
    "                  theme_classic()+\n",
    "                  theme(plot.title=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                        axis.title.y=element_blank(),\n",
    "                        axis.title.x=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                        axis.text.y=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                        axis.text.x=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                        axis.line.x = element_line(size = 0.2),\n",
    "                        axis.ticks = element_line(size = 0.2),\n",
    "                        axis.line.y = element_line(size = 0.2))+\n",
    "                  theme(legend.title = element_text(size=6),\n",
    "                        legend.text = element_text(size=6),\n",
    "                        legend.key.size = unit(0.25, 'cm'), #change legend key size\n",
    "                        legend.key.height = unit(0.25, 'cm'), #change legend key height\n",
    "                        legend.key.width = unit(0.25, 'cm'), #change legend key width\n",
    "                        legend.position=\"bottom\")+\n",
    "                  ggeasy::easy_center_title()\n",
    "                \n",
    "                \n",
    "                \n",
    "               \n",
    "                \n",
    "                dataframe_subset_ENSG_NO_NA_sel<-dataframe_subset_ENSG_NO_NA[which(dataframe_subset_ENSG_NO_NA$AS_ID == AS_ID_sel),]\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"dataframe_subset_ENSG_NO_NA_sel_2\\n\")\n",
    "                  cat(str(dataframe_subset_ENSG_NO_NA_sel))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                \n",
    "                indx.selected_genes<-which(gene_tracks_sel$ensembl_gene_id%in%dataframe_subset_ENSG_NO_NA_sel$ensembl_gene_id)\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"indx.selected_genes_0\\n\")\n",
    "                  cat(str(indx.selected_genes))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                graph_genes2<-graph_genes+\n",
    "                  geom_text_repel(data=gene_tracks_sel,\n",
    "                                  aes(x=definitive_start,\n",
    "                                      y=ymax,\n",
    "                                      label=hgnc),\n",
    "                                  # Repel away from the left edge, not from the right.\n",
    "                                  xlim = c(NA, Inf),\n",
    "                                  # Do not repel from top or bottom edges.\n",
    "                                  ylim = c(-Inf, Inf),\n",
    "                                  family=\"sans\",\n",
    "                                  color='black',\n",
    "                                  fontface='bold.italic',\n",
    "                                  size=2)\n",
    "                \n",
    "                graph_genes<-graph_genes+\n",
    "                  geom_text_repel(data=gene_tracks_sel[indx.selected_genes,],\n",
    "                                  aes(x=definitive_start,\n",
    "                                      y=ymax,\n",
    "                                      label=hgnc),\n",
    "                                  # Repel away from the left edge, not from the right.\n",
    "                                  xlim = c(NA, Inf),\n",
    "                                  # Do not repel from top or bottom edges.\n",
    "                                  ylim = c(-Inf, Inf),\n",
    "                                  family=\"sans\",\n",
    "                                  color='black',\n",
    "                                  fontface='bold.italic',\n",
    "                                  size=2)\n",
    "                \n",
    "                setwd(path_graphs_rs_AS)\n",
    "                \n",
    "                svgname<-paste(paste(\"Genes\",'COORD_order', sep='_'),\".svg\",sep='')\n",
    "                makesvg = TRUE\n",
    "                \n",
    "                if (makesvg == TRUE)\n",
    "                {\n",
    "                  ggsave(svgname, plot= graph_genes2,\n",
    "                         device=\"svg\",\n",
    "                         height=dataframe_to_print_SIT_plots_sel_AS_sel$Height, width=dataframe_to_print_SIT_plots_sel_AS_sel$Width)\n",
    "                }\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"genes plot\\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                \n",
    "             \n",
    "                \n",
    "              }else{\n",
    "                \n",
    "                graph_genes<-NA\n",
    "                \n",
    "              }#dim(gene_tracks_sel)[1] >0\n",
    "\n",
    "           #################### ATAC plot --------------------------------------\n",
    "\n",
    "\n",
    "      \n",
    "      \n",
    "      \n",
    "      \n",
    "              ATAC_tracks_subset.m_sel<-ATAC_tracks_subset.m[which(ATAC_tracks_subset.m$AS_ID == AS_ID_sel),]\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"ATAC_tracks_subset.m_sel_0\\n\")\n",
    "                cat(str(ATAC_tracks_subset.m_sel))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "              \n",
    "                      if(dim(ATAC_tracks_subset.m_sel)[1] >0)\n",
    "                      {\n",
    "                        finemap_beta_dot_plot<-finemap_beta_dot_plot+\n",
    "                          theme(axis.text.x=element_blank(),\n",
    "                                axis.line.x = element_blank(),\n",
    "                                axis.ticks.x = element_blank())\n",
    "                        \n",
    "                        graph_genes<-graph_genes+\n",
    "                          theme(axis.text.x=element_blank(),\n",
    "                                axis.line.x = element_blank(),\n",
    "                                axis.ticks.x = element_blank())\n",
    "                \n",
    "                        dataframe_subset_ATAC_NO_NA_sel<-dataframe_subset_ATAC_NO_NA[which(dataframe_subset_ATAC_NO_NA$AS_ID == AS_ID_sel),]\n",
    "                        \n",
    "                        if(DEBUG == 1)\n",
    "                        {\n",
    "                          cat(\"dataframe_subset_ATAC_NO_NA_sel_0\\n\")\n",
    "                          cat(str(dataframe_subset_ATAC_NO_NA_sel))\n",
    "                          cat(\"\\n\")\n",
    "                        }\n",
    "                        \n",
    "                        \n",
    "                       \n",
    "                      \n",
    "                        \n",
    "                        ATAC_tracks_subset.m_sel$Cell_Type<-factor(ATAC_tracks_subset.m_sel$Cell_Type,\n",
    "                                                                   levels = unique(dataframe_subset_ATAC_NO_NA_sel$Cell_Type),\n",
    "                                                                   ordered=T)\n",
    "                        \n",
    "                        \n",
    "                        if(DEBUG == 1)\n",
    "                        {\n",
    "                          cat(\"ATAC_tracks_subset.m_sel_2\\n\")\n",
    "                          cat(str(ATAC_tracks_subset.m_sel))\n",
    "                          cat(\"\\n\")\n",
    "                          cat(sprintf(as.character(names(summary(ATAC_tracks_subset.m_sel$Cell_Type)))))\n",
    "                          cat(\"\\n\")\n",
    "                          cat(sprintf(as.character(summary(ATAC_tracks_subset.m_sel$Cell_Type))))\n",
    "                          cat(\"\\n\")\n",
    "                          \n",
    "                        }\n",
    "                        \n",
    "                        \n",
    "                        \n",
    "                        ATAC_tracks_subset.m_sel$counts_by_10<-round(ATAC_tracks_subset.m_sel$counts,0)/10\n",
    "                        \n",
    "                        if(DEBUG == 1)\n",
    "                        {\n",
    "                          cat(\"ATAC_tracks_subset.m_sel_3\\n\")\n",
    "                          cat(str(ATAC_tracks_subset.m_sel))\n",
    "                          cat(\"\\n\")\n",
    "                          cat(sprintf(as.character(names(summary(ATAC_tracks_subset.m_sel$Cell_Type)))))\n",
    "                          cat(\"\\n\")\n",
    "                          cat(sprintf(as.character(summary(ATAC_tracks_subset.m_sel$Cell_Type))))\n",
    "                          cat(\"\\n\")\n",
    "                          \n",
    "                        }\n",
    "                        \n",
    "                        \n",
    "                        breaks_counts<-unique(sort(unique(c(0,50,max(ATAC_tracks_subset.m_sel$counts_by_10)))))\n",
    "                        labels_counts<-as.character(round(breaks_counts,1))\n",
    "                        \n",
    "                        if(DEBUG == 1)\n",
    "                        {\n",
    "                          cat(\"step_counts\\n\")\n",
    "                          cat(sprintf(as.character(step)))\n",
    "                          cat(\"\\n\")\n",
    "                          cat(\"breaks_counts\\n\")\n",
    "                          cat(sprintf(as.character(breaks_counts)))\n",
    "                          cat(\"\\n\")\n",
    "                          cat(\"labels_counts\\n\")\n",
    "                          cat(sprintf(as.character(labels_counts)))\n",
    "                          cat(\"\\n\")\n",
    "                        }\n",
    "                         \n",
    "                        ATAC_dot_plot<-ggplot()+\n",
    "                          geom_vline(xintercept=variants_file_sel$pos37, color=\"gray\", linetype='dashed',size=0.5)+\n",
    "                          geom_rect(data=ATAC_tracks_subset.m_sel,\n",
    "                                    aes(xmin=start,\n",
    "                                        xmax=end,\n",
    "                                        ymin=0,\n",
    "                                        ymax=counts_by_10,\n",
    "                                        color=counts_by_10),\n",
    "                                        size=0.3)+\n",
    "                          scale_y_continuous(name=\"ATAC-seq\",breaks=NULL,\n",
    "                                             limits=c(breaks_counts[1],breaks_counts[length(breaks_counts)]+1))+\n",
    "                          scale_x_continuous(name=NULL,breaks=c(start_sel,end_sel),labels=as.character(c(start_sel,end_sel)),\n",
    "                                             limits=c(start_sel-1,end_sel+1))+\n",
    "                          scale_color_viridis(\n",
    "                            alpha = 1,\n",
    "                            begin = 0,\n",
    "                            end = 1,\n",
    "                            direction = 1,\n",
    "                            discrete = FALSE,\n",
    "                            option = \"D\",\n",
    "                            aesthetics = \"color\",\n",
    "                            name=paste('ATAC counts',\"x10\",sep=\"\\n\"),\n",
    "                            na.value = NA,\n",
    "                            breaks=breaks_counts,\n",
    "                            labels=labels_counts,\n",
    "                            limits=c(breaks_counts[1],\n",
    "                                     breaks_counts[length(breaks_counts)])\n",
    "                          )\n",
    "                          \n",
    "                          # scale_color_gradient2(\n",
    "                          #   midpoint = 50,\n",
    "                          #   low = \"black\",\n",
    "                          #   mid = \"yellow\",\n",
    "                          #   high = \"red\",\n",
    "                          #   breaks=breaks_counts,labels=labels_counts,\n",
    "                          #   limits=c(breaks_counts[1],breaks_counts[length(breaks_counts)]),name=paste('ATAC reads',\"x10\",sep=\"\\n\"),na.value = \"white\")\n",
    "                        \n",
    "                        # scale_y_continuous(name=\"ATAC reads x10\",breaks=breaks_counts,labels=labels_counts,\n",
    "                        #                    limits=c(breaks_counts[1],breaks_counts[length(breaks_counts)]+1))+\n",
    "                        \n",
    "                       \n",
    "                        \n",
    "                        \n",
    "                        if(DEBUG == 1)\n",
    "                        {\n",
    "                          cat(\"ATAC plot I \\n\")\n",
    "                          \n",
    "                        }\n",
    "                        \n",
    "                        # setwd(path_graphs)\n",
    "                        # \n",
    "                        # svgname<-paste(paste(\"test\", sep='_'),\".svg\",sep='')\n",
    "                        # makesvg = TRUE\n",
    "                        # \n",
    "                        # if (makesvg == TRUE)\n",
    "                        # {\n",
    "                        #   ggsave(svgname, plot= ATAC_dot_plot,\n",
    "                        #          device=\"svg\",\n",
    "                        #          height=10, width=4)\n",
    "                        # }\n",
    "                        \n",
    "                        \n",
    "                        ATAC_dot_plot<-ATAC_dot_plot+\n",
    "                          facet_grid(Cell_Type ~ ., scales='free_x', space='free_x', switch=\"y\") +\n",
    "                          theme_cowplot(font_size = 8)+\n",
    "                          theme( strip.background = element_blank(),\n",
    "                                 strip.placement = \"outside\",\n",
    "                                 strip.text = element_text(size=6),\n",
    "                                 panel.spacing = unit(0.2, \"lines\"),\n",
    "                                 panel.background=element_rect(fill=\"white\"),\n",
    "                                 panel.border=element_rect(colour=\"white\",size=0,5),\n",
    "                                 panel.grid.major = element_blank(),\n",
    "                                 panel.grid.minor = element_blank())+\n",
    "                          theme_classic()+\n",
    "                          theme(plot.title=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                                axis.title.y=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                                axis.title.x=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                                axis.text.y=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                                axis.text.x=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                                axis.line.x = element_line(size = 0.2),\n",
    "                                axis.ticks = element_line(size = 0.2),\n",
    "                                axis.line.y = element_line(size = 0.2))+\n",
    "                          theme(legend.title = element_text(size=6),\n",
    "                                legend.text = element_text(size=6),\n",
    "                                legend.key.size = unit(0.25, 'cm'), #change legend key size\n",
    "                                legend.key.height = unit(0.25, 'cm'), #change legend key height\n",
    "                                legend.key.width = unit(0.25, 'cm'), #change legend key width\n",
    "                                legend.position=\"bottom\")+\n",
    "                          ggeasy::easy_center_title()\n",
    "                        \n",
    "                        \n",
    "                        \n",
    "                        \n",
    "                        if(DEBUG == 1)\n",
    "                        {\n",
    "                          cat(\"ATAC plot II \\n\")\n",
    "                          \n",
    "                        }\n",
    "                        \n",
    "                       \n",
    "                        setwd(path_graphs_rs_AS)\n",
    "                \n",
    "                        svgname<-paste(paste(\"ATAC\",'COORD_order', sep='_'),\".svg\",sep='')\n",
    "                        makesvg = TRUE\n",
    "                \n",
    "                        if (makesvg == TRUE)\n",
    "                        {\n",
    "                          ggsave(svgname, plot= ATAC_dot_plot,\n",
    "                                 device=\"svg\",\n",
    "                                 height=dataframe_to_print_SIT_plots_sel_AS_sel$Height, width=dataframe_to_print_SIT_plots_sel_AS_sel$Width)\n",
    "                        }\n",
    "                        \n",
    "                      }else{\n",
    "                        \n",
    "                        ATAC_dot_plot<-NA\n",
    "                      }#dim(ATAC_tracks_subset.m_sel)[1] >0\n",
    "\n",
    "            \n",
    "           #### PCHiC plot --------------------------------------------------------------------------------------------------------------\n",
    "          \n",
    "      \n",
    "      \n",
    "      \n",
    "              scale_df.m_subset_sel<-scale_df.m_subset[which(scale_df.m_subset$AS_ID == AS_ID_sel),]\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"scale_df.m_subset_sel_0\\n\")\n",
    "                cat(str(scale_df.m_subset_sel))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "              \n",
    "              if(dim(scale_df.m_subset_sel)[1] > 0)\n",
    "              {\n",
    "                \n",
    "                if(dim(ATAC_tracks_subset.m_sel)[1] >0)\n",
    "                {\n",
    "                  ATAC_dot_plot<-ATAC_dot_plot+\n",
    "                    theme(axis.text.x=element_blank(),\n",
    "                          axis.line.x = element_blank(),\n",
    "                          axis.ticks.x = element_blank())\n",
    "                  \n",
    "                }else{\n",
    "                  graph_genes<-graph_genes+\n",
    "                    theme(axis.text.x=element_blank(),\n",
    "                          axis.line.x = element_blank(),\n",
    "                          axis.ticks.x = element_blank())  \n",
    "                  finemap_beta_dot_plot<-finemap_beta_dot_plot+\n",
    "                    theme(axis.text.x=element_blank(),\n",
    "                          axis.line.x = element_blank(),\n",
    "                          axis.ticks.x = element_blank())\n",
    "                  \n",
    "                  }#dim(ATAC_tracks_subset.m_sel)[1] >0\n",
    "                \n",
    "                \n",
    "                dataframe_subset_PCHiC_NO_NA_sel<-dataframe_subset_PCHiC_NO_NA[which(dataframe_subset_PCHiC_NO_NA$AS_ID == AS_ID_sel),]\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"dataframe_subset_PCHiC_NO_NA_sel_0\\n\")\n",
    "                  cat(str(dataframe_subset_PCHiC_NO_NA_sel))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                #### arc ----\n",
    "                \n",
    "                \n",
    "                \n",
    "                indx.arc<-c(which(colnames(scale_df.m_subset_sel) == 'AS_ID'),\n",
    "                            which(colnames(scale_df.m_subset_sel) == 'oeStart'),which(colnames(scale_df.m_subset_sel) == 'oeEnd'),which(colnames(scale_df.m_subset_sel) == 'oeID'),\n",
    "                            which(colnames(scale_df.m_subset_sel) == 'baitChr'),which(colnames(scale_df.m_subset_sel) == 'baitStart'),which(colnames(scale_df.m_subset_sel) == 'baitEnd'),which(colnames(scale_df.m_subset_sel) == 'baitID'),\n",
    "                            which(colnames(scale_df.m_subset_sel) == 'baitSYMBOL'),\n",
    "                            which(colnames(scale_df.m_subset_sel) == 'Cell_Type'),which(colnames(scale_df.m_subset_sel) == 'ChicagoScore'))\n",
    "                \n",
    "                \n",
    "                \n",
    "                arc_df<-unique(scale_df.m_subset_sel[,indx.arc])\n",
    "                arc_df$chr<-paste('chr',arc_df$baitChr, sep='')\n",
    "                arc_df$oe_middle_point<-as.integer(arc_df$oeStart + ((arc_df$oeEnd - arc_df$oeStart)/2))\n",
    "                \n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"arc_df_0\\n\")\n",
    "                  cat(str(arc_df))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "              \n",
    "                \n",
    "                arc_df$Cell_Type<-factor(arc_df$Cell_Type,\n",
    "                                                  levels = unique(dataframe_subset_PCHiC_NO_NA_sel$Cell_Type),\n",
    "                                                  ordered=T)\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"arc_df_1\\n\")\n",
    "                  cat(str(arc_df))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(names(summary(arc_df$Cell_Type)))))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(summary(arc_df$Cell_Type))))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                arc_df$start<-NA\n",
    "                \n",
    "                arc_df$start<-arc_df$oe_middle_point\n",
    "                \n",
    "                # arc_df$start[which(arc_df$oe_middle_point >= arc_df$baitStart)]<-arc_df$baitStart\n",
    "                # arc_df$start[which(arc_df$oe_middle_point < arc_df$baitStart)]<-arc_df$oe_middle_point\n",
    "                \n",
    "                arc_df$end<-NA\n",
    "                \n",
    "                arc_df$end<-arc_df$baitStart\n",
    "                \n",
    "                # arc_df$end[which(arc_df$oe_middle_point >= arc_df$baitStart)]<-arc_df$oe_middle_point\n",
    "                # arc_df$end[which(arc_df$oe_middle_point < arc_df$baitStart)]<-arc_df$baitStart\n",
    "                # arc_df$name<-paste(arc_df$baitSYMBOL,arc_df$oeID, sep='__')\n",
    "                arc_df$name<-arc_df$baitSYMBOL\n",
    "                \n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"arc_df_2\\n\")\n",
    "                  cat(str(arc_df))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(names(summary(arc_df$Cell_Type)))))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(summary(arc_df$Cell_Type))))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                indx.int<-c(which(colnames(arc_df) == 'chr'),which(colnames(arc_df) == 'start'),which(colnames(arc_df) == 'end'),\n",
    "                            which(colnames(arc_df) == 'name'),which(colnames(arc_df) == 'Cell_Type'),\n",
    "                            which(colnames(arc_df) == 'ChicagoScore'))\n",
    "                \n",
    "                arc_df_FINAL<-unique(arc_df[,indx.int])\n",
    "                \n",
    "                arc_df_FINAL$Feature<-'arc'\n",
    "                arc_df_FINAL$y_start<-2\n",
    "                arc_df_FINAL$y_end<-1\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"arc_df_FINAL_0\\n\")\n",
    "                  cat(str(arc_df_FINAL))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                #### oe  and bait ----\n",
    "                \n",
    "                indx.oe<-c(which(colnames(scale_df.m_subset_sel) == 'oeChr'),which(colnames(scale_df.m_subset_sel) == 'oeStart'),which(colnames(scale_df.m_subset_sel) == 'oeEnd'),which(colnames(scale_df.m_subset_sel) == 'oeID'))\n",
    "                \n",
    "                oe_df<-unique(scale_df.m_subset_sel[,indx.oe])\n",
    "                oe_df$chr<-paste('chr',oe_df$oeChr, sep='')\n",
    "                oe_df$start<-oe_df$oeStart\n",
    "                oe_df$end<-oe_df$oeEnd\n",
    "                oe_df$name<-paste('oeID',as.character(oe_df$oeID),sep=' ')\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"oe_df_0\\n\")\n",
    "                  cat(str(oe_df))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                indx.int<-c(which(colnames(oe_df) == 'chr'),which(colnames(oe_df) == 'start'),which(colnames(oe_df) == 'end'),\n",
    "                            which(colnames(oe_df) == 'name'))\n",
    "                \n",
    "                oe_df_subset<-unique(oe_df[,indx.int])\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"oe_df_subset_0\\n\")\n",
    "                  cat(str(oe_df_subset))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(oe_df_subset$name)))\n",
    "                  cat(\"\\n\")\n",
    "                }\n",
    "                \n",
    "                oe_df_subset_expanded<-do.call(\"rbind\", replicate(length(levels(arc_df_FINAL$Cell_Type)), oe_df_subset, simplify = FALSE))\n",
    "                oe_df_subset_expanded$Cell_Type<-rep(levels(arc_df_FINAL$Cell_Type), dim(oe_df_subset)[1])\n",
    "                \n",
    "                oe_df_subset_expanded$Cell_Type<-factor(oe_df_subset_expanded$Cell_Type,\n",
    "                                                        levels = levels(arc_df_FINAL$Cell_Type),\n",
    "                                                        ordered=T)\n",
    "                \n",
    "                oe_df_subset_expanded$ChicagoScore<-NA\n",
    "                oe_df_subset_expanded$Feature<-'oe'\n",
    "                oe_df_subset_expanded$y_start<-2\n",
    "                oe_df_subset_expanded$y_end<-2\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"oe_df_subset_expanded_0\\n\")\n",
    "                  cat(str(oe_df_subset_expanded))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(names(summary(oe_df_subset_expanded$Cell_Type)))))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(summary(oe_df_subset_expanded$Cell_Type))))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                \n",
    "                indx.bait<-c(which(colnames(scale_df.m_subset_sel) == 'baitChr'),which(colnames(scale_df.m_subset_sel) == 'baitStart'),\n",
    "                             which(colnames(scale_df.m_subset_sel) == 'baitEnd'),which(colnames(scale_df.m_subset_sel) == 'baitID'),which(colnames(scale_df.m_subset_sel) == 'baitSYMBOL'))\n",
    "                \n",
    "                bait_df<-unique(scale_df.m_subset_sel[,indx.bait])\n",
    "                bait_df$chr<-paste('chr',bait_df$baitChr, sep='')\n",
    "                bait_df$start<-bait_df$baitStart\n",
    "                bait_df$end<-bait_df$baitEnd\n",
    "                # bait_df$name<-bait_df$baitSYMBOL #paste(bait_df$baitID,bait_df$baitSYMBOL, sep= '__')\n",
    "                bait_df$name<-paste('bait:',bait_df$baitSYMBOL,sep=\"\\n\")\n",
    "                \n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"bait_df_\\n\")\n",
    "                  cat(str(bait_df))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                indx.int<-c(which(colnames(bait_df) == 'chr'),which(colnames(bait_df) == 'start'),which(colnames(bait_df) == 'end'),\n",
    "                            which(colnames(bait_df) == 'name'))\n",
    "                \n",
    "                bait_df_subset<-unique(bait_df[,indx.int])\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"bait_df_subset_0\\n\")\n",
    "                  cat(str(bait_df_subset))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                \n",
    "                bait_df_subset_expanded<-do.call(\"rbind\", replicate(length(levels(arc_df_FINAL$Cell_Type)), bait_df_subset, simplify = FALSE))\n",
    "                bait_df_subset_expanded$Cell_Type<-rep(levels(arc_df_FINAL$Cell_Type), dim(bait_df_subset)[1])\n",
    "                \n",
    "                bait_df_subset_expanded$Cell_Type<-factor(bait_df_subset_expanded$Cell_Type,\n",
    "                                                          levels = levels(arc_df_FINAL$Cell_Type),\n",
    "                                                          ordered=T)\n",
    "                \n",
    "                bait_df_subset_expanded$ChicagoScore<-NA\n",
    "                bait_df_subset_expanded$Feature<-'oe'\n",
    "                bait_df_subset_expanded$y_start<-1\n",
    "                bait_df_subset_expanded$y_end<-1\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"bait_df_subset_expanded_0\\n\")\n",
    "                  cat(str(bait_df_subset_expanded))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(names(summary(bait_df_subset_expanded$Cell_Type)))))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(summary(bait_df_subset_expanded$Cell_Type))))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                \n",
    "                REP<-rbind(oe_df_subset_expanded,bait_df_subset_expanded,arc_df_FINAL)\n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"REP_0\\n\")\n",
    "                  cat(str(REP))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(names(summary(REP$Cell_Type)))))\n",
    "                  cat(\"\\n\")\n",
    "                  cat(sprintf(as.character(summary(REP$Cell_Type))))\n",
    "                  cat(\"\\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                #### PCHiC graph ----\n",
    "                \n",
    "             \n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"Graph_Part_START:\\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                graph_PCHiC<-ggplot()+\n",
    "                  geom_curve(data=REP[which(REP$Feature == 'arc'),],\n",
    "                             aes(x=start,\n",
    "                                 xend=end,\n",
    "                                 y=y_start,\n",
    "                                 yend=y_end),\n",
    "                                 color='black',\n",
    "                             curvature = -0.25, size=1)+\n",
    "                  geom_segment(data=REP[which(REP$Feature == 'oe'),],\n",
    "                               aes(x=start,\n",
    "                                   xend=end,\n",
    "                                   y=y_start,\n",
    "                                   yend=y_end),\n",
    "                               color=\"black\", size=2)+\n",
    "                  geom_segment(data=REP[which(REP$Feature == 'Bait'),],\n",
    "                               aes(x=start,\n",
    "                                   xend=end,\n",
    "                                   y=y_start,\n",
    "                                   yend=y_end),\n",
    "                               color=\"black\", size=2)+\n",
    "                  scale_x_continuous(name=NULL,breaks=c(start_sel,end_sel),\n",
    "                                     labels=as.character(c(start_sel,end_sel)),\n",
    "                                     limits=c(start_sel-1,end_sel+1))+\n",
    "                  scale_y_continuous(name='PCHi-C',\n",
    "                                     breaks=NULL,\n",
    "                                     limits=c(0.75,2.25))\n",
    "                  \n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"PCHiC plot I \\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                graph_PCHiC<-graph_PCHiC+\n",
    "                  facet_grid(Cell_Type ~ ., scales='free_x', space='free_x', switch=\"y\") +\n",
    "                  theme_cowplot(font_size = 8)+\n",
    "                  theme( strip.background = element_blank(),\n",
    "                         strip.placement = \"outside\",\n",
    "                         strip.text = element_text(size=6),\n",
    "                         panel.spacing = unit(0.2, \"lines\"),\n",
    "                         panel.background=element_rect(fill=\"white\"),\n",
    "                         panel.border=element_rect(colour=\"white\",size=0.5),\n",
    "                         panel.grid.major = element_blank(),\n",
    "                         panel.grid.minor = element_blank())+\n",
    "                  theme_classic()+\n",
    "                  theme(plot.title=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                        axis.title.y=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                        axis.title.x=element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                        axis.text.y=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                        axis.text.x=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                        axis.line.x = element_line(size = 0.2),\n",
    "                        axis.ticks = element_line(size = 0.2),\n",
    "                        axis.line.y = element_line(size = 0.2))+\n",
    "                  theme(legend.title = element_text(size=6),\n",
    "                        legend.text = element_text(size=6),\n",
    "                        legend.key.size = unit(0.25, 'cm'), #change legend key size\n",
    "                        legend.key.height = unit(0.25, 'cm'), #change legend key height\n",
    "                        legend.key.width = unit(0.25, 'cm'), #change legend key width\n",
    "                        legend.position=\"bottom\")+\n",
    "                  geom_text_repel(data=bait_df_subset,\n",
    "                                  aes(x=end+1,\n",
    "                                      y=1,\n",
    "                                      label=name),\n",
    "                                  box.padding = 0.25,\n",
    "                                  color='black',\n",
    "                                  max.overlaps = Inf,\n",
    "                                  show.legend = FALSE,\n",
    "                                  size=2)+\n",
    "                  geom_text_repel(data=oe_df_subset,\n",
    "                                  aes(x=start-1,\n",
    "                                      y=2,\n",
    "                                      label=name),\n",
    "                                  box.padding = 0.25,\n",
    "                                  color='black',\n",
    "                                  max.overlaps = Inf,\n",
    "                                  show.legend = FALSE,\n",
    "                                  size=2)+\n",
    "                  geom_vline(xintercept=variants_file_sel$pos37, color=\"gray\", linetype='dashed',size=0.5)+\n",
    "                  ggeasy::easy_center_title()\n",
    "                \n",
    "                \n",
    "                \n",
    "                \n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"PCHiC plot II \\n\")\n",
    "                  \n",
    "                }\n",
    "                \n",
    "                              \n",
    "                \n",
    "                \n",
    "        \n",
    "                setwd(path_graphs_rs_AS)\n",
    "        \n",
    "                svgname<-paste(paste(\"PCHiC\",'COORD_order', sep='_'),\".svg\",sep='')\n",
    "                makesvg = TRUE\n",
    "        \n",
    "                if (makesvg == TRUE)\n",
    "                {\n",
    "                  ggsave(svgname, plot= graph_PCHiC,\n",
    "                         device=\"svg\",\n",
    "                         height=dataframe_to_print_SIT_plots_sel_AS_sel$Height, width=dataframe_to_print_SIT_plots_sel_AS_sel$Width)\n",
    "                }\n",
    "                \n",
    "               \n",
    "                \n",
    "              }else{\n",
    "                \n",
    "                graph_PCHiC<-NA\n",
    "                \n",
    "              }#dim(scale_df.m_subset_sel)[1] > 0\n",
    "\n",
    "\n",
    "           #### FINAL GRAPH----\n",
    "\n",
    "              setwd(path_graphs_rs_AS)\n",
    "              \n",
    "              svgname<-paste(paste(\"Finemap_Beta\",'COORD_order', sep='_'),\".svg\",sep='')\n",
    "              makesvg = TRUE\n",
    "              \n",
    "              if (makesvg == TRUE)\n",
    "              {\n",
    "                ggsave(svgname, plot= finemap_beta_dot_plot,\n",
    "                       device=\"svg\",\n",
    "                       height=dataframe_to_print_SIT_plots_sel_AS_sel$Height, width=dataframe_to_print_SIT_plots_sel_AS_sel$Width)\n",
    "              }\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"dataframe_to_print_SIT_plots_sel_AS_sel_REMEMBER\\n\")\n",
    "                cat(str(dataframe_to_print_SIT_plots_sel_AS_sel))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "              }\n",
    "              \n",
    "              \n",
    "              graph_DEF<-plot_grid(finemap_beta_dot_plot,graph_genes,ATAC_dot_plot,graph_PCHiC,\n",
    "                                   nrow = 4,\n",
    "                                   ncol = 1,                                   \n",
    "                                   rel_heights=c(dataframe_to_print_SIT_plots_sel_AS_sel$weight_A,\n",
    "                                                 dataframe_to_print_SIT_plots_sel_AS_sel$weight_B,\n",
    "                                                 dataframe_to_print_SIT_plots_sel_AS_sel$weight_C,\n",
    "                                                 dataframe_to_print_SIT_plots_sel_AS_sel$weight_D))\n",
    "\n",
    "               graph_DEF_with_title <- ggdraw() +\n",
    "                  draw_plot(graph_DEF, x = 0, y = 0, width = 1, height = 0.92) + # Draw your combined plots\n",
    "                  draw_label(rs_sel,\n",
    "                             x = 0.02, y = 0.95, # Adjust x, y for position (0,0 is bottom-left, 1,1 is top-right)\n",
    "                             size = 8,         # Font size for the title (adjust as needed)\n",
    "                             fontfamily = \"sans\", # Sans-serif font\n",
    "                             fontface = \"bold\",   # Make it bold for emphasis\n",
    "                             vjust = 0.5, \n",
    "                             hjust = 0) # Adjust vertical and horizontal justification\n",
    "        \n",
    "              \n",
    "        \n",
    "        \n",
    "              setwd(path_graphs_rs_AS)\n",
    "        \n",
    "              svgname<-paste(paste(\"GRAPH_DEF\",'COORD_order', sep='_'),\".svg\",sep='')\n",
    "              makesvg = TRUE\n",
    "        \n",
    "              if (makesvg == TRUE)\n",
    "              {\n",
    "                ggsave(svgname, plot= graph_DEF,\n",
    "                       device=\"svg\",\n",
    "                       height=dataframe_to_print_SIT_plots_sel_AS_sel$Height, width=dataframe_to_print_SIT_plots_sel_AS_sel$Width)\n",
    "                \n",
    "              }\n",
    "\n",
    "\n",
    "                path_figures = paste(\"/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/test/Figure_graphs/\",sep='')\n",
    "\n",
    "               if(file.exists(path_figures)){\n",
    "\n",
    "                }else{\n",
    "                    dir.create(path_figures)\n",
    "                }\n",
    "\n",
    "           makesvg = FALSE\n",
    "           setwd(path_figures)\n",
    "\n",
    "           if(rs_sel == 'rs139141690'){\n",
    "  \n",
    "              svgname<-paste(paste(\"Figure_4_\",'panel_A', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "              makesvg = TRUE\n",
    "              \n",
    "              \n",
    "            }else{\n",
    "              \n",
    "              if(rs_sel == 'rs12733073'){\n",
    "                \n",
    "                \n",
    "                svgname<-paste(paste(\"Figure_S4_\",'panel_A', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                makesvg = TRUE\n",
    "                \n",
    "              }else{\n",
    "                \n",
    "                if(rs_sel == 'rs569022307'){\n",
    "                  \n",
    "                  svgname<-paste(paste(\"Figure_S4_\",'panel_D', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                  makesvg = TRUE\n",
    "                  \n",
    "                }else{\n",
    "                  \n",
    "                  if(rs_sel == 'rs150649461'){\n",
    "                    \n",
    "                    svgname<-paste(paste(\"Figure_S4_\",'panel_G', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                    makesvg = TRUE\n",
    "                    \n",
    "                    \n",
    "                  }else{\n",
    "                    \n",
    "                    if(rs_sel == 'rs150221884'){\n",
    "                      \n",
    "                      svgname<-paste(paste(\"Figure_S4_\",'panel_I', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                      makesvg = TRUE\n",
    "                      \n",
    "                      \n",
    "                    }else{\n",
    "                      \n",
    "                      if(rs_sel == 'rs543594419'){\n",
    "                        \n",
    "                        svgname<-paste(paste(\"Figure_S4_\",'panel_J', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                        makesvg = TRUE\n",
    "                        \n",
    "                        \n",
    "                      }else{\n",
    "                        \n",
    "                        if(rs_sel == 'rs572476245'){\n",
    "                          \n",
    "                          svgname<-paste(paste(\"Figure_S4_\",'panel_K', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                          makesvg = TRUE\n",
    "                          \n",
    "                          \n",
    "                        }else{\n",
    "                          \n",
    "                          if(rs_sel == 'rs114050631'){\n",
    "                            \n",
    "                            svgname<-paste(paste(\"Figure_S2_\",'panel_A', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                            makesvg = TRUE\n",
    "                            \n",
    "                            \n",
    "                          }else{\n",
    "                            \n",
    "                            if(rs_sel == 'rs150813342'){\n",
    "                              \n",
    "                              svgname<-paste(paste(\"Figure_S2_\",'panel_B_1', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                              makesvg = TRUE\n",
    "                              \n",
    "                              \n",
    "                            }else{\n",
    "                              \n",
    "                              if(rs_sel == 'rs187715179'){\n",
    "                                \n",
    "                                svgname<-paste(paste(\"Figure_S2_\",'panel_B_2', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                makesvg = TRUE\n",
    "                                \n",
    "                              }else{                                     \n",
    "                                \n",
    "                                \n",
    "                                if(rs_sel == 'rs182089148'){\n",
    "                                  \n",
    "                                  \n",
    "                                  svgname<-paste(paste(\"Figure_S4_\",'panel_B', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                  makesvg = TRUE\n",
    "                                  \n",
    "                                  \n",
    "                                }else{\n",
    "                                  \n",
    "                                  if(rs_sel == 'rs112401631'){\n",
    "                                    \n",
    "                                    \n",
    "                                    svgname<-paste(paste(\"Figure_S4_\",'panel_C', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                    makesvg = TRUE\n",
    "                                    \n",
    "                                  }else{\n",
    "                                    \n",
    "                                    if(rs_sel == 'rs17758695'){\n",
    "                                      \n",
    "                                      \n",
    "                                      svgname<-paste(paste(\"Figure_S4_\",'panel_E', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                      makesvg = TRUE\n",
    "                                      \n",
    "                                    }else{\n",
    "                                      \n",
    "                                      \n",
    "                                      if(rs_sel == 'rs149489081'){\n",
    "                                        \n",
    "                                        \n",
    "                                        svgname<-paste(paste(\"Figure_S4_\",'panel_F', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                        makesvg = TRUE\n",
    "                                        \n",
    "                                      }else{\n",
    "                                        \n",
    "                                        if(rs_sel == 'rs35592432'){\n",
    "                                          \n",
    "                                          \n",
    "                                          svgname<-paste(paste(\"Figure_S4_\",'panel_H', 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                          makesvg = TRUE\n",
    "                                          \n",
    "                                        }else{\n",
    "                                          \n",
    "                                          svgname<-paste(paste(\"Other_locus\",rs_sel, 'Locus_plot',AS_ID_sel, VAR_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                          makesvg = FALSE\n",
    "                                          \n",
    "                                        }#rs_sel == 'rs35592432'\n",
    "                                        \n",
    "                                      }#rs_sel == 'rs149489081'\n",
    "                                      \n",
    "                                    }#rs_sel == 'rs17758695'                                                                                \n",
    "                                    \n",
    "                                  }#rs_sel == 'rs112401631'            \n",
    "                                  \n",
    "                                }#  rs_sel == 'rs182089148'                                                          \n",
    "                              }#if(rs_sel == 'rs187715179'\n",
    "                              \n",
    "                            }#if(rs_sel == 'rs150813342')    \n",
    "                            \n",
    "                          }#rs_sel == 'rs114050631'                                                  \n",
    "                          \n",
    "                          \n",
    "                        }#rs_sel == 'rs572476245'\n",
    "                        \n",
    "                      }#rs_sel == 'rs543594419'\n",
    "                      \n",
    "                    }#rs_sel == 'rs150221884'\n",
    "                    \n",
    "                  }#rs_sel == 'rs150649461'\n",
    "                  \n",
    "                  \n",
    "                }#rs_sel == 'rs569022307'\n",
    "                \n",
    "              }#rs_sel == 'rs12733073'\n",
    "              \n",
    "            }#rs_sel == 'rs139141690'\n",
    "\n",
    "            \n",
    "                                                \n",
    "          if (makesvg == TRUE)\n",
    "          {\n",
    "            ggsave(svgname, plot= graph_DEF_with_title,\n",
    "                   device=\"svg\",\n",
    "                   height=dataframe_to_print_SIT_plots_sel_AS_sel$Height, width=dataframe_to_print_SIT_plots_sel_AS_sel$Width)\n",
    "            \n",
    "          }\n",
    "\n",
    "\n",
    "        }# k in 1:length(AS_ID_array)      \n",
    "}#i in START:length(VARS)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "73d375aa-e132-4e99-8ea9-9337dd6cc37d",
   "metadata": {},
   "source": [
    "# Discarded plotting the Chicago Score"
   ]
  },
  {
   "cell_type": "raw",
   "id": "752cc26f-95ce-455e-8d5f-8146fd54ca91",
   "metadata": {},
   "source": [
    "geom_curve(data=REP[which(REP$Feature == 'arc'),],\n",
    "                             aes(x=start,\n",
    "                                 xend=end,\n",
    "                                 y=y_start,\n",
    "                                 yend=y_end,\n",
    "                                 color=ChicagoScore),\n",
    "                             curvature = -0.025, size=1)+\n",
    "\n",
    "+\n",
    "                  scale_color_gradient2(\n",
    "                    low = \"black\",\n",
    "                    mid = \"yellow\",\n",
    "                    high = \"red\",\n",
    "                    midpoint = 5,\n",
    "                    breaks=breaks_CS,labels=labels_CS,\n",
    "                    limits=c(breaks_CS[1],breaks_CS[length(breaks_CS)]),name=paste('PCHi-C Chicago',\"score\",sep=\"\\n\"),na.value = \"white\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "163b4651-8640-4623-a767-3475b71c88ce",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13ca09d3-0de9-45f5-a0f1-dd7cefba28ef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3906bc68-4f5a-4125-b924-8f54aa9c6784",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d0e2229-002a-46ad-8288-58a8e48f3d24",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4de7f9f2-5593-436b-a9c7-db65175977a6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6b86a27e-208a-4b72-a147-fc8c1343cd84",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81255fc5-9493-456d-b90d-44655e81714d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55bd6659-3220-4f5a-ae77-f9b852f06044",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e53ff2df-ce7b-496c-ab72-d17f46aa6a37",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "general_purpose",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
