{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5ba6e839-d30f-48b7-bade-2bebc44d281d",
   "metadata": {},
   "source": [
    "# Load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "898e4e86-7d07-4ed9-8353-945efc378c08",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:17:51.272247Z",
     "iopub.status.busy": "2025-06-24T16:17:51.271225Z",
     "iopub.status.idle": "2025-06-24T16:17:51.333438Z",
     "shell.execute_reply": "2025-06-24T16:17:51.332408Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/latex": [
       "'/home/manuel.tardaguila/R/x86\\_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/markdown": [
       "'/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/plain": [
       "[1] \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/latex": [
       "'/home/manuel.tardaguila/R/x86\\_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/markdown": [
       "'/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/plain": [
       "[1] \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    ".libPaths()\n",
    "\n",
    "assign(\".lib.loc\", \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\", envir = environment(.libPaths))\n",
    ".libPaths()\n",
    "\n",
    "suppressMessages(library(\"plyr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"data.table\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"withr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggplot2\", lib.loc = \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"farver\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"labeling\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"optparse\", lib.loc = \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"dplyr\", lib.loc = \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"withr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"backports\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"broom\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"rstudioapi\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"tzdb\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"svglite\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggeasy\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"tidyverse\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"BiocGenerics\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"S4Vectors\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"IRanges\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"GenomeInfoDb\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"GenomicRanges\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"Biobase\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"AnnotationDbi\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"GO.db\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"org.Hs.eg.db\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggrepel\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"RColorBrewer\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"svglite\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"cowplot\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"Matrix\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"rtracklayer\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"Biostrings\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggrepel\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggarchery\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggnewscale\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"splitstackshape\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"viridis\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggtranscript\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggpubr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "8af0ed16-9a2b-441f-91b6-553a182e9163",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:17:51.451292Z",
     "iopub.status.busy": "2025-06-24T16:17:51.450392Z",
     "iopub.status.idle": "2025-06-24T16:17:51.457843Z",
     "shell.execute_reply": "2025-06-24T16:17:51.456854Z"
    }
   },
   "outputs": [],
   "source": [
    "DEBUG<-1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6c920f4-552e-48fb-87f1-2981231f94c4",
   "metadata": {},
   "source": [
    "# Read input files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d85b45c-b82d-485b-8fc1-f9e6a57afb27",
   "metadata": {},
   "source": [
    "## Read path to INTERVAL results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "e5e0b472-a307-490b-bfba-621424831ae1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:17:52.153758Z",
     "iopub.status.busy": "2025-06-24T16:17:52.152794Z",
     "iopub.status.idle": "2025-06-24T16:17:52.167225Z",
     "shell.execute_reply": "2025-06-24T16:17:52.166203Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "path_INTERVAL_\n",
      "/group/soranzo/manuel.tardaguila/INTERVAL_results/\n"
     ]
    }
   ],
   "source": [
    "#### READ and transform path_INTERVAL ----\n",
    "  \n",
    "path_INTERVAL = '/group/soranzo/manuel.tardaguila/INTERVAL_results/'\n",
    "\n",
    "cat(\"path_INTERVAL_\\n\")\n",
    "cat(sprintf(as.character(path_INTERVAL)))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7d64069-b586-497e-b338-de2cb753fed4",
   "metadata": {},
   "source": [
    "## Read path to BluePrint results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "79a81c99-4db2-4208-9ee8-a4cae7f79a56",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:17:52.547329Z",
     "iopub.status.busy": "2025-06-24T16:17:52.546431Z",
     "iopub.status.idle": "2025-06-24T16:17:52.560571Z",
     "shell.execute_reply": "2025-06-24T16:17:52.559550Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "path_BP_\n",
      "/group/soranzo/manuel.tardaguila/BP_results/\n"
     ]
    }
   ],
   "source": [
    "#### READ and transform path_BP ----\n",
    "\n",
    "path_BP = '/group/soranzo/manuel.tardaguila/BP_results/'\n",
    "\n",
    "cat(\"path_BP_\\n\")\n",
    "cat(sprintf(as.character(path_BP)))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0b06d54-e73f-43fd-8d8b-81141813a4bd",
   "metadata": {},
   "source": [
    "## Read INTERVAL transcript expression matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "432681b8-e18c-41ae-92cc-5937b70d6f51",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:17:52.929077Z",
     "iopub.status.busy": "2025-06-24T16:17:52.928160Z",
     "iopub.status.idle": "2025-06-24T16:18:02.197683Z",
     "shell.execute_reply": "2025-06-24T16:18:02.196479Z"
    }
   },
   "outputs": [],
   "source": [
    " \n",
    "Transposed_Isoform_Expression<-readRDS(\"/group/soranzo/manuel.tardaguila/INTERVAL_results/Transposed_Isoform_Expression_df.rds\")\n",
    "colnames(Transposed_Isoform_Expression)[which(colnames(Transposed_Isoform_Expression) == \"Sample_id\")]<-\"sample_id\"\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "04645c4b-f8d2-449a-8b8d-113661fc806d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T13:27:28.789999Z",
     "iopub.status.busy": "2025-06-24T13:27:28.789058Z",
     "iopub.status.idle": "2025-06-24T13:27:29.174610Z",
     "shell.execute_reply": "2025-06-24T13:27:29.173457Z"
    }
   },
   "source": [
    "cat(\"Transposed_Isoform_Expression_0\\n\")\n",
    "cat(str(Transposed_Isoform_Expression))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c875f99-010c-4e0f-b024-5ba88f5d125d",
   "metadata": {},
   "source": [
    "## Read ENSEMBL gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "02e54da4-56f0-448a-9f15-e35ac33517c5",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:18:02.200363Z",
     "iopub.status.busy": "2025-06-24T16:18:02.199435Z",
     "iopub.status.idle": "2025-06-24T16:18:24.550690Z",
     "shell.execute_reply": "2025-06-24T16:18:24.549490Z"
    }
   },
   "outputs": [],
   "source": [
    "#### READ ensembl_gtf ----\n",
    "\n",
    "ensembl_gtf = readGFF(\"/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/Reference_files/Homo_sapiens.GRCh37.87.gtf.gz\")\n",
    "\n",
    "\n",
    "# cat(\"ensembl_gtf_0\\n\")\n",
    "# cat(str(ensembl_gtf))\n",
    "# cat(\"\\n\")\n",
    "# cat(sprintf(as.character(names(summary(as.factor(ensembl_gtf$type))))))\n",
    "# cat(\"\\n\")\n",
    "# cat(sprintf(as.character(summary(as.factor(ensembl_gtf$type)))))\n",
    "# cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91053bce-d1a7-466c-834d-bb76edaca89c",
   "metadata": {},
   "source": [
    "## Read Table S6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "02b4d32e-4ec6-48ba-b53a-8bb2f0db74a2",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:18:24.553391Z",
     "iopub.status.busy": "2025-06-24T16:18:24.552513Z",
     "iopub.status.idle": "2025-06-24T16:18:24.676846Z",
     "shell.execute_reply": "2025-06-24T16:18:24.675886Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Table_S6_0\n",
      "'data.frame':\t94 obs. of  25 variables:\n",
      " $ VAR                        : chr  \"chr1_29217311_G_A\" \"chr1_91606142_A_G\" \"chr1_92925654_G_C\" \"chr1_92981236_T_G\" ...\n",
      " $ rs                         : chr  \"rs115820925\" \"rs188951504\" \"rs150649461\" \"rs184721335\" ...\n",
      " $ Mechanistic_Class          : Ord.factor w/ 5 levels \"DE\"<\"ATU\"<\"DE + ATU\"<..: 2 4 3 4 4 4 4 4 1 1 ...\n",
      " $ Manual_curation            : Ord.factor w/ 7 levels \"R in candidate\"<..: 1 6 1 3 6 6 6 3 1 1 ...\n",
      " $ Candidate_effector         : chr  \"EPB41\" \"ABSENT\" \"EVI5, GFI1\" \"EVI5, GFI1\" ...\n",
      " $ OpenTargets_QTL            : chr  \"RCC1\" \"NOT FOUND\" \"EVI5, GFI1\" \"NOT FOUND\" ...\n",
      " $ Whole_blood_DE_HGNC_string : chr  NA NA \"EVI5;GFI1\" NA ...\n",
      " $ Whole_blood_DTU_HGNC_string: chr  NA NA \"EVI5\" NA ...\n",
      " $ Monocyte_DE_HGNC_string    : chr  NA NA \"GFI1\" NA ...\n",
      " $ Tcell_DE_HGNC_string       : chr  NA NA NA NA ...\n",
      " $ Neutrophil_DE_HGNC_string  : chr  NA NA NA NA ...\n",
      " $ Neutrophil_DTU_HGNC_string : chr  NA NA NA NA ...\n",
      " $ Tcell_DTU_HGNC_string      : chr  \"EPB41\" NA NA NA ...\n",
      " $ Monocyte_DTU_HGNC_string   : chr  NA NA NA NA ...\n",
      " $ Proxy_rsid_string          : chr  \"rs183557510__0.52__PHACTR4(NCT,INTRON)|rs139376744__0.53__PHACTR4(NCT,INTRON)|rs186394361__0.52__RCC1(UPSTREAM,\"| __truncated__ \"rs189955976__0.55__NA(INTERGENIC)\" \"rs141094656__0.95__GLMN(INTRON,UPSTREAM,NMD)|rs139795227__0.99__NA(REGULATORY);RPAP2(INTRON,NCT)\" \"rs568310492__0.68__BTBD8(UPSTREAM);SETSIP(UPSTREAM)|rs116830401__0.61__BTBD8(INTRON);GAPDHP46(UPSTREAM)|rs57533\"| __truncated__ ...\n",
      " $ MPRA_CLASS                 : Ord.factor w/ 2 levels \"MPRA negative\"<..: 1 2 1 1 1 1 1 2 2 2 ...\n",
      " $ genIE_CLASS                : Ord.factor w/ 3 levels \"genIE negative\"<..: 3 3 3 3 3 3 3 3 3 3 ...\n",
      " $ Replication_OT_QTL         : chr  \"eQTL_OT_FOUND\" NA \"eQTL_OT_FOUND\" NA ...\n",
      " $ VAR_38                     : chr  \"chr1_28890799_G_A\" \"chr1_91140585_A_G\" \"chr1_92460097_G_C\" \"chr1_92515679_T_G\" ...\n",
      " $ Multi_Lineage              : Ord.factor w/ 2 levels \"Lineage restricted\"<..: 1 1 1 1 1 1 1 1 1 2 ...\n",
      " $ Lineage_string             : chr  \"erythroid lineage\" \"granulocyte monocyte lineage\" \"granulocyte monocyte lineage\" \"granulocyte monocyte lineage\" ...\n",
      " $ phenotype_DEF_string       : chr  \"MRV\" \"MONO#\" \"MONO#;MONO%;LYMPH%;WBC#\" \"MONO%\" ...\n",
      " $ maf_origin                 : num  0.00759 0.00162 0.01475 0.00426 0.00777 ...\n",
      " $ VEP_DEF_LABELS_wCSQ        : chr  \"INTRON\" \"INTERGENIC\" \"INTERGENIC\" \"INTRON\" ...\n",
      " $ integration_category       : Ord.factor w/ 4 levels \"Double positives\"<..: 3 2 3 4 4 4 4 2 1 1 ...\n",
      "\n",
      " chr [1:94] \"chr1_29217311_G_A\" \"chr1_91606142_A_G\" \"chr1_92925654_G_C\" ...\n",
      "\n",
      "Table_S6_subset_0\n",
      "'data.frame':\t24 obs. of  25 variables:\n",
      " $ VAR                        : chr  \"chr1_29217311_G_A\" \"chr1_92925654_G_C\" \"chr2_24091099_C_T\" \"chr2_219020958_C_T\" ...\n",
      " $ rs                         : chr  \"rs115820925\" \"rs150649461\" \"rs150221884\" \"rs114050631\" ...\n",
      " $ Mechanistic_Class          : Ord.factor w/ 5 levels \"DE\"<\"ATU\"<\"DE + ATU\"<..: 2 3 2 3 2 3 2 2 3 2 ...\n",
      " $ Manual_curation            : Ord.factor w/ 7 levels \"R in candidate\"<..: 1 1 3 1 1 1 1 1 2 2 ...\n",
      " $ Candidate_effector         : chr  \"EPB41\" \"EVI5, GFI1\" \"MFSD2B\" \"CXCR2\" ...\n",
      " $ OpenTargets_QTL            : chr  \"RCC1\" \"EVI5, GFI1\" \"NOT FOUND\" \"CXCR2\" ...\n",
      " $ Whole_blood_DE_HGNC_string : chr  NA \"EVI5;GFI1\" NA \"CXCR1;CXCR2;RP11-378A13.1\" ...\n",
      " $ Whole_blood_DTU_HGNC_string: chr  NA \"EVI5\" \"MFSD2B\" \"SLC11A1\" ...\n",
      " $ Monocyte_DE_HGNC_string    : chr  NA \"GFI1\" NA NA ...\n",
      " $ Tcell_DE_HGNC_string       : chr  NA NA NA NA ...\n",
      " $ Neutrophil_DE_HGNC_string  : chr  NA NA NA NA ...\n",
      " $ Neutrophil_DTU_HGNC_string : chr  NA NA NA NA ...\n",
      " $ Tcell_DTU_HGNC_string      : chr  \"EPB41\" NA NA NA ...\n",
      " $ Monocyte_DTU_HGNC_string   : chr  NA NA NA NA ...\n",
      " $ Proxy_rsid_string          : chr  \"rs183557510__0.52__PHACTR4(NCT,INTRON)|rs139376744__0.53__PHACTR4(NCT,INTRON)|rs186394361__0.52__RCC1(UPSTREAM,\"| __truncated__ \"rs141094656__0.95__GLMN(INTRON,UPSTREAM,NMD)|rs139795227__0.99__NA(REGULATORY);RPAP2(INTRON,NCT)\" \"rs151263370__0.54__ATAD2B(INTRON)|rs151263370__0.54__ATAD2B(INTRON)|rs188588191__0.96__ATAD2B(INTRON)|rs1389035\"| __truncated__ \"rs192280464__0.7__AC021016.7(UPSTREAM);AC021016.8(NCT);PNKD(DOWNSTREAM)\" ...\n",
      " $ MPRA_CLASS                 : Ord.factor w/ 2 levels \"MPRA negative\"<..: 1 1 2 1 2 2 1 2 1 1 ...\n",
      " $ genIE_CLASS                : Ord.factor w/ 3 levels \"genIE negative\"<..: 3 3 3 3 3 2 3 3 3 3 ...\n",
      " $ Replication_OT_QTL         : chr  \"eQTL_OT_FOUND\" \"eQTL_OT_FOUND\" NA \"eQTL_OT_FOUND\" ...\n",
      " $ VAR_38                     : chr  \"chr1_28890799_G_A\" \"chr1_92460097_G_C\" \"chr2_23868229_C_T\" \"chr2_218156235_C_T\" ...\n",
      " $ Multi_Lineage              : Ord.factor w/ 2 levels \"Lineage restricted\"<..: 1 1 2 1 1 1 1 2 1 1 ...\n",
      " $ Lineage_string             : chr  \"erythroid lineage\" \"granulocyte monocyte lineage\" \"erythroid lineage;megakaryocytic lineage\" \"granulocyte monocyte lineage\" ...\n",
      " $ phenotype_DEF_string       : chr  \"MRV\" \"MONO#;MONO%;LYMPH%;WBC#\" \"RET#;RET%;IRF;HLSR%;MPV\" \"MONO%;NEUT#;NEUT%;BASO#;LYMPH%;WBC#\" ...\n",
      " $ maf_origin                 : num  0.00759 0.01475 0.00464 0.01127 0.01093 ...\n",
      " $ VEP_DEF_LABELS_wCSQ        : chr  \"INTRON\" \"INTERGENIC\" \"INTRON\" \"INTERGENIC\" ...\n",
      " $ integration_category       : Ord.factor w/ 4 levels \"Double positives\"<..: 3 3 1 3 1 1 3 1 3 3 ...\n",
      "\n",
      " chr [1:24] \"chr1_29217311_G_A\" \"chr1_92925654_G_C\" \"chr2_24091099_C_T\" ...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in type.convert.default(unlist(x, use.names = FALSE)):\n",
      "“'as.is' should be specified by the caller; using TRUE”\n",
      "Warning message in type.convert.default(unlist(x, use.names = FALSE)):\n",
      "“'as.is' should be specified by the caller; using TRUE”\n",
      "Warning message in type.convert.default(unlist(x, use.names = FALSE)):\n",
      "“'as.is' should be specified by the caller; using TRUE”\n",
      "Warning message in type.convert.default(unlist(x, use.names = FALSE)):\n",
      "“'as.is' should be specified by the caller; using TRUE”\n",
      "Warning message in melt(Table_S6_subset_ATU, id.vars = c(\"VAR\", \"rs\"), variable.name = \"RNASeq_source\", :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Table_S6_subset_ATU). In the next version, this warning will become an error.”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Monocyte Neutrophil Tcell Whole_blood\n",
      "26 26 26 26\n",
      "Table_S6_subset_ATU.m_NO_NA_0\n",
      "'data.frame':\t29 obs. of  4 variables:\n",
      " $ VAR          : chr  \"chr1_92925654_G_C\" \"chr2_24091099_C_T\" \"chr2_219020958_C_T\" \"chr3_128317978_C_T\" ...\n",
      " $ rs           : chr  \"rs150649461\" \"rs150221884\" \"rs114050631\" \"rs191451841\" ...\n",
      " $ RNASeq_source: chr  \"Whole_blood\" \"Whole_blood\" \"Whole_blood\" \"Whole_blood\" ...\n",
      " $ HGNC         : chr  \"EVI5\" \"MFSD2B\" \"SLC11A1\" \"GATA2\" ...\n",
      "\n",
      "EVI5 MFSD2B SLC11A1 GATA2 EEFSEC TAF8 IKZF1 PILRB ANK1 GFI1B ATL1 NPRL3 ELP5 VMP1 KDSR RASAL3 TYMP ARSA ODF3B BCL2 PLCL2 FOXP1 CCR7 EPB41 PRMT5-AS1\n",
      "Monocyte Neutrophil Tcell Whole_blood\n",
      "3 1 4 21\n"
     ]
    }
   ],
   "source": [
    "#### READ Table_S6 ----\n",
    "  \n",
    "  Table_S6<-readRDS(file=\"/group/soranzo/manuel.tardaguila/Paper_bits/FIX_TABLES/Provisional_Tables/Table_S6_Provisional.rds\")\n",
    "  \n",
    "  \n",
    "  cat(\"Table_S6_0\\n\")\n",
    "  cat(str(Table_S6))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S6$VAR)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  Table_S6_subset<-Table_S6[which(Table_S6$Mechanistic_Class%in%c('ATU','DE + ATU')),]\n",
    "  \n",
    "  cat(\"Table_S6_subset_0\\n\")\n",
    "  cat(str(Table_S6_subset))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S6_subset$VAR)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  \n",
    "  Table_S6_subset<-unique(as.data.frame(cSplit(Table_S6_subset,sep = ';', direction = \"long\",\n",
    "                                              splitCols = \"Whole_blood_DTU_HGNC_string\"),stringsAsFactors=F))\n",
    "  \n",
    "  # cat(\"Table_S6_subset_1\\n\")\n",
    "  # cat(str(Table_S6_subset))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(unique(Table_S6_subset$Whole_blood_DTU_HGNC_string))))\n",
    "  # cat(\"\\n\")\n",
    "  \n",
    "  Table_S6_subset<-unique(as.data.frame(cSplit(Table_S6_subset,sep = ';', direction = \"long\",\n",
    "                                               splitCols = \"Neutrophil_DTU_HGNC_string\"),stringsAsFactors=F))\n",
    "  \n",
    "  # cat(\"Table_S6_subset_2\\n\")\n",
    "  # cat(str(Table_S6_subset))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(unique(Table_S6_subset$Neutrophil_DTU_HGNC_string))))\n",
    "  # cat(\"\\n\")\n",
    " \n",
    "  Table_S6_subset<-unique(as.data.frame(cSplit(Table_S6_subset,sep = ';', direction = \"long\",\n",
    "                                               splitCols = \"Monocyte_DTU_HGNC_string\"),stringsAsFactors=F))\n",
    "  \n",
    "  # cat(\"Table_S6_subset_3\\n\")\n",
    "  # cat(str(Table_S6_subset))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(unique(Table_S6_subset$Monocyte_DTU_HGNC_string))))\n",
    "  # cat(\"\\n\")\n",
    "  \n",
    "  Table_S6_subset<-unique(as.data.frame(cSplit(Table_S6_subset,sep = ';', direction = \"long\",\n",
    "                                               splitCols = \"Tcell_DTU_HGNC_string\"),stringsAsFactors=F))\n",
    "  \n",
    "  # cat(\"Table_S6_subset_4\\n\")\n",
    "  # cat(str(Table_S6_subset))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(unique(Table_S6_subset$Tcell_DTU_HGNC_string))))\n",
    "  # cat(\"\\n\")\n",
    "  \n",
    "  Table_S6_subset_ATU<-unique(Table_S6_subset[,c(which(colnames(Table_S6_subset) == 'VAR'),which(colnames(Table_S6_subset) == 'rs'),\n",
    "                                          which(colnames(Table_S6_subset) == 'Whole_blood_DTU_HGNC_string'),\n",
    "                                          which(colnames(Table_S6_subset) == 'Neutrophil_DTU_HGNC_string'),\n",
    "                                          which(colnames(Table_S6_subset) == 'Monocyte_DTU_HGNC_string'),\n",
    "                                          which(colnames(Table_S6_subset) == 'Tcell_DTU_HGNC_string'))])\n",
    "  \n",
    "  \n",
    "  # cat(\"Table_S6_subset_ATU_0\\n\")\n",
    "  # cat(str(Table_S6_subset_ATU))\n",
    "  # cat(\"\\n\")\n",
    "  \n",
    "  Table_S6_subset_ATU.m<-melt(Table_S6_subset_ATU, id.vars=c('VAR','rs'), variable.name='RNASeq_source', value.name='HGNC')\n",
    "  \n",
    "  # cat(\"Table_S6_subset_ATU.m_0\\n\")\n",
    "  # cat(str(Table_S6_subset_ATU.m))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(unique(Table_S6_subset_ATU.m$HGNC))))\n",
    "  # cat(\"\\n\")\n",
    "  \n",
    "  Table_S6_subset_ATU.m$RNASeq_source<-gsub(\"_DTU_HGNC_string\",\"\",Table_S6_subset_ATU.m$RNASeq_source)\n",
    "  \n",
    "  cat(sprintf(as.character(names(summary(as.factor(Table_S6_subset_ATU.m$RNASeq_source))))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(summary(as.factor(Table_S6_subset_ATU.m$RNASeq_source)))))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  \n",
    "  # check<-Table_S6_subset_ATU.m[which(Table_S6_subset_ATU.m$VAR == 'chr3_71355240_G_C'),]\n",
    "  # \n",
    "  # cat(\"check_0\\n\")\n",
    "  # cat(str(check))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(unique(check$HGNC))))\n",
    "  # cat(\"\\n\")\n",
    "  \n",
    "  \n",
    "  \n",
    "  \n",
    "  Table_S6_subset_ATU.m_NO_NA<-Table_S6_subset_ATU.m[which(Table_S6_subset_ATU.m$HGNC != \"NA\"),]\n",
    "  \n",
    "  cat(\"Table_S6_subset_ATU.m_NO_NA_0\\n\")\n",
    "  cat(str(Table_S6_subset_ATU.m_NO_NA))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(unique(Table_S6_subset_ATU.m_NO_NA$HGNC))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(names(summary(as.factor(Table_S6_subset_ATU.m_NO_NA$RNASeq_source))))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(summary(as.factor(Table_S6_subset_ATU.m_NO_NA$RNASeq_source)))))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  Table_S6_subset_ATU.m_NO_NA<-unique(Table_S6_subset_ATU.m_NO_NA[,-which(colnames(Table_S6_subset_ATU.m_NO_NA) == 'RNASeq_source')])\n",
    "  \n",
    "  \n",
    " \n",
    "  # check<-Table_S6_subset_ATU.m_NO_NA[which(Table_S6_subset_ATU.m_NO_NA$VAR == 'chr3_71355240_G_C'),]\n",
    "  # \n",
    "  # cat(\"check_2\\n\")\n",
    "  # cat(str(check))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(unique(check$HGNC))))\n",
    "  # cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef2ebc31-697a-4459-b0c8-f9b6d8a79518",
   "metadata": {},
   "source": [
    "## DE/ATU results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "95c20449-8f97-4ae2-8610-a62a31968808",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:18:24.679363Z",
     "iopub.status.busy": "2025-06-24T16:18:24.678514Z",
     "iopub.status.idle": "2025-06-24T16:18:24.790725Z",
     "shell.execute_reply": "2025-06-24T16:18:24.789762Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Table_S7_0\n",
      "'data.frame':\t4544 obs. of  12 variables:\n",
      " $ VAR                              : chr  \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" ...\n",
      " $ HGNC                             : chr  \"DPH5\" \"DPH5\" \"DPH5\" \"EXTL2\" ...\n",
      " $ ensembl_gene_id                  : chr  \"ENSG00000117543\" \"ENSG00000117543\" \"ENSG00000117543\" \"ENSG00000162694\" ...\n",
      " $ rs                               : Ord.factor w/ 88 levels \"rs115820925\"<..: 6 6 6 6 6 6 6 6 6 6 ...\n",
      " $ RNASeq_source                    : Ord.factor w/ 4 levels \"Whole blood\"<..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ Analysis                         : Ord.factor w/ 2 levels \"DE\"<\"ATU\": 1 2 2 1 2 2 1 1 1 1 ...\n",
      " $ transcript_id                    : chr  NA \"ENST00000427040\" \"ENST00000370109\" NA ...\n",
      " $ Beta                             : num  -0.00182 0 0.11236 -0.28817 0.3839 ...\n",
      " $ Beta_Z_score                     : num  -0.0232 -0.038 0.588 -0.0823 2.1008 ...\n",
      " $ adjusted_pval                    : num  1 NA 0.708 0.871 0.708 ...\n",
      " $ adjusted_minus_logpval           : num  0 NA 0.15 0.06 0.15 NA 0 0.1 0 0.01 ...\n",
      " $ regulated_gene_in_Gene_body_CLASS: Ord.factor w/ 2 levels \"Variant_outside_gene_body_of_regulated_gene\"<..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "\n",
      " chr [1:88] \"chr1_101744646_C_G\" \"chr1_158613314_G_A\" \"chr1_158638260_G_A\" ...\n",
      "\n",
      " chr [1:906] \"ENSG00000117543\" \"ENSG00000162694\" \"ENSG00000235795\" ...\n",
      "\n",
      "Table_S7_ATU_0\n",
      "'data.frame':\t2715 obs. of  12 variables:\n",
      " $ VAR                              : chr  \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" ...\n",
      " $ HGNC                             : chr  \"DPH5\" \"DPH5\" \"EXTL2\" \"EXTL2\" ...\n",
      " $ ensembl_gene_id                  : chr  \"ENSG00000117543\" \"ENSG00000117543\" \"ENSG00000162694\" \"ENSG00000162694\" ...\n",
      " $ rs                               : Ord.factor w/ 88 levels \"rs115820925\"<..: 6 6 6 6 6 6 7 7 7 7 ...\n",
      " $ RNASeq_source                    : Ord.factor w/ 4 levels \"Whole blood\"<..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ Analysis                         : Ord.factor w/ 2 levels \"DE\"<\"ATU\": 2 2 2 2 2 2 2 2 2 2 ...\n",
      " $ transcript_id                    : chr  \"ENST00000427040\" \"ENST00000370109\" \"ENST00000370113\" \"ENST00000370114\" ...\n",
      " $ Beta                             : num  0 0.112 0.384 0 0 ...\n",
      " $ Beta_Z_score                     : num  -0.038 0.588 2.101 -0.038 -0.038 ...\n",
      " $ adjusted_pval                    : num  NA 0.708 0.708 NA NA ...\n",
      " $ adjusted_minus_logpval           : num  NA 0.15 0.15 NA NA 0.15 0.11 NA 0.11 NA ...\n",
      " $ regulated_gene_in_Gene_body_CLASS: Ord.factor w/ 2 levels \"Variant_outside_gene_body_of_regulated_gene\"<..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "\n",
      " chr [1:85] \"chr1_101744646_C_G\" \"chr1_158613314_G_A\" \"chr1_158638260_G_A\" ...\n",
      "\n",
      " chr [1:492] \"ENSG00000117543\" \"ENSG00000162694\" \"ENSG00000162695\" ...\n",
      "\n",
      "Whole blood Monocytes Neutrophils naive T-CD4 Cells\n",
      "1404 437 437 437\n",
      "Table_S7_ATU_subset_0\n",
      "'data.frame':\t194 obs. of  12 variables:\n",
      " $ VAR                              : chr  \"chr1_29217311_G_A\" \"chr1_29217311_G_A\" \"chr1_29217311_G_A\" \"chr1_29217311_G_A\" ...\n",
      " $ rs                               : Ord.factor w/ 88 levels \"rs115820925\"<..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ HGNC                             : chr  \"EPB41\" \"EPB41\" \"EPB41\" \"EPB41\" ...\n",
      " $ ensembl_gene_id                  : chr  \"ENSG00000159023\" \"ENSG00000159023\" \"ENSG00000159023\" \"ENSG00000159023\" ...\n",
      " $ RNASeq_source                    : Ord.factor w/ 4 levels \"Whole blood\"<..: 2 2 3 3 4 4 4 1 1 3 ...\n",
      " $ Analysis                         : Ord.factor w/ 2 levels \"DE\"<\"ATU\": 2 2 2 2 2 2 2 2 2 2 ...\n",
      " $ transcript_id                    : chr  \"ENST00000398863\" \"ENST00000373798\" \"ENST00000398863\" \"ENST00000347529\" ...\n",
      " $ Beta                             : num  0.548 0.642 -0.924 -1.078 1.048 ...\n",
      " $ Beta_Z_score                     : num  0.631 0.769 -0.5 -0.56 1.112 ...\n",
      " $ adjusted_pval                    : num  0.4571 0.4571 0.4677 0.4677 0.0741 ...\n",
      " $ adjusted_minus_logpval           : num  0.34 0.34 0.33 0.33 1.13 1.46 1.46 0.23 NA 0.33 ...\n",
      " $ regulated_gene_in_Gene_body_CLASS: Ord.factor w/ 2 levels \"Variant_outside_gene_body_of_regulated_gene\"<..: 2 2 2 2 2 2 2 2 2 2 ...\n",
      "\n",
      " chr [1:24] \"chr1_29217311_G_A\" \"chr1_92925654_G_C\" \"chr14_23587046_A_C\" ...\n",
      "\n",
      " chr [1:25] \"ENSG00000159023\" \"ENSG00000067208\" \"ENSG00000237054\" ...\n",
      "\n",
      "Whole blood Monocytes Neutrophils naive T-CD4 Cells\n",
      "68 42 42 42\n",
      "EPB41 EVI5 PRMT5-AS1 ATL1 NPRL3 CCR7 VMP1 ELP5 BCL2 KDSR RASAL3 SLC11A1 MFSD2B ARSA ODF3B TYMP GATA2 EEFSEC PLCL2 FOXP1 TAF8 PILRB IKZF1 ANK1 GFI1B\n"
     ]
    }
   ],
   "source": [
    " #### READ Table_S7 ----\n",
    "  \n",
    "  Table_S7<-readRDS(file=\"/group/soranzo/manuel.tardaguila/Paper_bits/FIX_TABLES/Provisional_Tables/Table_S7_Provisional.rds\")\n",
    "  \n",
    "  cat(\"Table_S7_0\\n\")\n",
    "  cat(str(Table_S7))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S7$VAR)))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S7$ensembl_gene_id)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  Table_S7$Analysis[which(Table_S7$Analysis == 'DTU')]<-'ATU'\n",
    "  \n",
    "  Table_S7_ATU<-Table_S7[which(Table_S7$Analysis == 'ATU'),]\n",
    "  \n",
    " \n",
    "  cat(\"Table_S7_ATU_0\\n\")\n",
    "  cat(str(Table_S7_ATU))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S7_ATU$VAR)))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S7_ATU$ensembl_gene_id)))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(names(summary(as.factor(Table_S7_ATU$RNASeq_source))))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(summary(as.factor(Table_S7_ATU$RNASeq_source)))))\n",
    "  cat(\"\\n\")\n",
    "  # \n",
    "  # check<-Table_S7_ATU[which(Table_S7_ATU$VAR == 'chr3_71355240_G_C'),]\n",
    "  # \n",
    "  # cat(\"check_3\\n\")\n",
    "  # cat(str(check))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(unique(check$HGNC))))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(str(unique(check$transcript_id)))\n",
    "  # cat(\"\\n\")\n",
    "  \n",
    "  \n",
    "  Table_S7_ATU_subset<-merge(Table_S7_ATU,\n",
    "                             Table_S6_subset_ATU.m_NO_NA,\n",
    "                             by=c(\"VAR\",'rs',\"HGNC\"))\n",
    "  \n",
    "  \n",
    "  \n",
    "  cat(\"Table_S7_ATU_subset_0\\n\")\n",
    "  cat(str(Table_S7_ATU_subset))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S7_ATU_subset$VAR)))\n",
    "  cat(\"\\n\")\n",
    "  cat(str(unique(Table_S7_ATU_subset$ensembl_gene_id)))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(names(summary(as.factor(Table_S7_ATU_subset$RNASeq_source))))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(summary(as.factor(Table_S7_ATU_subset$RNASeq_source)))))\n",
    "  cat(\"\\n\")\n",
    "  cat(sprintf(as.character(unique(Table_S7_ATU_subset$HGNC))))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  # check<-Table_S7_ATU_subset[which(Table_S7_ATU_subset$VAR == 'chr3_71355240_G_C'),]\n",
    "  # \n",
    "  # cat(\"check_4\\n\")\n",
    "  # cat(str(check))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(unique(check$HGNC))))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(str(unique(check$transcript_id)))\n",
    "  # cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "bfe2ed88-6dd7-4f12-beed-7e53a0f053c7",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:18:24.793196Z",
     "iopub.status.busy": "2025-06-24T16:18:24.792326Z",
     "iopub.status.idle": "2025-06-24T16:18:24.808026Z",
     "shell.execute_reply": "2025-06-24T16:18:24.807036Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>VAR</th><th scope=col>rs</th><th scope=col>HGNC</th><th scope=col>ensembl_gene_id</th><th scope=col>RNASeq_source</th><th scope=col>Analysis</th><th scope=col>transcript_id</th><th scope=col>Beta</th><th scope=col>Beta_Z_score</th><th scope=col>adjusted_pval</th><th scope=col>adjusted_minus_logpval</th><th scope=col>regulated_gene_in_Gene_body_CLASS</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;ord&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;ord&gt;</th><th scope=col>&lt;ord&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;ord&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>12</th><td>chr1_92925654_G_C</td><td>rs150649461</td><td>EVI5</td><td>ENSG00000067208</td><td>Whole blood</td><td>ATU</td><td>ENST00000466778</td><td>-0.3370677</td><td>-1.91579878</td><td>0.003090295</td><td>2.51</td><td>Variant_outside_gene_body_of_regulated_gene</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>chr1_92925654_G_C</td><td>rs150649461</td><td>EVI5</td><td>ENSG00000067208</td><td>Whole blood</td><td>ATU</td><td>ENST00000370331</td><td> 0.0000000</td><td>-0.03795929</td><td>         NA</td><td>  NA</td><td>Variant_outside_gene_body_of_regulated_gene</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & VAR & rs & HGNC & ensembl\\_gene\\_id & RNASeq\\_source & Analysis & transcript\\_id & Beta & Beta\\_Z\\_score & adjusted\\_pval & adjusted\\_minus\\_logpval & regulated\\_gene\\_in\\_Gene\\_body\\_CLASS\\\\\n",
       "  & <chr> & <ord> & <chr> & <chr> & <ord> & <ord> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <ord>\\\\\n",
       "\\hline\n",
       "\t12 & chr1\\_92925654\\_G\\_C & rs150649461 & EVI5 & ENSG00000067208 & Whole blood & ATU & ENST00000466778 & -0.3370677 & -1.91579878 & 0.003090295 & 2.51 & Variant\\_outside\\_gene\\_body\\_of\\_regulated\\_gene\\\\\n",
       "\t13 & chr1\\_92925654\\_G\\_C & rs150649461 & EVI5 & ENSG00000067208 & Whole blood & ATU & ENST00000370331 &  0.0000000 & -0.03795929 &          NA &   NA & Variant\\_outside\\_gene\\_body\\_of\\_regulated\\_gene\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 12\n",
       "\n",
       "| <!--/--> | VAR &lt;chr&gt; | rs &lt;ord&gt; | HGNC &lt;chr&gt; | ensembl_gene_id &lt;chr&gt; | RNASeq_source &lt;ord&gt; | Analysis &lt;ord&gt; | transcript_id &lt;chr&gt; | Beta &lt;dbl&gt; | Beta_Z_score &lt;dbl&gt; | adjusted_pval &lt;dbl&gt; | adjusted_minus_logpval &lt;dbl&gt; | regulated_gene_in_Gene_body_CLASS &lt;ord&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 12 | chr1_92925654_G_C | rs150649461 | EVI5 | ENSG00000067208 | Whole blood | ATU | ENST00000466778 | -0.3370677 | -1.91579878 | 0.003090295 | 2.51 | Variant_outside_gene_body_of_regulated_gene |\n",
       "| 13 | chr1_92925654_G_C | rs150649461 | EVI5 | ENSG00000067208 | Whole blood | ATU | ENST00000370331 |  0.0000000 | -0.03795929 |          NA |   NA | Variant_outside_gene_body_of_regulated_gene |\n",
       "\n"
      ],
      "text/plain": [
       "   VAR               rs          HGNC ensembl_gene_id RNASeq_source Analysis\n",
       "12 chr1_92925654_G_C rs150649461 EVI5 ENSG00000067208 Whole blood   ATU     \n",
       "13 chr1_92925654_G_C rs150649461 EVI5 ENSG00000067208 Whole blood   ATU     \n",
       "   transcript_id   Beta       Beta_Z_score adjusted_pval adjusted_minus_logpval\n",
       "12 ENST00000466778 -0.3370677 -1.91579878  0.003090295   2.51                  \n",
       "13 ENST00000370331  0.0000000 -0.03795929           NA     NA                  \n",
       "   regulated_gene_in_Gene_body_CLASS          \n",
       "12 Variant_outside_gene_body_of_regulated_gene\n",
       "13 Variant_outside_gene_body_of_regulated_gene"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Table_S7_ATU_subset[which(Table_S7_ATU_subset$VAR == 'chr1_92925654_G_C'),]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40af5a55-7003-4038-9759-4999e9eea354",
   "metadata": {},
   "source": [
    "# Subset ensembl_gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "44cf446b-7213-4bd1-9fe8-e2eb9a4bd0c1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:18:24.810462Z",
     "iopub.status.busy": "2025-06-24T16:18:24.809611Z",
     "iopub.status.idle": "2025-06-24T16:18:24.923199Z",
     "shell.execute_reply": "2025-06-24T16:18:24.922049Z"
    }
   },
   "outputs": [],
   "source": [
    " #### subset the ensembl gtf ----\n",
    "  \n",
    "  \n",
    "  ensembl_gtf_sel<-unique(ensembl_gtf[which(ensembl_gtf$gene_id%in%Table_S7_ATU_subset$ensembl_gene_id &\n",
    "                                       ensembl_gtf$type%in%c('gene','transcript','exon','CDS','five_prime_utr','three_prime_utr')),])\n",
    "  \n",
    "  # cat(\"ensembl_gtf_sel_0\\n\")\n",
    "  # cat(str(ensembl_gtf_sel))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(str(unique(ensembl_gtf_sel$gene_id)))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(names(summary(as.factor(ensembl_gtf_sel$transcript_biotype))))))\n",
    "  # cat(\"\\n\")\n",
    "  # cat(sprintf(as.character(summary(as.factor(ensembl_gtf_sel$transcript_biotype)))))\n",
    "  # cat(\"\\n\")\n",
    "\n",
    "rm(ensembl_gtf)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "607d9065-d45a-498c-9536-ba245167e844",
   "metadata": {},
   "source": [
    "# General path graphs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "a3781acc-b3cd-4c77-9c00-3d76eb65895a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:18:24.925757Z",
     "iopub.status.busy": "2025-06-24T16:18:24.924875Z",
     "iopub.status.idle": "2025-06-24T16:18:24.939575Z",
     "shell.execute_reply": "2025-06-24T16:18:24.938590Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_graphs<-paste('/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/test/','Graphs_selected_ATU','/',sep='')\n",
    "  \n",
    "  if (file.exists(path_graphs)){\n",
    "    \n",
    "    \n",
    "  }else{\n",
    "    \n",
    "    dir.create(file.path(path_graphs))\n",
    "    \n",
    "  }#path_graphs\n",
    "\n",
    "path_figures = paste(\"/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/test/Figure_graphs/\",sep='')\n",
    "    \n",
    "    if(file.exists(path_figures)){\n",
    "      \n",
    "    }else{\n",
    "      dir.create(path_figures)\n",
    "    }"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1d2b1ff-8020-4598-af79-b0b17fd649ee",
   "metadata": {},
   "source": [
    "# LOOP to print graphs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "131a952c-fb4a-4efa-a555-650dcbfbbb94",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:18:24.942027Z",
     "iopub.status.busy": "2025-06-24T16:18:24.941168Z",
     "iopub.status.idle": "2025-06-24T16:18:24.955059Z",
     "shell.execute_reply": "2025-06-24T16:18:24.954055Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VARS_0\n",
      " chr [1:24] \"chr1_29217311_G_A\" \"chr1_92925654_G_C\" \"chr14_23587046_A_C\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "VARS<-unique(Table_S7_ATU_subset$VAR)\n",
    "  \n",
    "  # VARS<-'chr7_101499930_G_A'\n",
    "  \n",
    "  # VARS<-'chr1_92925654_G_C'\n",
    "  \n",
    "  cat(\"VARS_0\\n\")\n",
    "  cat(str(VARS))\n",
    "  cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "6c097343-cc7c-4eed-ba68-288fa877db9d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:18:24.957537Z",
     "iopub.status.busy": "2025-06-24T16:18:24.956653Z",
     "iopub.status.idle": "2025-06-24T16:18:24.963700Z",
     "shell.execute_reply": "2025-06-24T16:18:24.962724Z"
    }
   },
   "outputs": [],
   "source": [
    "START<-1"
   ]
  },
  {
   "cell_type": "raw",
   "id": "fe107130-6c43-4d50-9c28-0ccb9fca5138",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:05:22.746995Z",
     "iopub.status.busy": "2025-06-24T16:05:22.745995Z",
     "iopub.status.idle": "2025-06-24T16:05:22.757326Z",
     "shell.execute_reply": "2025-06-24T16:05:22.756273Z"
    }
   },
   "source": [
    "START<-which(VARS == 'chr3_71355240_G_C')\n",
    "START"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "e0b75116-d26a-4577-a811-5f210709206a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-06-24T16:18:24.966758Z",
     "iopub.status.busy": "2025-06-24T16:18:24.965928Z",
     "iopub.status.idle": "2025-06-24T16:19:03.397149Z",
     "shell.execute_reply": "2025-06-24T16:19:03.395993Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------------------------------------------------->\t1\trs115820925\tchr1_29217311_G_A\n",
      "--------------------------->\t1\tEPB41\n",
      "Printed_transcript model\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----------------> printed BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t2\trs150649461\tchr1_92925654_G_C\n",
      "--------------------------->\t1\tEVI5\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t3\trs147453535\tchr14_23587046_A_C\n",
      "--------------------------->\t1\tPRMT5-AS1\n",
      "Printed_transcript model\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----------------> printed BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t4\trs550463489\tchr14_51132622_T_C\n",
      "--------------------------->\t1\tATL1\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t5\trs187174553\tchr16_155132_C_T\n",
      "--------------------------->\t1\tNPRL3\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t6\trs112401631\tchr17_38764524_T_A\n",
      "--------------------------->\t1\tCCR7\n",
      "Printed_transcript model\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----------------> printed BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t7\trs142122062\tchr17_58602131_G_A\n",
      "--------------------------->\t1\tVMP1\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t8\trs200489612\tchr17_7106378_G_A\n",
      "--------------------------->\t1\tELP5\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t9\trs17758695\tchr18_60920854_C_T\n",
      "--------------------------->\t1\tBCL2\n",
      "Printed_transcript model\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----------------> printed BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "--------------------------->\t2\tKDSR\n",
      "Printed_transcript model\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 3 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 3 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 3 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 3 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----------------> printed BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t10\trs569022307\tchr19_15653669_T_C\n",
      "--------------------------->\t1\tRASAL3\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t11\trs114050631\tchr2_219020958_C_T\n",
      "--------------------------->\t1\tSLC11A1\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 2\n",
      "---------------------------------------------------------------------------------->\t12\trs150221884\tchr2_24091099_C_T\n",
      "--------------------------->\t1\tMFSD2B\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t13\trs543594419\tchr22_50949811_T_C\n",
      "--------------------------->\t1\tARSA\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 2\n",
      "--------------------------->\t2\tODF3B\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 2\n",
      "--------------------------->\t3\tTYMP\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 2\n",
      "---------------------------------------------------------------------------------->\t14\trs191451841\tchr3_128317978_C_T\n",
      "--------------------------->\t1\tGATA2\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t15\trs112097551\tchr3_128322617_G_A\n",
      "--------------------------->\t1\tEEFSEC\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t16\trs116577908\tchr3_17098399_A_G\n",
      "--------------------------->\t1\tPLCL2\n",
      "Printed_transcript model\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----------------> printed BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t17\trs35592432\tchr3_71355240_G_C\n",
      "--------------------------->\t1\tFOXP1\n",
      "Printed_transcript model\n",
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------->Hello world\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----------------> printed BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 1 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t18\trs112233623\tchr6_41924998_C_T\n",
      "--------------------------->\t1\tTAF8\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t19\trs113267280\tchr6_41952511_T_G\n",
      "--------------------------->\t1\tTAF8\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t20\trs546810441\tchr7_100083971_G_A\n",
      "--------------------------->\t1\tPILRB\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t21\trs150640087\tchr7_50444152_G_T\n",
      "--------------------------->\t1\tIKZF1\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t22\trs185672063\tchr7_99760955_G_C\n",
      "--------------------------->\t1\tPILRB\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 1\n",
      "---------------------------------------------------------------------------------->\t23\trs149489081\tchr8_41589736_T_G\n",
      "--------------------------->\t1\tANK1\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in melt(Transposed_Isoform_Expression_sel, id.vars = c(\"sample_id\"), :\n",
      "“The melt generic in data.table has been passed a data.frame and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is deprecated, and this redirection is now deprecated as well. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace like reshape2::melt(Transposed_Isoform_Expression_sel). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_summary).”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_ydensity).”\n",
      "Warning message:\n",
      "“Removed 2 rows containing non-finite values (stat_summary).”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Printed transcripts INTERVAL\n",
      "---------------------------------------------------------------------------------->\t24\trs187715179\tchr9_135920196_C_T\n",
      "--------------------------->\t1\tGFI1B\n",
      "Printed_transcript model\n",
      "-----------------> No BluePrint results:\tNo results INTERVAL 2\n"
     ]
    }
   ],
   "source": [
    "DEBUG<-0\n",
    "\n",
    "for(i in START:length(VARS))\n",
    "{\n",
    "  VAR_sel<-VARS[i]\n",
    "  \n",
    "  \n",
    "  \n",
    "  Table_S7_ATU_subset_sel<-Table_S7_ATU_subset[which(Table_S7_ATU_subset$VAR == VAR_sel),]\n",
    "  \n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"Table_S7_ATU_subset_sel_0\\n\")\n",
    "    cat(str(Table_S7_ATU_subset_sel))\n",
    "    cat(\"\\n\")\n",
    "    \n",
    "  }\n",
    "  \n",
    "  rs_sel<-unique(as.character(Table_S7_ATU_subset_sel$rs))\n",
    "  \n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"rs_sel_0\\n\")\n",
    "    cat(str(rs_sel))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "  \n",
    "  path_graphs_rs<-paste(path_graphs,paste(rs_sel,VAR_sel, sep='__'),'/',sep='')\n",
    "  \n",
    "  if (file.exists(path_graphs_rs)){\n",
    "    \n",
    "    \n",
    "  }else{\n",
    "    \n",
    "    dir.create(file.path(path_graphs_rs))\n",
    "    \n",
    "  }#path_graphs_rs\n",
    "  \n",
    "  ENSG_array<-unique(Table_S7_ATU_subset_sel$ensembl_gene_id)\n",
    "  \n",
    "  if(DEBUG == 1)\n",
    "  {\n",
    "    cat(\"ENSG_array_0\\n\")\n",
    "    cat(str(ENSG_array))\n",
    "    cat(\"\\n\")\n",
    "  }\n",
    "  \n",
    "  cat(\"---------------------------------------------------------------------------------->\\t\")\n",
    "  cat(sprintf(as.character(i)))\n",
    "  cat(\"\\t\")\n",
    "  cat(sprintf(as.character(rs_sel)))\n",
    "  cat(\"\\t\")\n",
    "  cat(sprintf(as.character(VAR_sel)))\n",
    "  cat(\"\\n\")\n",
    "  \n",
    "  \n",
    "  for(k in 1:length(ENSG_array))\n",
    "  {\n",
    "    ENSG_array_sel<-ENSG_array[k]\n",
    "    \n",
    "    Table_S7_ATU_subset_sel_ENSG_sel<-Table_S7_ATU_subset_sel[which(Table_S7_ATU_subset_sel$ensembl_gene_id == ENSG_array_sel),]\n",
    "    \n",
    "    Table_S7_ATU_subset_sel_ENSG_sel$REF<-NA\n",
    "    \n",
    "    Table_S7_ATU_subset_sel_ENSG_sel$REF[is.na(Table_S7_ATU_subset_sel_ENSG_sel$adjusted_minus_logpval)]<-'reference'\n",
    "    Table_S7_ATU_subset_sel_ENSG_sel$REF[!is.na(Table_S7_ATU_subset_sel_ENSG_sel$adjusted_minus_logpval)]<-'not_reference'\n",
    "    \n",
    "    Table_S7_ATU_subset_sel_ENSG_sel$REF<-factor(Table_S7_ATU_subset_sel_ENSG_sel$REF,\n",
    "                                                 levels=c('not_reference','reference'),\n",
    "                                                 ordered=T)\n",
    "    \n",
    "    \n",
    "    Table_S7_ATU_subset_sel_ENSG_sel[order(Table_S7_ATU_subset_sel_ENSG_sel$RNASeq_source,Table_S7_ATU_subset_sel_ENSG_sel$REF),]\n",
    "    \n",
    "    order_transcript_id<-unique(Table_S7_ATU_subset_sel_ENSG_sel$transcript_id)\n",
    "    \n",
    "    newG<-c(\"HOM_REF\",\"HET\")\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"Table_S7_ATU_subset_sel_ENSG_sel_0\\n\")\n",
    "      cat(str(Table_S7_ATU_subset_sel_ENSG_sel))\n",
    "      cat(\"\\n\")\n",
    "      cat(str(unique(Table_S7_ATU_subset_sel_ENSG_sel$ensembl_gene_id)))\n",
    "      cat(\"\\n\")\n",
    "      cat(sprintf(as.character(order_transcript_id)))\n",
    "      cat(\"\\n\")\n",
    "      \n",
    "      cat(\"ORDER transcript_id\\n\")\n",
    "      cat(sprintf(as.character(Table_S7_ATU_subset_sel_ENSG_sel$transcript_id)))\n",
    "      cat(\"\\n\")\n",
    "      \n",
    "      \n",
    "    }\n",
    "    \n",
    "    ensembl_gtf_sel_ENSG_sel<-ensembl_gtf_sel[which(ensembl_gtf_sel$gene_id == ENSG_array_sel &\n",
    "                                                      ensembl_gtf_sel$type == 'gene'),]\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"ensembl_gtf_sel_ENSG_sel_0\\n\")\n",
    "      cat(str(ensembl_gtf_sel_ENSG_sel))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    HGNC_sel<-unique(ensembl_gtf_sel_ENSG_sel$gene_name)\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"HGNC_sel_0\\n\")\n",
    "      cat(str(HGNC_sel))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    \n",
    "    chr_sel<-unique(as.character(ensembl_gtf_sel_ENSG_sel$seqid))\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"chr_sel_0\\n\")\n",
    "      cat(str(chr_sel))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    start_sel<-unique(ensembl_gtf_sel_ENSG_sel$start)\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"start_sel_0\\n\")\n",
    "      cat(str(start_sel))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    end_sel<-unique(ensembl_gtf_sel_ENSG_sel$end)\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"end_sel_0\\n\")\n",
    "      cat(str(end_sel))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    strand_sel<-unique(ensembl_gtf_sel_ENSG_sel$strand)\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"strand_sel_0\\n\")\n",
    "      cat(str(strand_sel))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    cat(\"--------------------------->\\t\")    \n",
    "    cat(sprintf(as.character(k)))\n",
    "    cat(\"\\t\")\n",
    "    cat(sprintf(as.character(HGNC_sel)))\n",
    "    cat(\"\\n\")\n",
    "    \n",
    "    \n",
    "    path_graphs_rs_HGNC<-paste(path_graphs_rs,HGNC_sel,'/',sep='')\n",
    "    \n",
    "    if (file.exists(path_graphs_rs_HGNC)){\n",
    "      \n",
    "      \n",
    "    }else{\n",
    "      \n",
    "      dir.create(file.path(path_graphs_rs_HGNC))\n",
    "      \n",
    "    }#path_graphs_rs\n",
    "    \n",
    "    \n",
    "    #### transcript plot -------------------------------------------------------------------------------------------------------\n",
    "    \n",
    "    transcript_id_array<-unique(Table_S7_ATU_subset_sel_ENSG_sel$transcript_id)\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"transcript_id_array_0\\n\")\n",
    "      cat(str(transcript_id_array))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    \n",
    "    ensembl_gtf_sel_transcript_sel<-ensembl_gtf_sel[which(ensembl_gtf_sel$transcript_id %in% transcript_id_array &\n",
    "                                                            ensembl_gtf_sel$type == 'transcript'),]\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"ensembl_gtf_sel_transcript_sel_0\\n\")\n",
    "      cat(str(ensembl_gtf_sel_transcript_sel))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    \n",
    "    check<-Table_S7_ATU_subset_sel_ENSG_sel[-which(Table_S7_ATU_subset_sel_ENSG_sel$transcript_id%in%ensembl_gtf_sel_transcript_sel$transcript_id),]\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"check_0\\n\")\n",
    "      cat(str(check))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    if(dim(check)[1]>0)\n",
    "    {\n",
    "      \n",
    "      ### None\n",
    "      \n",
    "    }#dim(check)[1]>0\n",
    "    \n",
    "    \n",
    "    REP<-data.frame()\n",
    "    for(l in 1:length(transcript_id_array))\n",
    "    {\n",
    "      \n",
    "      transcript_id_array_sel<-transcript_id_array[l]\n",
    "      \n",
    "      \n",
    "      # cat(\"---------------------------------------------------------------------------------->\\t\")\n",
    "      # cat(sprintf(as.character(l)))\n",
    "      # cat(\"\\t\")\n",
    "      # cat(sprintf(as.character(transcript_id_array_sel)))\n",
    "      # cat(\"\\n\")\n",
    "      \n",
    "      ensembl_gtf_sel_transcript_sel<-ensembl_gtf_sel[which(ensembl_gtf_sel$transcript_id == transcript_id_array_sel &\n",
    "                                                              ensembl_gtf_sel$type != 'gene'),]\n",
    "      # \n",
    "      # if(DEBUG == 1)\n",
    "      # {\n",
    "      #   cat(\"ensembl_gtf_sel_transcript_sel_0\\n\")\n",
    "      #   cat(str(ensembl_gtf_sel_transcript_sel))\n",
    "      #   cat(\"\\n\")\n",
    "      # }\n",
    "      \n",
    "      ensembl_gtf_sel_transcript_sel$COORD<-l\n",
    "      \n",
    "      REP<-rbind(ensembl_gtf_sel_transcript_sel,REP)\n",
    "      \n",
    "    }#l in 1:length(transcript_id_array)\n",
    "    \n",
    "    \n",
    "    colnames(REP)[which(colnames(REP) == 'seqid')]<-'seqnames'\n",
    "    \n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"REP_0\\n\")\n",
    "      cat(str(REP))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    REP$transcript_biotype<-factor(REP$transcript_biotype,\n",
    "                                   levels=c(\"protein_coding\",\"retained_intron\",\"nonsense_mediated_decay\",\"processed_transcript\",\"antisense\"),\n",
    "                                   ordered=T)\n",
    "    \n",
    "    REP$transcript_id<-factor(REP$transcript_id,\n",
    "                              levels=rev(order_transcript_id),\n",
    "                              ordered=T)\n",
    "    \n",
    "    \n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"REP_1\\n\")\n",
    "      cat(str(REP))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    \n",
    "    vector.fill<-c(brewer.pal(length(levels(REP$transcript_biotype)), \"Dark2\"))\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"vector.fill_0\\n\")\n",
    "      cat(str(vector.fill))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    \n",
    "    \n",
    "    # extract exons\n",
    "    \n",
    "    REP_exons <- REP %>% dplyr::filter(type == \"exon\")\n",
    "    \n",
    "    REP_exons_rescaled <- shorten_gaps(\n",
    "      REP_exons, \n",
    "      to_intron(REP_exons, \"transcript_name\"), \n",
    "      group_var = \"transcript_name\"\n",
    "    )\n",
    "    \n",
    "    transcript_plot<-REP_exons_rescaled %>%\n",
    "      dplyr::filter(type == \"exon\") %>%\n",
    "      ggplot(aes(\n",
    "        xstart = start,\n",
    "        xend = end,\n",
    "        y = transcript_id\n",
    "      )) +\n",
    "      geom_range(\n",
    "        aes(fill = transcript_biotype)\n",
    "      ) +\n",
    "      geom_intron(\n",
    "        data = REP_exons_rescaled %>% dplyr::filter(type == \"intron\"),\n",
    "        aes(strand=strand),\n",
    "        arrow = grid::arrow(ends = \"last\", length = grid::unit(0.25, \"cm\")),\n",
    "        arrow.min.intron.length = 200\n",
    "      )\n",
    "    \n",
    "    transcript_plot<-transcript_plot+\n",
    "      scale_fill_manual(values=vector.fill,drop=F)+\n",
    "      theme_classic()+\n",
    "      theme(axis.title.y=element_blank(),\n",
    "            axis.title.x=element_blank(),\n",
    "            axis.text.y=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "            axis.text.x=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "            axis.line.x = element_line(size = 0.2),\n",
    "            axis.ticks = element_line(size = 0.2),\n",
    "            axis.line.y = element_line(size = 0.2))+\n",
    "      theme(legend.title = element_blank(),\n",
    "            legend.text = element_text(size=6),\n",
    "            legend.key.size = unit(0.25, 'cm'), #change legend key size\n",
    "            legend.key.height = unit(0.25, 'cm'), #change legend key height\n",
    "            legend.key.width = unit(0.25, 'cm'), #change legend key width\n",
    "            legend.position=\"bottom\")\n",
    "    \n",
    "    # scale_x_continuous(name=NULL,breaks=c(start_sel,end_sel),\n",
    "    #                    labels=as.character(c(start_sel,end_sel)),\n",
    "    #                    limits=c(start_sel-1,end_sel+1))+\n",
    "    \n",
    "    setwd(path_graphs_rs_HGNC)\n",
    "    \n",
    "    svgname<-paste(paste('Transcripts_analysed', sep='_'),\".svg\",sep='')\n",
    "    makesvg = TRUE\n",
    "    \n",
    "    if (makesvg == TRUE)\n",
    "    {\n",
    "      ggsave(svgname, plot= transcript_plot,\n",
    "             device=\"svg\",\n",
    "             height=4, width=4)\n",
    "    }\n",
    "    \n",
    "    cat(\"Printed_transcript model\\n\")\n",
    "    \n",
    "    #### violin plots of expression  BluePrint -------------------------------------------------------------------------------------------------------\n",
    "\n",
    "      \n",
    "    BLUEPRINT_files<-paste(path_BP,VAR_sel,'/', sep='')\n",
    "    filename<-paste(BLUEPRINT_files,'BP_Transcript_EXP_for_LM.rds',sep='')\n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"BLUEPRINT_files\\n\")\n",
    "      cat(sprintf(as.character(BLUEPRINT_files)))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    \n",
    "    \n",
    "    if (file.exists(filename)){\n",
    "      \n",
    "      setwd(BLUEPRINT_files)\n",
    "      \n",
    "      EXP_BP_df = readRDS(file='BP_Transcript_EXP_for_LM.rds')\n",
    "      \n",
    "      if(DEBUG == 1)\n",
    "      {\n",
    "        cat(\"EXP_BP_df\\n\")\n",
    "        cat(str(EXP_BP_df))\n",
    "        cat(\"\\n\")\n",
    "        cat(str(unique(EXP_BP_df$ensembl_gene_id)))\n",
    "        cat(\"\\n\")\n",
    "        cat(str(unique(EXP_BP_df$transcript_id)))\n",
    "        cat(\"\\n\")\n",
    "      }\n",
    "      \n",
    "      EXP_BP_df_sel<-EXP_BP_df[which(EXP_BP_df$ensembl_gene_id == ENSG_array_sel &\n",
    "                                       EXP_BP_df$transcript_id%in%order_transcript_id),]\n",
    "      \n",
    "      if(DEBUG == 1)\n",
    "      {\n",
    "        cat(\"EXP_BP_df_sel\\n\")\n",
    "        cat(str(EXP_BP_df_sel))\n",
    "        cat(\"\\n\")\n",
    "        cat(str(unique(EXP_BP_df_sel$ensembl_gene_id)))\n",
    "        cat(\"\\n\")\n",
    "        cat(str(unique(EXP_BP_df_sel$transcript_id)))\n",
    "        cat(\"\\n\")\n",
    "        \n",
    "        cat(\"EXP_BP_df_sel$transcript_id\\n\")\n",
    "        cat(sprintf(as.character(unique(EXP_BP_df_sel$transcript_id))))\n",
    "        cat(\"\\n\")\n",
    "        \n",
    "        cat(\"order_transcript_id\\n\")\n",
    "        cat(sprintf(as.character(order_transcript_id)))\n",
    "        cat(\"\\n\")\n",
    "        \n",
    "        cat(\"newG\\n\")\n",
    "        cat(sprintf(as.character(newG)))\n",
    "        cat(\"\\n\")\n",
    "      }\n",
    "      \n",
    "      \n",
    "      EXP_BP_df_sel$FPKM<-2^(EXP_BP_df_sel$value)\n",
    "      \n",
    "      EXP_BP_df_sel$transcript_id<-factor(EXP_BP_df_sel$transcript_id,\n",
    "                                          levels=order_transcript_id,\n",
    "                                          ordered=T)\n",
    "      \n",
    "      EXP_BP_df_sel$Cell_Type<-factor(EXP_BP_df_sel$Cell_Type, \n",
    "                                      levels=c(\"Monocyte\",\"Neutrophil\",\"Tcell\"), \n",
    "                                      ordered=T)\n",
    "      \n",
    "      \n",
    "      EXP_BP_df_sel$Genotype<-factor(EXP_BP_df_sel$Genotype,\n",
    "                                     levels=newG,\n",
    "                                     ordered=T)\n",
    "      \n",
    "      \n",
    "      if(DEBUG == 1)\n",
    "      {\n",
    "        cat(\"EXP_BP_df_sel\\n\")\n",
    "        cat(str(EXP_BP_df_sel))\n",
    "        cat(\"\\n\")\n",
    "        cat(sprintf(as.character(names(summary(EXP_BP_df_sel$Cell_Type)))))\n",
    "        cat(\"\\n\")\n",
    "        cat(sprintf(as.character(summary(EXP_BP_df_sel$Cell_Type))))\n",
    "        cat(\"\\n\")\n",
    "        cat(sprintf(as.character(names(summary(EXP_BP_df_sel$transcript_id)))))\n",
    "        cat(\"\\n\")\n",
    "        cat(sprintf(as.character(summary(EXP_BP_df_sel$transcript_id))))\n",
    "        cat(\"\\n\")\n",
    "        cat(sprintf(as.character(names(summary(EXP_BP_df_sel$Genotype)))))\n",
    "        cat(\"\\n\")\n",
    "        cat(sprintf(as.character(summary(EXP_BP_df_sel$Genotype))))\n",
    "        cat(\"\\n\")\n",
    "      }    \n",
    "      \n",
    "      \n",
    "      \n",
    "      \n",
    "      array_transcript_id<-unique(EXP_BP_df_sel$transcript_id)\n",
    "      \n",
    "      \n",
    "      if(DEBUG == 1)\n",
    "      {\n",
    "        cat(\"array_transcript_id_0\\n\")\n",
    "        cat(str(array_transcript_id))\n",
    "        cat(\"\\n\")\n",
    "      }\n",
    "      \n",
    "      for(iteration_array_transcript_id in 1:length(array_transcript_id)){\n",
    "        \n",
    "        array_transcript_id_sel<-array_transcript_id[iteration_array_transcript_id]\n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          cat(sprintf(as.character(iteration_array_transcript_id)))\n",
    "          cat(\"\\t\")\n",
    "          cat(sprintf(as.character(array_transcript_id_sel)))\n",
    "          cat(\"\\n\")\n",
    "        }\n",
    "        \n",
    "        EXP_BP_df_sel_transcript_id_sel<-droplevels(EXP_BP_df_sel[which(EXP_BP_df_sel$transcript_id == array_transcript_id_sel),])\n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          cat(\"EXP_BP_df_sel_transcript_id_sel_0\\n\")\n",
    "          cat(str(EXP_BP_df_sel_transcript_id_sel))\n",
    "          cat(\"\\n\")\n",
    "        }\n",
    "        \n",
    "        \n",
    "        array_Cell_Type<-unique(EXP_BP_df_sel_transcript_id_sel$Cell_Type)\n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          cat(\"array_Cell_Type_0\\n\")\n",
    "          cat(str(array_Cell_Type))\n",
    "          cat(\"\\n\")\n",
    "        }\n",
    "        \n",
    "        for(iteration_array_Cell_Type in 1:length(array_Cell_Type)){\n",
    "          \n",
    "          array_Cell_Type_sel<-array_Cell_Type[iteration_array_Cell_Type]\n",
    "          \n",
    "          if(DEBUG == 1)\n",
    "          {\n",
    "            cat(sprintf(as.character(iteration_array_Cell_Type)))\n",
    "            cat(\"\\t\")\n",
    "            cat(sprintf(as.character(array_Cell_Type_sel)))\n",
    "            cat(\"\\n\")\n",
    "          }\n",
    "          \n",
    "          EXP_BP_df_sel_transcript_id_sel_Cell_Type_sel<-droplevels(EXP_BP_df_sel_transcript_id_sel[which(EXP_BP_df_sel_transcript_id_sel$Cell_Type == array_Cell_Type_sel),])\n",
    "          \n",
    "          if(DEBUG == 1)\n",
    "          {\n",
    "            cat(\"EXP_BP_df_sel_transcript_id_sel_Cell_Type_sel_0\\n\")\n",
    "            cat(str(EXP_BP_df_sel_transcript_id_sel_Cell_Type_sel))\n",
    "            cat(\"\\n\")\n",
    "          }\n",
    "          \n",
    "          A<-round(summary(EXP_BP_df_sel_transcript_id_sel_Cell_Type_sel$value[!is.na(EXP_BP_df_sel_transcript_id_sel_Cell_Type_sel$value)]),2)\n",
    "          \n",
    "          \n",
    "          # cat(\"summary_GENE_EXP\\n\")\n",
    "          # cat(sprintf(as.character(names(A))))\n",
    "          # cat(\"\\n\")\n",
    "          # cat(sprintf(as.character(A)))\n",
    "          # cat(\"\\n\")\n",
    "          \n",
    "          step<-abs(A[6]-A[1])/4\n",
    "          \n",
    "          if(step == 0)\n",
    "          {\n",
    "            \n",
    "            step<-1\n",
    "          }\n",
    "          \n",
    "          breaks.Rank<-sort(unique(c(0,A[6],seq(from= A[1], to=A[6],by=step))))\n",
    "          labels.Rank<-as.character(round(breaks.Rank,1))\n",
    "          \n",
    "          \n",
    "          if(DEBUG == 1)\n",
    "          {\n",
    "            cat(\"labels.Rank:\\t\")\n",
    "            cat(sprintf(as.character(labels.Rank)))\n",
    "            cat(\"\\n\")\n",
    "            \n",
    "            \n",
    "            \n",
    "            # quit(status = 1)\n",
    "          }\n",
    "          \n",
    "          \n",
    "          \n",
    "          BP_per_transcript_violin<-ggplot(data=EXP_BP_df_sel_transcript_id_sel_Cell_Type_sel,\n",
    "                                           aes(y=value, x=Genotype, fill=Genotype)) +\n",
    "            geom_violin()+\n",
    "            stat_summary(fun = median, fun.min = median, fun.max = median,\n",
    "                         geom = \"crossbar\", width = 0.5)+\n",
    "            scale_y_continuous(name=\"log2 FPKM\",breaks=breaks.Rank,labels=labels.Rank, limits=c(breaks.Rank[1],breaks.Rank[length(breaks.Rank)]))+\n",
    "            scale_fill_manual(values=c(\"#E7B800\",\"#00AFBB\",\"#FC4E07\"),drop=F)+\n",
    "            scale_x_discrete(name=NULL, drop=F)\n",
    "          \n",
    "          \n",
    "          \n",
    "          BP_per_transcript_violin<-BP_per_transcript_violin+\n",
    "            facet_grid(transcript_id ~ Cell_Type, scales='free_x', space='free_x', switch=\"y\", drop=F)+\n",
    "            theme_cowplot(font_size = 4)+\n",
    "            theme( strip.background = element_blank(),\n",
    "                   strip.placement = \"outside\",\n",
    "                   strip.text = element_text(size=6),\n",
    "                   panel.spacing = unit(0.2, \"lines\"),\n",
    "                   panel.background=element_rect(fill=\"white\"),\n",
    "                   panel.border=element_rect(colour=\"white\",size=0,5),\n",
    "                   panel.grid.major = element_blank(),\n",
    "                   panel.grid.minor = element_blank())+\n",
    "            theme_classic()+\n",
    "            theme(axis.title.y=element_text(size=8, color=\"black\", family=\"sans\"),\n",
    "                  axis.title.x=element_blank(),\n",
    "                  axis.text.y=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                  axis.text.x=element_blank(),\n",
    "                  axis.line.x = element_line(size = 0.2),\n",
    "                  axis.ticks.x = element_blank(),\n",
    "                  axis.ticks.y = element_line(size = 0.2),\n",
    "                  axis.line.y = element_line(size = 0.2))+\n",
    "            theme(legend.title = element_text(size=6),\n",
    "                  legend.text = element_text(size=6),\n",
    "                  legend.key.size = unit(0.25, 'cm'), #change legend key size\n",
    "                  legend.key.height = unit(0.25, 'cm'), #change legend key height\n",
    "                  legend.key.width = unit(0.25, 'cm'), #change legend key width\n",
    "                  legend.position=\"hidden\")+\n",
    "            ggeasy::easy_center_title()\n",
    "          \n",
    "          \n",
    "          setwd(path_graphs_rs_HGNC)\n",
    "          \n",
    "          svgname<-paste(paste('BP_per_transcript_violin',array_transcript_id_sel,array_Cell_Type_sel, sep='_'),\".svg\",sep='')\n",
    "          makesvg = TRUE\n",
    "          \n",
    "          if (makesvg == TRUE)\n",
    "          {\n",
    "            ggsave(svgname, plot= BP_per_transcript_violin,\n",
    "                   device=\"svg\",\n",
    "                   height=2, width=2)\n",
    "          }\n",
    "\n",
    "            #levels=c(\"Monocyte\",\"Neutrophil\",\"Tcell\"), \n",
    "\n",
    "             if(rs_sel == 'rs17758695' & array_Cell_Type_sel %in%c('Tcell','Neutrophil')){\n",
    "\n",
    "                 setwd(path_figures)\n",
    "\n",
    "                   svgname<-paste(paste('NEW_possible',rs_sel,HGNC_sel,array_transcript_id_sel,array_Cell_Type_sel, sep='_'),\".svg\",sep='')\n",
    "                  makesvg = TRUE\n",
    "                  \n",
    "                  if (makesvg == TRUE)\n",
    "                  {\n",
    "                    ggsave(svgname, plot= BP_per_transcript_violin,\n",
    "                           device=\"svg\",\n",
    "                           height=2, width=2)\n",
    "                  }\n",
    "\n",
    "            }else{\n",
    "\n",
    "                 if(rs_sel == 'rs35592432' & array_Cell_Type_sel%in%c('Tcell','Monocyte')){\n",
    "\n",
    "                     cat(\"---------------------------------->Hello world\\n\")\n",
    "\n",
    "                        setwd(path_figures)\n",
    "\n",
    "\n",
    "                      svgname<-paste(paste('NEW_possible',rs_sel,HGNC_sel,array_transcript_id_sel,array_Cell_Type_sel, sep='_'),\".svg\",sep='')\n",
    "                      makesvg = TRUE\n",
    "                      \n",
    "                      if (makesvg == TRUE)\n",
    "                      {\n",
    "                        ggsave(svgname, plot= BP_per_transcript_violin,\n",
    "                               device=\"svg\",\n",
    "                               height=2, width=2)\n",
    "                      }\n",
    "\n",
    "                }else{\n",
    "\n",
    "                     if(rs_sel == 'rs112401631' & array_Cell_Type_sel%in%c('Monocyte')){\n",
    "\n",
    "                          setwd(path_figures)\n",
    "\n",
    "                          svgname<-paste(paste('NEW_possible',rs_sel,HGNC_sel,array_transcript_id_sel,array_Cell_Type_sel, sep='_'),\".svg\",sep='')\n",
    "                          makesvg = TRUE\n",
    "                          \n",
    "                          if (makesvg == TRUE)\n",
    "                          {\n",
    "                            ggsave(svgname, plot= BP_per_transcript_violin,\n",
    "                                   device=\"svg\",\n",
    "                                   height=2, width=2)\n",
    "                          }\n",
    "                    }#rs_sel == 'rs112401631' & %in%c('Monocyte')\n",
    "                     \n",
    "                }#rs_sel == 'rs35592432' & %in%c('Tcell','Monocyte')\n",
    "\n",
    "            }#rs_sel == 'rs17758695' & array_Cell_Type_sel %in%c('Tcell','Neutrophil')\n",
    "\n",
    "                \n",
    "          \n",
    "        }#iteration_array_Cell_Type in 1:length(array_Cell_Type)\n",
    "      }#iteration_array_transcript_id in 1:length(array_transcript_id\n",
    "      \n",
    "      cat(\"-----------------> printed BluePrint results:\\t\")\n",
    "      \n",
    "      \n",
    "    }else{\n",
    "      \n",
    "      BLUEPRINT_violin<-NA\n",
    "      \n",
    "      cat(\"-----------------> No BluePrint results:\\t\")\n",
    "      \n",
    "    }#file.exists(BLUEPRINT_files)\n",
    "    \n",
    "    #### violin plots of expression  INTERVAL whole blood----\n",
    "    \n",
    "\n",
    "    \n",
    "    INTERVAL_files<-paste(path_INTERVAL,VAR_sel,'/', sep='')\n",
    "    \n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"INTERVAL_files\\n\")\n",
    "      cat(sprintf(as.character(INTERVAL_files)))\n",
    "      cat(\"\\n\")\n",
    "    }\n",
    "    \n",
    "    if (file.exists(INTERVAL_files)){\n",
    "      \n",
    "      setwd(INTERVAL_files)\n",
    "      \n",
    "      filename<-paste(\"INTERVAL_covariates_and_PEER_factors_\",VAR_sel,\".rds\", sep='')\n",
    "      \n",
    "      if (file.exists(filename)){\n",
    "        \n",
    "        INTERVAL_covariates_and_PEER_factors_sel<-readRDS(file=filename)\n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          cat(\"INTERVAL_covariates_and_PEER_factors_sel\\n\")\n",
    "          cat(str(INTERVAL_covariates_and_PEER_factors_sel))\n",
    "          cat(\"\\n\")\n",
    "        }\n",
    "        \n",
    "        Transposed_Isoform_Expression_sel<-Transposed_Isoform_Expression[,c(which(colnames(Transposed_Isoform_Expression) == \"sample_id\"),\n",
    "                                                                            which(colnames(Transposed_Isoform_Expression) %in% order_transcript_id))]\n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          cat(\"Transposed_Isoform_Expression_sel_1\\n\")\n",
    "          cat(str(Transposed_Isoform_Expression_sel))\n",
    "          cat(\"\\n\")\n",
    "        }\n",
    "        \n",
    "        Transposed_Isoform_Expression_sel.m<-melt(Transposed_Isoform_Expression_sel, id.vars=c(\"sample_id\"),value.name = \"log2TPM\",variable.name = \"transcript_id\")\n",
    "        Transposed_Isoform_Expression_sel.m$TPM<--1+2^(Transposed_Isoform_Expression_sel.m$log2TPM)\n",
    "        \n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          cat(\"Transposed_Isoform_Expression_sel.m_0\\n\")\n",
    "          cat(str(Transposed_Isoform_Expression_sel.m))\n",
    "          cat(\"\\n\")\n",
    "          \n",
    "          cat(\"sample_id\\n\")\n",
    "          cat(str(unique(Transposed_Isoform_Expression_sel.m$sample_id)))\n",
    "          cat(\"\\n\")\n",
    "          \n",
    "          cat(\"transcript_id\\n\")\n",
    "          cat(str(unique(Transposed_Isoform_Expression_sel.m$transcript_id)))\n",
    "          cat(\"\\n\")\n",
    "          \n",
    "          # ########################################\n",
    "          # quit(status = 1)\n",
    "        }\n",
    "        \n",
    "        ### calculate summatory TPM per sample\n",
    "        \n",
    "        Transposed_Isoform_Expression_sel.m.dt<-data.table(Transposed_Isoform_Expression_sel.m,\n",
    "                                                           key=c(\"sample_id\"))\n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          cat(\"Transposed_Isoform_Expression_sel.m.dt_0\\n\")\n",
    "          cat(str(Transposed_Isoform_Expression_sel.m.dt))\n",
    "          cat(\"\\n\")\n",
    "        }\n",
    "        \n",
    "        \n",
    "        Summary_table_GENE_EXP<-as.data.frame(Transposed_Isoform_Expression_sel.m.dt[, .(sum_GENE_EXP=sum(TPM)),\n",
    "                                                                                     by=key(Transposed_Isoform_Expression_sel.m.dt)],stringsAsFactors=F)\n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          \n",
    "          cat(\"Summary_table_GENE_EXP_0\\n\")\n",
    "          cat(str(Summary_table_GENE_EXP))\n",
    "          cat(\"\\n\")\n",
    "          \n",
    "          # quit(status = 1)\n",
    "        }\n",
    "        \n",
    "        \n",
    "        Transposed_Isoform_Expression_sel.m<-merge(Transposed_Isoform_Expression_sel.m,\n",
    "                                                   Summary_table_GENE_EXP,\n",
    "                                                   by=\"sample_id\")\n",
    "        \n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          cat(\"Transposed_Isoform_Expression_sel.m_1\\n\")\n",
    "          cat(str(Transposed_Isoform_Expression_sel.m))\n",
    "          cat(\"\\n\")\n",
    "          \n",
    "        }\n",
    "        \n",
    "        Transposed_Isoform_Expression_sel.m.dt<-data.table(Transposed_Isoform_Expression_sel.m,\n",
    "                                                           key=c(\"sample_id\",\"transcript_id\"))\n",
    "        \n",
    "        Ratio_df<-as.data.frame(Transposed_Isoform_Expression_sel.m.dt[, .(log2TPM=log2TPM,\n",
    "                                                                           TPM=TPM,\n",
    "                                                                           sum_GENE_EXP=sum_GENE_EXP,\n",
    "                                                                           Ratio=TPM/sum_GENE_EXP),by=key(Transposed_Isoform_Expression_sel.m.dt)],stringsAsFactors=F)\n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          \n",
    "          cat(\"Ratio_df_0\\n\")\n",
    "          cat(str(Ratio_df))\n",
    "          cat(\"\\n\")\n",
    "          #\n",
    "          # cat(\"distrib_ratios\\n\")\n",
    "          # cat(sprintf(as.character(names(summary(Ratio_df$Ratio)))))\n",
    "          # cat(\"\\n\")\n",
    "          # cat(sprintf(as.character(summary(Ratio_df$Ratio))))\n",
    "          # cat(\"\\n\")\n",
    "          #\n",
    "          #\n",
    "          # quit(status = 1)\n",
    "        }\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        #### Merge with covariates matrix & keep HET ----\n",
    "        \n",
    "        \n",
    "        \n",
    "        Ratio_df<-merge(Ratio_df,\n",
    "                        INTERVAL_covariates_and_PEER_factors_sel,\n",
    "                        by=c(\"sample_id\"),\n",
    "                        all=T)\n",
    "        \n",
    "        \n",
    "        \n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          \n",
    "          cat(\"Ratio_df_3\\n\")\n",
    "          cat(str(Ratio_df))\n",
    "          cat(\"\\n\")\n",
    "          \n",
    "          # quit(status = 1)\n",
    "          \n",
    "        }\n",
    "        \n",
    "        Ratio_df_HET<-droplevels(Ratio_df[which(Ratio_df$Genotype != \"HOM\"),])\n",
    "        \n",
    "        Ratio_df_HET$transcript_id<-factor(Ratio_df_HET$transcript_id,\n",
    "                                           levels=order_transcript_id,\n",
    "                                           ordered=T)\n",
    "        \n",
    "        Ratio_df_HET$Cell_Type<-'Whole blood'\n",
    "        \n",
    "        \n",
    "        Ratio_df_HET$Genotype<-factor(Ratio_df_HET$Genotype,\n",
    "                                      levels=newG,\n",
    "                                      ordered=T)\n",
    "        \n",
    "        \n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          \n",
    "          cat(\"Ratio_df_HET_0\\n\")\n",
    "          cat(str(Ratio_df_HET))\n",
    "          cat(\"\\n\")\n",
    "          \n",
    "          \n",
    "          # quit(status = 1)\n",
    "          \n",
    "        }\n",
    "        \n",
    "        \n",
    "        \n",
    "        array_transcript_id<-unique(Ratio_df_HET$transcript_id)\n",
    "        \n",
    "        \n",
    "        if(DEBUG == 1)\n",
    "        {\n",
    "          cat(\"array_transcript_id_0\\n\")\n",
    "          cat(str(array_transcript_id))\n",
    "          cat(\"\\n\")\n",
    "        }\n",
    "        \n",
    "        for(iteration_array_transcript_id in 1:length(array_transcript_id)){\n",
    "          \n",
    "          array_transcript_id_sel<-array_transcript_id[iteration_array_transcript_id]\n",
    "          \n",
    "          if(DEBUG == 1)\n",
    "          {\n",
    "            cat(sprintf(as.character(iteration_array_transcript_id)))\n",
    "            cat(\"\\t\")\n",
    "            cat(sprintf(as.character(array_transcript_id_sel)))\n",
    "            cat(\"\\n\")\n",
    "          }\n",
    "          \n",
    "          Ratio_df_HET_transcript_id_sel<-droplevels(Ratio_df_HET[which(Ratio_df_HET$transcript_id == array_transcript_id_sel),])\n",
    "          \n",
    "          if(DEBUG == 1)\n",
    "          {\n",
    "            cat(\"Ratio_df_HET_transcript_id_sel_0\\n\")\n",
    "            cat(str(Ratio_df_HET_transcript_id_sel))\n",
    "            cat(\"\\n\")\n",
    "          }\n",
    "          \n",
    "          \n",
    "          A<-round(summary(Ratio_df_HET_transcript_id_sel$TPM[!is.na(Ratio_df_HET_transcript_id_sel$TPM)]),2)\n",
    "          \n",
    "          \n",
    "          # cat(\"summary_GENE_EXP\\n\")\n",
    "          # cat(sprintf(as.character(names(A))))\n",
    "          # cat(\"\\n\")\n",
    "          # cat(sprintf(as.character(A)))\n",
    "          # cat(\"\\n\")\n",
    "          \n",
    "          step<-abs(A[6]-A[1])/4\n",
    "          \n",
    "          if(step == 0)\n",
    "          {\n",
    "            \n",
    "            step<-1\n",
    "          }\n",
    "          \n",
    "          breaks.Rank<-sort(unique(c(A[6],seq(from= A[1], to=A[6],by=step))))\n",
    "          labels.Rank<-as.character(round(breaks.Rank,1))\n",
    "          \n",
    "          \n",
    "          if(DEBUG == 1)\n",
    "          {\n",
    "            cat(\"labels.Rank:\\t\")\n",
    "            cat(sprintf(as.character(labels.Rank)))\n",
    "            cat(\"\\n\")\n",
    "            \n",
    "            \n",
    "            \n",
    "            # quit(status = 1)\n",
    "          }\n",
    "          \n",
    "          \n",
    "          \n",
    "          \n",
    "          \n",
    "          \n",
    "          \n",
    "          WB_per_transcript_violin<-ggplot(data=Ratio_df_HET_transcript_id_sel,\n",
    "                                           aes(y=TPM, x=Genotype, fill=Genotype)) +\n",
    "            geom_violin()+\n",
    "            stat_summary(fun = median, fun.min = median, fun.max = median,\n",
    "                         geom = \"crossbar\", width = 0.5)+\n",
    "            scale_y_continuous(name=\"TPM\",breaks=breaks.Rank,labels=labels.Rank, limits=c(breaks.Rank[1],breaks.Rank[length(breaks.Rank)]))+\n",
    "            scale_fill_manual(values=c(\"#E7B800\",\"#00AFBB\",\"#FC4E07\"),drop=F)+\n",
    "            scale_x_discrete(name=NULL, drop=F)\n",
    "          \n",
    "          \n",
    "          \n",
    "          WB_per_transcript_violin<-WB_per_transcript_violin+\n",
    "            facet_grid(transcript_id ~ Cell_Type, scales='free_x', space='free_x', switch=\"y\", drop=F)+\n",
    "            theme_cowplot(font_size = 4)+\n",
    "            theme( strip.background = element_blank(),\n",
    "                   strip.placement = \"outside\",\n",
    "                   strip.text = element_text(size=6),\n",
    "                   panel.spacing = unit(0.2, \"lines\"),\n",
    "                   panel.background=element_rect(fill=\"white\"),\n",
    "                   panel.border=element_rect(colour=\"white\",size=0,5),\n",
    "                   panel.grid.major = element_blank(),\n",
    "                   panel.grid.minor = element_blank())+\n",
    "            theme_classic()+\n",
    "            theme(axis.title.y=element_text(size=8, color=\"black\", family=\"sans\"),\n",
    "                  axis.title.x=element_blank(),\n",
    "                  axis.text.y=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                  axis.text.x=element_blank(),\n",
    "                  axis.line.x = element_line(size = 0.2),\n",
    "                  axis.ticks.x = element_blank(),\n",
    "                  axis.ticks.y = element_line(size = 0.2),\n",
    "                  axis.line.y = element_line(size = 0.2))+\n",
    "            theme(legend.title = element_text(size=6),\n",
    "                  legend.text = element_text(size=6),\n",
    "                  legend.key.size = unit(0.25, 'cm'), #change legend key size\n",
    "                  legend.key.height = unit(0.25, 'cm'), #change legend key height\n",
    "                  legend.key.width = unit(0.25, 'cm'), #change legend key width\n",
    "                  legend.position=\"hidden\")+\n",
    "            ggeasy::easy_center_title()\n",
    "          \n",
    "          \n",
    "          setwd(path_graphs_rs_HGNC)\n",
    "          \n",
    "          svgname<-paste(paste('WB_per_transcript_violin',array_transcript_id_sel, sep='_'),\".svg\",sep='')\n",
    "          makesvg = TRUE\n",
    "          \n",
    "          if (makesvg == TRUE)\n",
    "          {\n",
    "            ggsave(svgname, plot= WB_per_transcript_violin,\n",
    "                   device=\"svg\",\n",
    "                   height=2, width=2)\n",
    "          }\n",
    "\n",
    "            if(rs_sel == 'rs150649461'){\n",
    "\n",
    "                setwd(path_figures)\n",
    "          \n",
    "                  svgname<-paste(paste('Figure_S4_panel_C',array_transcript_id_sel, sep='_'),\".svg\",sep='')\n",
    "                  makesvg = TRUE\n",
    "                  \n",
    "                  if (makesvg == TRUE)\n",
    "                  {\n",
    "                    ggsave(svgname, plot= WB_per_transcript_violin,\n",
    "                           device=\"svg\",\n",
    "                           height=2, width=2)\n",
    "                  }\n",
    "\n",
    "            }else{\n",
    "\n",
    "                if(rs_sel == 'rs150221884'){\n",
    "\n",
    "                    setwd(path_figures)\n",
    "          \n",
    "                      svgname<-paste(paste('Figure_S4_panel_D',array_transcript_id_sel, sep='_'),\".svg\",sep='')\n",
    "                      makesvg = TRUE\n",
    "                      \n",
    "                      if (makesvg == TRUE)\n",
    "                      {\n",
    "                        ggsave(svgname, plot= WB_per_transcript_violin,\n",
    "                               device=\"svg\",\n",
    "                               height=2, width=2)\n",
    "                      }\n",
    "\n",
    "                }else{\n",
    "\n",
    "                    if(rs_sel == 'rs569022307'){\n",
    "\n",
    "                        setwd(path_figures)\n",
    "          \n",
    "                          svgname<-paste(paste('Figure_S4_panel_B',array_transcript_id_sel, sep='_'),\".svg\",sep='')\n",
    "                          makesvg = TRUE\n",
    "                          \n",
    "                          if (makesvg == TRUE)\n",
    "                          {\n",
    "                            ggsave(svgname, plot= WB_per_transcript_violin,\n",
    "                                   device=\"svg\",\n",
    "                                   height=2, width=2)\n",
    "                          }\n",
    "\n",
    "                    }else{\n",
    "\n",
    "                        if(rs_sel == 'rs191451841'){\n",
    "\n",
    "                            setwd(path_figures)\n",
    "          \n",
    "                              svgname<-paste(paste('NEW_possible',rs_sel, HGNC_sel, array_transcript_id_sel, sep='_'),\".svg\",sep='')\n",
    "                              makesvg = TRUE\n",
    "                              \n",
    "                              if (makesvg == TRUE)\n",
    "                              {\n",
    "                                ggsave(svgname, plot= WB_per_transcript_violin,\n",
    "                                       device=\"svg\",\n",
    "                                       height=2, width=2)\n",
    "                              }\n",
    "\n",
    "                        }else{\n",
    "\n",
    "                            if(rs_sel == 'rs112097551'){\n",
    "\n",
    "                                setwd(path_figures)\n",
    "              \n",
    "                                  svgname<-paste(paste('NEW_possible',rs_sel, HGNC_sel, array_transcript_id_sel, sep='_'),\".svg\",sep='')\n",
    "                                  makesvg = TRUE\n",
    "                                  \n",
    "                                  if (makesvg == TRUE)\n",
    "                                  {\n",
    "                                    ggsave(svgname, plot= WB_per_transcript_violin,\n",
    "                                           device=\"svg\",\n",
    "                                           height=2, width=2)\n",
    "                                  }\n",
    "                            }else{\n",
    "\n",
    "                                 if(rs_sel == 'rs149489081'){\n",
    "\n",
    "                                    setwd(path_figures)\n",
    "                  \n",
    "                                      svgname<-paste(paste('NEW_possible',rs_sel, HGNC_sel, array_transcript_id_sel, sep='_'),\".svg\",sep='')\n",
    "                                      makesvg = TRUE\n",
    "                                      \n",
    "                                      if (makesvg == TRUE)\n",
    "                                      {\n",
    "                                        ggsave(svgname, plot= WB_per_transcript_violin,\n",
    "                                               device=\"svg\",\n",
    "                                               height=2, width=2)\n",
    "                                      }\n",
    "                                }else{\n",
    "                                     \n",
    "                                }# rs_sel == 'rs149489081'                               \n",
    "                            }#rs_sel == 'rs112097551'                            \n",
    "                        }#rs_sel == 'rs191451841'                       \n",
    "                    }# rs_sel == 'rs569022307'                    \n",
    "                }#rs_sel == 'rs150221884'              \n",
    "            }#rs_sel == 'rs150649461'\n",
    "          \n",
    "        }#iteration_array_transcript_id in 1:length(array_transcript_id\n",
    "        \n",
    "        \n",
    "        cat(\"Printed transcripts INTERVAL\\n\")\n",
    "        \n",
    "        \n",
    "      }else{\n",
    "        \n",
    "        Blood_violin<-NA\n",
    "        \n",
    "        cat(\"No results INTERVAL 2\\n\")\n",
    "        \n",
    "      }#file.exists(filename)\n",
    "    }else{\n",
    "      \n",
    "      Blood_violin<-NA\n",
    "      \n",
    "      cat(\"No results INTERVAL 1\\n\")\n",
    "      \n",
    "      \n",
    "    }#file.exists(INTERVAL_files) \n",
    "    \n",
    "    \n",
    "  }# k in 1:length(ENSG_array\n",
    "\n",
    "}#i in START:length(VARS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f06090a-62b9-4647-908d-4eaf35705a19",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e8601f34-e871-4fdd-8383-155d26242a69",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94761e1a-4d65-4522-b1b5-5d96683553d2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "beb0e649-8bdd-4ae2-80f7-4dc910ab6474",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "general_purpose",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
