{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5ba6e839-d30f-48b7-bade-2bebc44d281d",
   "metadata": {},
   "source": [
    "# Load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "898e4e86-7d07-4ed9-8353-945efc378c08",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T14:39:34.333853Z",
     "iopub.status.busy": "2025-07-05T14:39:34.332218Z",
     "iopub.status.idle": "2025-07-05T14:40:00.736275Z",
     "shell.execute_reply": "2025-07-05T14:40:00.735019Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/home/manuel.tardaguila/conda_envs/general_purpose/lib/R/library'"
      ],
      "text/latex": [
       "'/home/manuel.tardaguila/conda\\_envs/general\\_purpose/lib/R/library'"
      ],
      "text/markdown": [
       "'/home/manuel.tardaguila/conda_envs/general_purpose/lib/R/library'"
      ],
      "text/plain": [
       "[1] \"/home/manuel.tardaguila/conda_envs/general_purpose/lib/R/library\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/latex": [
       "'/home/manuel.tardaguila/R/x86\\_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/markdown": [
       "'/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/'"
      ],
      "text/plain": [
       "[1] \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    ".libPaths()\n",
    "\n",
    "assign(\".lib.loc\", \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\", envir = environment(.libPaths))\n",
    ".libPaths()\n",
    "\n",
    "suppressMessages(library(\"plyr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"data.table\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"withr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggplot2\", lib.loc = \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"farver\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"labeling\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"optparse\", lib.loc = \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"dplyr\", lib.loc = \"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"withr\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"backports\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"broom\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"rstudioapi\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"tzdb\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"svglite\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggeasy\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"tidyverse\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"BiocGenerics\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"S4Vectors\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"IRanges\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"GenomeInfoDb\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"GenomicRanges\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"Biobase\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"AnnotationDbi\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"GO.db\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"org.Hs.eg.db\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggrepel\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"RColorBrewer\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"svglite\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"cowplot\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"Matrix\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"rtracklayer\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"Biostrings\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggrepel\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggarchery\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"ggnewscale\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"splitstackshape\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n",
    "suppressMessages(library(\"viridis\", lib.loc=\"/home/manuel.tardaguila/R/x86_64-pc-linux-gnu-library/4.1/\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8af0ed16-9a2b-441f-91b6-553a182e9163",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T14:40:00.762020Z",
     "iopub.status.busy": "2025-07-05T14:40:00.738105Z",
     "iopub.status.idle": "2025-07-05T14:40:00.770971Z",
     "shell.execute_reply": "2025-07-05T14:40:00.769830Z"
    }
   },
   "outputs": [],
   "source": [
    "DEBUG<-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "10a98b7d-21f2-4f70-a3ee-0b4bbc59090b",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T14:40:00.773794Z",
     "iopub.status.busy": "2025-07-05T14:40:00.772790Z",
     "iopub.status.idle": "2025-07-05T14:40:00.781107Z",
     "shell.execute_reply": "2025-07-05T14:40:00.780001Z"
    }
   },
   "outputs": [],
   "source": [
    "build_files_from_pipelines_dir<-'/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/test/Build_files/'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6c920f4-552e-48fb-87f1-2981231f94c4",
   "metadata": {},
   "source": [
    "# Read input files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "96bfc149-eadb-4df5-9596-053d1586d4dd",
   "metadata": {},
   "source": [
    "## haplotypes file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3c36d8b2-5f46-44d8-8a07-0192ce591cb7",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T14:40:00.784031Z",
     "iopub.status.busy": "2025-07-05T14:40:00.782960Z",
     "iopub.status.idle": "2025-07-05T14:40:03.403548Z",
     "shell.execute_reply": "2025-07-05T14:40:03.402394Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rescue_haplotypes_0\n",
      "'data.frame':\t165051 obs. of  26 variables:\n",
      " $ VAR                           : chr  \"chr9_135920196_C_T\" \"chr9_135920196_C_T\" \"chr9_135920196_C_T\" \"chr9_135920196_C_T\" ...\n",
      " $ Proxy_VAR                     : chr  \"chr9_135864513_C_T\" \"chr9_135864513_C_T\" \"chr9_135864513_C_T\" \"chr9_135864513_C_T\" ...\n",
      " $ comparison                    : chr  \"HOM_REF__HET|HOM_REF\" \"HOM_REF__HET|HOM_REF\" \"HOM_REF__HET|HOM_REF\" \"HOM_REF__HET|HOM_REF\" ...\n",
      " $ ensembl_gene_id               : chr  \"ENSG00000000003\" \"ENSG00000000419\" \"ENSG00000000457\" \"ENSG00000000460\" ...\n",
      " $ HGNC                          : chr  \"TSPAN6\" \"DPM1\" \"SCYL3\" \"C1orf112\" ...\n",
      " $ CIS_gene_minuslogpvalue       : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ CIS_Beta                      : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ n_breakdown_string            : chr  NA NA NA NA ...\n",
      " $ n_SIG_CIS                     : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ n_tested_CIS                  : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Perc_SIG_CIS                  : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Block_PCHiC_minuslogpvalue    : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Block_PCHiC_Beta              : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Block_PCHiC_n_breakdown_string: chr  NA NA NA NA ...\n",
      " $ n_SIG_Block                   : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ n_tested_Block                : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Perc_SIG_Block                : int  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Genome_wide_minuslogpvalue    : num  0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ Genome_wide_Beta              : num  -0.1023 0.1286 -0.0172 -0.166 -0.1872 ...\n",
      " $ Genome_wide_n_breakdown_string: chr  \"HOM_REF__2861;HET|HOM_REF__13\" \"HOM_REF__2861;HET|HOM_REF__13\" \"HOM_REF__2861;HET|HOM_REF__13\" \"HOM_REF__2861;HET|HOM_REF__13\" ...\n",
      " $ n_SIG_GW                      : int  17 17 17 17 17 17 17 17 17 17 ...\n",
      " $ n_tested_GW                   : int  18339 18339 18339 18339 18339 18339 18339 18339 18339 18339 ...\n",
      " $ Perc_SIG_GW                   : num  0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1 ...\n",
      " $ CIS_Beta_Z_score              : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Block_PCHiC_Beta_Z_score      : num  NA NA NA NA NA NA NA NA NA NA ...\n",
      " $ Genome_wide_Beta_Z_score      : num  -0.000524 0.000388 -0.000188 -0.000776 -0.000859 ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#### Read rescue_haplotypes----\n",
    "  \n",
    "rescue_haplotypes<-as.data.frame(fread(file=\"/group/soranzo/manuel.tardaguila/INTERVAL_results/chr9_135920196_C_T_ALL_BY_ALL_DE_LM_FPKM_results_Haplotypes.tsv\", sep=\"\\t\", header=T), stringsAsFactors=F)\n",
    "\n",
    "\n",
    "\n",
    "cat(\"rescue_haplotypes_0\\n\")\n",
    "cat(str(rescue_haplotypes))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18476377-f46d-46d0-9076-b5beef15bd15",
   "metadata": {},
   "source": [
    "## dataframe for sit plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "01f95fe1-050d-4f34-be2b-fd6fc0ab6ba2",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T14:40:03.406212Z",
     "iopub.status.busy": "2025-07-05T14:40:03.405282Z",
     "iopub.status.idle": "2025-07-05T14:40:03.451429Z",
     "shell.execute_reply": "2025-07-05T14:40:03.450361Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_to_print_SIT_plots_0\n",
      "'data.frame':\t23 obs. of  15 variables:\n",
      " $ VAR        : chr  \"chr22_28761148_C_T\" \"chr22_28761148_C_T\" \"chr1_92925654_G_C\" \"chr1_198680015_G_A\" ...\n",
      " $ rs         : chr  \"rs62237617\" \"rs62237617\" \"rs150649461\" \"rs12733073\" ...\n",
      " $ AS_ID      : chr  \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" ...\n",
      " $ Genes      : chr  \"ENSG00000183765;ENSG00000100154;ENSG00000159873;ENSG00000100209\" \"ENSG00000183765;ENSG00000100154;ENSG00000159873;ENSG00000100209\" \"ENSG00000162676;ENSG00000067208\" \"ENSG00000081237\" ...\n",
      " $ ATAC_cells : chr  \"CD4;CD8;NK\" \"Mega\" \"CMP;GMP_A\" NA ...\n",
      " $ oeID       : chr  \"432044\" \"432044\" NA \"49883\" ...\n",
      " $ PCHiC_cells: chr  \"tCD4;tCD8\" \"MK\" NA \"tCD4\" ...\n",
      " $ Height     : num  7 7 6 3 6 3 2 3.5 4 6 ...\n",
      " $ Width      : int  4 4 4 4 4 4 4 4 4 4 ...\n",
      " $ weight_A   : num  0.1 0.1 0.3 0.1 0.3 0.1 0.1 0.1 0.1 0.3 ...\n",
      " $ weight_B   : num  0.4 0.4 0.4 0.2 0.4 0.2 0.2 0.18 0.1 0.4 ...\n",
      " $ weight_C   : num  0.8 0.8 0.6 0 0.6 0 0 0 0 0.6 ...\n",
      " $ weight_D   : num  0.5 0.5 0.3 0.2 0.4 0.2 0 0.2 0.2 0.4 ...\n",
      " $ RNA_source : chr  \"Whole blood\" \"Whole blood\" \"Whole blood\" \"Whole blood\" ...\n",
      " $ beta_span  : chr  \"-0.25;-0.1;0;0.1;0.25\" \"-0.25;-0.1;0;0.1;0.25\" \"-0.5;-0.4;-0.3;-0.2;-0.1;0;0.1;0.2;0.4;0.5;0.3\" \"-0.1;-0.2;0;0.1;0.2\" ...\n",
      "\n",
      " chr [1:22] \"lymph__304\" \"pct__350\" \"mono__27\" \"lymph__38\" \"pct__44\" ...\n",
      "\n",
      "Monocytes naive T-CD4 Cells Whole blood\n",
      "1 2 20\n"
     ]
    }
   ],
   "source": [
    " dataframe_to_print_SIT_plots<-as.data.frame(fread(file=\"/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/dataframe_to_print_SIT_plots_3.txt\", sep=\"\\t\", header=T), stringsAsFactors=F)\n",
    "  \n",
    "colnames(dataframe_to_print_SIT_plots)[which(colnames(dataframe_to_print_SIT_plots) == 'AS')]<-'AS_ID'\n",
    "\n",
    "dataframe_to_print_SIT_plots$beta_span<-gsub(\"\\\"\",\"\",dataframe_to_print_SIT_plots$beta_span)\n",
    "dataframe_to_print_SIT_plots$beta_span<-gsub(\"\\'\",\"\",dataframe_to_print_SIT_plots$beta_span)\n",
    "\n",
    "\n",
    "cat(\"dataframe_to_print_SIT_plots_0\\n\")\n",
    "cat(str(dataframe_to_print_SIT_plots))\n",
    "cat(\"\\n\")\n",
    "cat(str(unique(dataframe_to_print_SIT_plots$AS_ID)))\n",
    "cat(\"\\n\")\n",
    "cat(sprintf(as.character(names(summary(as.factor(dataframe_to_print_SIT_plots$RNA_source))))))\n",
    "cat(\"\\n\")\n",
    "cat(sprintf(as.character(summary(as.factor(dataframe_to_print_SIT_plots$RNA_source)))))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "07a61c27-2911-47ae-aa9e-dd6837ac572e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T14:40:03.454065Z",
     "iopub.status.busy": "2025-07-05T14:40:03.453137Z",
     "iopub.status.idle": "2025-07-05T14:40:03.469388Z",
     "shell.execute_reply": "2025-07-05T14:40:03.468276Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dataframe_subset_0\n",
      "'data.frame':\t20 obs. of  2 variables:\n",
      " $ VAR: chr  \"chr22_28761148_C_T\" \"chr1_92925654_G_C\" \"chr1_198680015_G_A\" \"chr1_202129205_G_A\" ...\n",
      " $ rs : chr  \"rs62237617\" \"rs150649461\" \"rs12733073\" \"rs147176673\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "dataframe_subset<-unique(dataframe_to_print_SIT_plots[which(colnames(dataframe_to_print_SIT_plots)%in%c('VAR','rs'))])\n",
    "\n",
    "cat(\"dataframe_subset_0\\n\")\n",
    "cat(str(dataframe_subset))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a9a5306-8d7f-475d-8669-5b520a6fffec",
   "metadata": {},
   "source": [
    "# MPRA results prior to meta-analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "124942f0-1eca-4537-9de8-4665c580b8a9",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T16:43:50.293146Z",
     "iopub.status.busy": "2025-07-05T16:43:50.292158Z",
     "iopub.status.idle": "2025-07-05T16:43:50.411600Z",
     "shell.execute_reply": "2025-07-05T16:43:50.410515Z"
    }
   },
   "outputs": [],
   "source": [
    "MPRA_tiles<-read.table(file=\"/group/soranzo/manuel.tardaguila/MPRA_bc_synthesis/test/Analysis_filtered_Plus_Pseudocount/final_files/MPRA_S_no_global_metanalysis/Table_S3_MPRA_result_per_tile_and_cell_type.tsv\",\n",
    "                      sep=\"\\t\",\n",
    "                      header=T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "15690bdf-aaf5-47f7-ba99-ac77f78413bc",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T16:43:50.512483Z",
     "iopub.status.busy": "2025-07-05T16:43:50.511550Z",
     "iopub.status.idle": "2025-07-05T16:43:50.532152Z",
     "shell.execute_reply": "2025-07-05T16:43:50.531145Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MPRA_tiles_0\n",
      "\n",
      "K562 HL60 CHRF THP1\n",
      "TILE_1 TILE_2 TILE_3 TILE_4 TILE_5\n"
     ]
    }
   ],
   "source": [
    "cat(\"MPRA_tiles_0\\n\")\n",
    "#cat(str(MPRA_tiles))\n",
    "cat(\"\\n\")\n",
    "cat(sprintf(as.character(unique(MPRA_tiles$Cell_Type))))\n",
    "cat(\"\\n\")\n",
    "cat(sprintf(as.character(unique(MPRA_tiles$window))))\n",
    "cat(\"\\n\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "3571dd5b-d5ae-4886-ad8a-235e35890925",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T16:43:50.961563Z",
     "iopub.status.busy": "2025-07-05T16:43:50.960562Z",
     "iopub.status.idle": "2025-07-05T16:43:50.971042Z",
     "shell.execute_reply": "2025-07-05T16:43:50.970028Z"
    }
   },
   "outputs": [],
   "source": [
    "MPRA_tiles$Cell_Type<-factor(MPRA_tiles$Cell_Type,\n",
    "                            levels=c(\"K562\",\"CHRF\",\"HL60\",\"THP1\"),\n",
    "                            ordered=T)\n",
    "\n",
    "MPRA_tiles$window<-factor(MPRA_tiles$window,\n",
    "                            levels=c(\"TILE_1\",\"TILE_2\",\"TILE_3\",\"TILE_4\",\"TILE_5\"),\n",
    "                            ordered=T)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "e222dbe7-f388-4622-b128-b2f036f032c4",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T16:43:51.413270Z",
     "iopub.status.busy": "2025-07-05T16:43:51.412256Z",
     "iopub.status.idle": "2025-07-05T16:43:51.446276Z",
     "shell.execute_reply": "2025-07-05T16:43:51.445273Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "check_0\n",
      "'data.frame':\t80 obs. of  26 variables:\n",
      " $ SNP         : chr  \"chr1_202129196_T_C__chr1_202129205_G_A\" \"chr1_202129196_T_C__chr1_202129205_G_A\" \"chr1_202129196_T_C__chr1_202129205_G_A\" \"chr1_202129196_T_C__chr1_202129205_G_A\" ...\n",
      " $ comparison  : chr  \"ref_ref_vs_alt_ref\" \"ref_ref_vs_alt_ref\" \"ref_ref_vs_alt_ref\" \"ref_ref_vs_alt_ref\" ...\n",
      " $ window      : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 1 1 1 1 2 2 2 2 3 3 ...\n",
      " $ VAR         : chr  \"chr1_202129196_T_C\" \"chr1_202129196_T_C\" \"chr1_202129196_T_C\" \"chr1_202129196_T_C\" ...\n",
      " $ chr         : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\" ...\n",
      " $ start       : int  202129071 202129071 202129071 202129071 202129106 202129106 202129106 202129106 202129132 202129132 ...\n",
      " $ end         : int  202129218 202129218 202129218 202129218 202129253 202129253 202129253 202129253 202129279 202129279 ...\n",
      " $ ID          : chr  \"Element_12__TILE_1__REF\" \"Element_12__TILE_1__REF\" \"Element_12__TILE_1__REF\" \"Element_12__TILE_1__REF\" ...\n",
      " $ Cell_Type   : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 3 1 4 2 2 3 4 1 3 1 ...\n",
      " $ Project     : chr  \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" ...\n",
      " $ A_Ctrl_Mean : num  608 1759 322 733 2380 ...\n",
      " $ A_Exp_Mean  : num  578 1544 271 760 3590 ...\n",
      " $ A_log2FC    : num  -0.0788 -0.1916 -0.2127 0.0708 0.6002 ...\n",
      " $ A_log2FC_SE : num  0.0952 0.052 0.1631 0.1463 0.203 ...\n",
      " $ A_logPadj_BF: num  0 0.871 0 0 0 ...\n",
      " $ B_Ctrl_Mean : num  414 1148 214 510 3429 ...\n",
      " $ B_Exp_Mean  : num  459 1204 225 652 4940 ...\n",
      " $ B_log2FC    : num  0.1289 0.0548 0.065 0.354 0.5313 ...\n",
      " $ B_log2FC_SE : num  0.1128 0.0434 0.1736 0.1533 0.2151 ...\n",
      " $ B_logPadj_BF: num  0 0 0 0 0 ...\n",
      " $ log2Skew    : num  0.1965 0.246 0.2637 0.2792 -0.0749 ...\n",
      " $ Skew_SE     : num  0.119 0.0838 0.2384 0.2143 0.2799 ...\n",
      " $ Skew_logPadj: num  0.000937 1.415354 0.000105 0.056643 0.003627 ...\n",
      " $ log2FC      : num  0.129 -0.192 -0.213 0.354 0.6 ...\n",
      " $ log2FC_SE   : num  0.113 0.052 0.163 0.153 0.203 ...\n",
      " $ logPadj_BF  : num  0 0.871 0 0 0 ...\n",
      "\n",
      "chr1_202129196_T_C chr1_202129205_G_A chr1_202129196_T_C__chr1_202129205_G_A chr9_135920196_C_T\n"
     ]
    }
   ],
   "source": [
    "check<-MPRA_tiles[grep(\"Element_12\",MPRA_tiles$ID),]\n",
    "\n",
    "\n",
    "cat(\"check_0\\n\")\n",
    "cat(str(check))\n",
    "cat(\"\\n\")\n",
    "cat(sprintf(as.character(unique(check$VAR))))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33a4bbfb-16db-41df-9ea2-40322c2a7474",
   "metadata": {},
   "source": [
    "# MPRA results post to meta-analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "7df90299-1ae9-4607-8b9b-edde122aed5e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T16:44:12.297093Z",
     "iopub.status.busy": "2025-07-05T16:44:12.296044Z",
     "iopub.status.idle": "2025-07-05T16:44:12.328576Z",
     "shell.execute_reply": "2025-07-05T16:44:12.327369Z"
    }
   },
   "outputs": [],
   "source": [
    "MPRA_post_meta_analysis<-read.table(file=\"/group/soranzo/manuel.tardaguila/MPRA_bc_synthesis/test/Analysis_filtered_Plus_Pseudocount/final_files/MPRA_S_no_global_metanalysis/Table_S4_MPRA_results_meta_analysis_collapsed_by_SNP_and_Cell_Type_Threshold_log2FC_0.255_Threshold_FC_meta_padj_0.01_VAR_added.tsv\",\n",
    "                      sep=\"\\t\",\n",
    "                      header=T)\n",
    "\n",
    "MPRA_post_meta_analysis$Cell_Type<-factor(MPRA_post_meta_analysis$Cell_Type,\n",
    "                            levels=c(\"K562\",\"CHRF\",\"HL60\",\"THP1\"),\n",
    "                            ordered=T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "7592f293-a9df-45b1-af10-1282b7d32ce3",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T16:44:12.533104Z",
     "iopub.status.busy": "2025-07-05T16:44:12.532094Z",
     "iopub.status.idle": "2025-07-05T16:44:12.561940Z",
     "shell.execute_reply": "2025-07-05T16:44:12.560899Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MPRA_post_meta_analysis\n",
      "'data.frame':\t2536 obs. of  18 variables:\n",
      " $ SNP                : chr  \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" ...\n",
      " $ comparison         : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" ...\n",
      " $ Cell_Type          : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 4 4 4 4 4 3 3 3 3 3 ...\n",
      " $ Project            : chr  \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" ...\n",
      " $ log2FC_A_meta      : num  -0.232 -0.232 -0.232 -0.232 -0.232 ...\n",
      " $ log2FC_meta        : num  -0.251 -0.251 -0.251 -0.251 -0.251 ...\n",
      " $ log2FC_meta_SE     : num  0.0465 0.0465 0.0465 0.0465 0.0465 ...\n",
      " $ log2Skew_meta      : num  -0.0133 -0.0133 -0.0133 -0.0133 -0.0133 ...\n",
      " $ log2Skew_meta_SE   : num  0.0477 0.0477 0.0477 0.0477 0.0477 ...\n",
      " $ log2FC_meta_nlogp  : num  17.2 17.2 17.2 17.2 17.2 ...\n",
      " $ log2Skew_meta_nlogp: num  0.942 0.942 0.942 0.942 0.942 ...\n",
      " $ lof2FC_meta_padj   : num  14.5 14.5 14.5 14.5 14.5 ...\n",
      " $ lof2Skew_meta_padj : num  0.816 0.816 0.816 0.816 0.816 ...\n",
      " $ active_meta        : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ emVar_meta         : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ Activity           : chr  \"INACTIVE\" \"INACTIVE\" \"INACTIVE\" \"INACTIVE\" ...\n",
      " $ MPRA_CLASS         : chr  \"MPRA_negative\" \"MPRA_negative\" \"MPRA_negative\" \"MPRA_negative\" ...\n",
      " $ VAR                : chr  \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" \"chr1_101744646_C_G\" ...\n",
      "\n",
      "THP1 HL60 CHRF K562\n"
     ]
    }
   ],
   "source": [
    "cat(\"MPRA_post_meta_analysis\\n\")\n",
    "cat(str(MPRA_post_meta_analysis))\n",
    "cat(\"\\n\")\n",
    "cat(sprintf(as.character(unique(MPRA_post_meta_analysis$Cell_Type))))\n",
    "cat(\"\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "607d9065-d45a-498c-9536-ba245167e844",
   "metadata": {},
   "source": [
    "# General path graphs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "a3781acc-b3cd-4c77-9c00-3d76eb65895a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T16:44:13.253883Z",
     "iopub.status.busy": "2025-07-05T16:44:13.252862Z",
     "iopub.status.idle": "2025-07-05T16:44:13.264964Z",
     "shell.execute_reply": "2025-07-05T16:44:13.263951Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "path_graphs<-paste('/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/test/','Graphs_selected_MPRA','/',sep='')\n",
    "  \n",
    "  if (file.exists(path_graphs)){\n",
    "    \n",
    "    \n",
    "  }else{\n",
    "    \n",
    "    dir.create(file.path(path_graphs))\n",
    "    \n",
    "  }#path_graphs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1d2b1ff-8020-4598-af79-b0b17fd649ee",
   "metadata": {},
   "source": [
    "# LOOP to print graphs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "131a952c-fb4a-4efa-a555-650dcbfbbb94",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T17:43:10.136006Z",
     "iopub.status.busy": "2025-07-05T17:43:10.134936Z",
     "iopub.status.idle": "2025-07-05T17:43:10.152654Z",
     "shell.execute_reply": "2025-07-05T17:43:10.151591Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VARS_0\n",
      " chr [1:20] \"chr22_28761148_C_T\" \"chr1_92925654_G_C\" \"chr1_198680015_G_A\" ...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "VARS<-unique(dataframe_subset$VAR)\n",
    "\n",
    "\n",
    "\n",
    "cat(\"VARS_0\\n\")\n",
    "cat(str(VARS))\n",
    "cat(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "6c097343-cc7c-4eed-ba68-288fa877db9d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T17:45:35.715030Z",
     "iopub.status.busy": "2025-07-05T17:45:35.713972Z",
     "iopub.status.idle": "2025-07-05T17:45:35.722552Z",
     "shell.execute_reply": "2025-07-05T17:45:35.721531Z"
    }
   },
   "outputs": [],
   "source": [
    "START<-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "c51d6328-8f1a-4cf1-82d1-1b2c1d5903b7",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T17:45:36.397037Z",
     "iopub.status.busy": "2025-07-05T17:45:36.396073Z",
     "iopub.status.idle": "2025-07-05T17:45:36.406223Z",
     "shell.execute_reply": "2025-07-05T17:45:36.405145Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "1"
      ],
      "text/latex": [
       "1"
      ],
      "text/markdown": [
       "1"
      ],
      "text/plain": [
       "[1] 1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#START<-which(VARS == 'chr7_101499930_G_A')\n",
    "#START<-which(VARS == 'chr9_135864513_C_T')\n",
    "#START<-which(VARS == 'chr3_184091102_T_G')\n",
    "#START<-which(VARS == 'chr3_46354444_C_T')\n",
    "\n",
    "#START<-which(VARS == 'chr1_198680015_G_A')\n",
    "\n",
    "\n",
    "\n",
    "START"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "e0b75116-d26a-4577-a811-5f210709206a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-07-05T17:44:56.688898Z",
     "iopub.status.busy": "2025-07-05T17:44:56.687873Z",
     "iopub.status.idle": "2025-07-05T17:44:58.941344Z",
     "shell.execute_reply": "2025-07-05T17:44:58.940103Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------------->\t3\trs12733073\tchr1_198680015_G_A\n",
      "dataframe_subset_sel_0\n",
      "'data.frame':\t1 obs. of  2 variables:\n",
      " $ VAR: chr \"chr1_198680015_G_A\"\n",
      " $ rs : chr \"rs12733073\"\n",
      "\n",
      "Element_selection_sub_0\n",
      " chr [1:5] \"Element_11__TILE_1__REF\" \"Element_11__TILE_2__REF\" ...\n",
      "\n",
      "MPRA_tiles_sub_0\n",
      "'data.frame':\t20 obs. of  26 variables:\n",
      " $ SNP         : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ comparison  : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" ...\n",
      " $ window      : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 1 1 1 1 2 2 2 2 3 3 ...\n",
      " $ VAR         : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ chr         : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\" ...\n",
      " $ start       : int  198679881 198679881 198679881 198679881 198679916 198679916 198679916 198679916 198679942 198679942 ...\n",
      " $ end         : int  198680028 198680028 198680028 198680028 198680063 198680063 198680063 198680063 198680089 198680089 ...\n",
      " $ ID          : chr  \"Element_11__TILE_1__REF\" \"Element_11__TILE_1__REF\" \"Element_11__TILE_1__REF\" \"Element_11__TILE_1__REF\" ...\n",
      " $ Cell_Type   : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 1 3 2 4 2 3 1 4 3 2 ...\n",
      " $ Project     : chr  \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" ...\n",
      " $ A_Ctrl_Mean : num  41194 18059 22158 11261 14308 ...\n",
      " $ A_Exp_Mean  : num  43755 17706 38031 11369 21170 ...\n",
      " $ A_log2FC    : num  0.0873 -0.0293 0.7793 0.014 0.5646 ...\n",
      " $ A_log2FC_SE : num  0.0573 0.1085 0.1459 0.2795 0.1465 ...\n",
      " $ A_logPadj_BF: num  0 0 4.27 0 1.17 ...\n",
      " $ B_Ctrl_Mean : num  34468 14957 17605 8985 10177 ...\n",
      " $ B_Exp_Mean  : num  38246 15031 30116 8902 17029 ...\n",
      " $ B_log2FC    : num  0.15028 0.00609 0.77462 -0.01325 0.74293 ...\n",
      " $ B_log2FC_SE : num  0.0581 0.1043 0.1489 0.274 0.1544 ...\n",
      " $ B_logPadj_BF: num  0 0 3.94 0 3.06 ...\n",
      " $ log2Skew    : num  0.06034 0.0341 -0.00847 -0.03898 0.17306 ...\n",
      " $ Skew_SE     : num  0.115 0.134 0.195 0.314 0.205 ...\n",
      " $ Skew_logPadj: num  0.060435 0.000937 0.000754 0.001159 0.000754 ...\n",
      " $ log2FC      : num  0.1503 -0.0293 0.7793 0.014 0.7429 ...\n",
      " $ log2FC_SE   : num  0.0581 0.1085 0.1459 0.2795 0.1544 ...\n",
      " $ logPadj_BF  : num  0 0 4.27 0 3.06 ...\n",
      "\n",
      "MPRA_VAR_array_0\n",
      " chr \"chr1_198680015_G_A\"\n",
      "\n",
      "--------MPRA_VAR_array_sel---------------->\t1\tchr1\t198679871\t198680159\tchr1_198680015_G_A\n",
      "FLAG_diplotype_0\n",
      "0\n",
      "pos_array_0\n",
      "chr1_198680015_G_A\n",
      "pos_array_1\n",
      "198680015_G_A\n",
      "pos_array_2\n",
      "198680015\n",
      "MPRA_tiles_sub_VAR_sel_0\n",
      "'data.frame':\t20 obs. of  26 variables:\n",
      " $ SNP         : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ comparison  : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" ...\n",
      " $ window      : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 1 1 1 1 2 2 2 2 3 3 ...\n",
      " $ VAR         : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ chr         : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\" ...\n",
      " $ start       : int  198679881 198679881 198679881 198679881 198679916 198679916 198679916 198679916 198679942 198679942 ...\n",
      " $ end         : int  198680028 198680028 198680028 198680028 198680063 198680063 198680063 198680063 198680089 198680089 ...\n",
      " $ ID          : chr  \"Element_11__TILE_1__REF\" \"Element_11__TILE_1__REF\" \"Element_11__TILE_1__REF\" \"Element_11__TILE_1__REF\" ...\n",
      " $ Cell_Type   : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 1 3 2 4 2 3 1 4 3 2 ...\n",
      " $ Project     : chr  \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" ...\n",
      " $ A_Ctrl_Mean : num  41194 18059 22158 11261 14308 ...\n",
      " $ A_Exp_Mean  : num  43755 17706 38031 11369 21170 ...\n",
      " $ A_log2FC    : num  0.0873 -0.0293 0.7793 0.014 0.5646 ...\n",
      " $ A_log2FC_SE : num  0.0573 0.1085 0.1459 0.2795 0.1465 ...\n",
      " $ A_logPadj_BF: num  0 0 4.27 0 1.17 ...\n",
      " $ B_Ctrl_Mean : num  34468 14957 17605 8985 10177 ...\n",
      " $ B_Exp_Mean  : num  38246 15031 30116 8902 17029 ...\n",
      " $ B_log2FC    : num  0.15028 0.00609 0.77462 -0.01325 0.74293 ...\n",
      " $ B_log2FC_SE : num  0.0581 0.1043 0.1489 0.274 0.1544 ...\n",
      " $ B_logPadj_BF: num  0 0 3.94 0 3.06 ...\n",
      " $ log2Skew    : num  0.06034 0.0341 -0.00847 -0.03898 0.17306 ...\n",
      " $ Skew_SE     : num  0.115 0.134 0.195 0.314 0.205 ...\n",
      " $ Skew_logPadj: num  0.060435 0.000937 0.000754 0.001159 0.000754 ...\n",
      " $ log2FC      : num  0.1503 -0.0293 0.7793 0.014 0.7429 ...\n",
      " $ log2FC_SE   : num  0.0581 0.1085 0.1459 0.2795 0.1544 ...\n",
      " $ logPadj_BF  : num  0 0 4.27 0 3.06 ...\n",
      "\n",
      "REP_0\n",
      "'data.frame':\t20 obs. of  13 variables:\n",
      " $ SNP         : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ comparison  : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" ...\n",
      " $ window      : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 1 1 1 1 2 2 2 2 3 3 ...\n",
      " $ VAR         : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ chr         : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\" ...\n",
      " $ start       : int  198679881 198679881 198679881 198679881 198679916 198679916 198679916 198679916 198679942 198679942 ...\n",
      " $ end         : int  198680028 198680028 198680028 198680028 198680063 198680063 198680063 198680063 198680089 198680089 ...\n",
      " $ Cell_Type   : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 1 3 2 4 2 3 1 4 3 2 ...\n",
      " $ log2Skew    : num  0.06034 0.0341 -0.00847 -0.03898 0.17306 ...\n",
      " $ Skew_logPadj: num  0.060435 0.000937 0.000754 0.001159 0.000754 ...\n",
      " $ log2FC      : num  0.1503 -0.0293 0.7793 0.014 0.7429 ...\n",
      " $ logPadj_BF  : num  0 0 4.27 0 3.06 ...\n",
      " $ analysis    : chr  \"MPRA model\" \"MPRA model\" \"MPRA model\" \"MPRA model\" ...\n",
      "\n",
      "REP_1\n",
      "'data.frame':\t20 obs. of  13 variables:\n",
      " $ SNP          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ comparison   : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" ...\n",
      " $ window       : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 1 1 1 1 2 2 2 2 3 3 ...\n",
      " $ VAR          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ chr          : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\" ...\n",
      " $ start        : int  198679881 198679881 198679881 198679881 198679916 198679916 198679916 198679916 198679942 198679942 ...\n",
      " $ end          : int  198680028 198680028 198680028 198680028 198680063 198680063 198680063 198680063 198680089 198680089 ...\n",
      " $ Cell_Type    : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 1 3 2 4 2 3 1 4 3 2 ...\n",
      " $ log2Skew     : num  0.06034 0.0341 -0.00847 -0.03898 0.17306 ...\n",
      " $ log2Skew_Padj: num  0.060435 0.000937 0.000754 0.001159 0.000754 ...\n",
      " $ log2FC       : num  0.1503 -0.0293 0.7793 0.014 0.7429 ...\n",
      " $ log2FC_Padj  : num  0 0 4.27 0 3.06 ...\n",
      " $ analysis     : chr  \"MPRA model\" \"MPRA model\" \"MPRA model\" \"MPRA model\" ...\n",
      "\n",
      "----------------------------------->MPRA_post_meta_analysis_sel_0\n",
      "'data.frame':\t4 obs. of  18 variables:\n",
      " $ SNP                : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\"\n",
      " $ comparison         : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\"\n",
      " $ Cell_Type          : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 4 3 1 2\n",
      " $ Project            : chr  \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\" \"MPRA_bc_synthesis\"\n",
      " $ log2FC_A_meta      : num  -0.0356 -0.1032 -0.0105 0.2646\n",
      " $ log2FC_meta        : num  -0.0566 -0.1308 -0.0022 0.2621\n",
      " $ log2FC_meta_SE     : num  0.0878 0.0515 0.0211 0.0463\n",
      " $ log2Skew_meta      : num  -0.0395 -0.035 -0.0508 -0.062\n",
      " $ log2Skew_meta_SE   : num  0.099 0.0637 0.0306 0.0693\n",
      " $ log2FC_meta_nlogp  : num  1.35 5.2 0.78 18.71\n",
      " $ log2Skew_meta_nlogp: num  1.06 1.23 3.03 1.68\n",
      " $ lof2FC_meta_padj   : num  0 2.47 0 15.97\n",
      " $ lof2Skew_meta_padj : num  0.9 1.01 2.37 1.3\n",
      " $ active_meta        : logi  FALSE FALSE FALSE TRUE\n",
      " $ emVar_meta         : logi  FALSE FALSE FALSE TRUE\n",
      " $ Activity           : chr  \"INACTIVE\" \"INACTIVE\" \"INACTIVE\" \"ACTIVE\"\n",
      " $ MPRA_CLASS         : chr  \"MPRA_negative\" \"MPRA_negative\" \"MPRA_negative\" \"MPRA_positive\"\n",
      " $ VAR                : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\"\n",
      "\n",
      "REP_meta_0\n",
      "'data.frame':\t4 obs. of  12 variables:\n",
      " $ SNP               : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\"\n",
      " $ comparison        : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\"\n",
      " $ Cell_Type         : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 4 3 1 2\n",
      " $ log2FC_meta       : num  -0.0566 -0.1308 -0.0022 0.2621\n",
      " $ log2Skew_meta     : num  -0.0395 -0.035 -0.0508 -0.062\n",
      " $ lof2FC_meta_padj  : num  0 2.47 0 15.97\n",
      " $ lof2Skew_meta_padj: num  0.9 1.01 2.37 1.3\n",
      " $ VAR               : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\"\n",
      " $ chr               : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\"\n",
      " $ start             : int  198679881 198679881 198679881 198679881\n",
      " $ end               : int  198680149 198680149 198680149 198680149\n",
      " $ window            : chr  \"TILE_3\" \"TILE_3\" \"TILE_3\" \"TILE_3\"\n",
      "\n",
      "REP_meta_1\n",
      "'data.frame':\t4 obs. of  13 variables:\n",
      " $ SNP          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\"\n",
      " $ comparison   : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\"\n",
      " $ Cell_Type    : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 4 3 1 2\n",
      " $ log2FC       : num  -0.0566 -0.1308 -0.0022 0.2621\n",
      " $ log2Skew     : num  -0.0395 -0.035 -0.0508 -0.062\n",
      " $ log2FC_Padj  : num  0 2.47 0 15.97\n",
      " $ log2Skew_Padj: num  0.9 1.01 2.37 1.3\n",
      " $ VAR          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\"\n",
      " $ chr          : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\"\n",
      " $ start        : int  198679881 198679881 198679881 198679881\n",
      " $ end          : int  198680149 198680149 198680149 198680149\n",
      " $ window       : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 3 3 3 3\n",
      " $ analysis     : chr  \"meta analysis\" \"meta analysis\" \"meta analysis\" \"meta analysis\"\n",
      "\n",
      "DEF_0\n",
      "'data.frame':\t24 obs. of  13 variables:\n",
      " $ SNP          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ comparison   : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" ...\n",
      " $ Cell_Type    : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 4 3 1 2 1 3 2 4 2 3 ...\n",
      " $ log2FC       : num  -0.0566 -0.1308 -0.0022 0.2621 0.1503 ...\n",
      " $ log2Skew     : num  -0.0395 -0.035 -0.0508 -0.062 0.0603 ...\n",
      " $ log2FC_Padj  : num  0 2.47 0 15.97 0 ...\n",
      " $ log2Skew_Padj: num  0.9 1.0065 2.3746 1.2994 0.0604 ...\n",
      " $ VAR          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ chr          : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\" ...\n",
      " $ start        : int  198679881 198679881 198679881 198679881 198679881 198679881 198679881 198679881 198679916 198679916 ...\n",
      " $ end          : int  198680149 198680149 198680149 198680149 198680028 198680028 198680028 198680028 198680063 198680063 ...\n",
      " $ window       : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 3 3 3 3 1 1 1 1 2 2 ...\n",
      " $ analysis     : chr  \"meta analysis\" \"meta analysis\" \"meta analysis\" \"meta analysis\" ...\n",
      "\n",
      "DEF_1\n",
      "'data.frame':\t24 obs. of  13 variables:\n",
      " $ SNP          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ comparison   : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" ...\n",
      " $ Cell_Type    : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 4 3 1 2 1 3 2 4 2 3 ...\n",
      " $ log2FC       : num  -0.0566 -0.1308 -0.0022 0.2621 0.1503 ...\n",
      " $ log2Skew     : num  -0.0395 -0.035 -0.0508 -0.062 0.0603 ...\n",
      " $ log2FC_Padj  : num  0 2.47 0 15.97 0 ...\n",
      " $ log2Skew_Padj: num  0.9 1.0065 2.3746 1.2994 0.0604 ...\n",
      " $ VAR          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ chr          : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\" ...\n",
      " $ start        : int  198679881 198679881 198679881 198679881 198679881 198679881 198679881 198679881 198679916 198679916 ...\n",
      " $ end          : int  198680149 198680149 198680149 198680149 198680028 198680028 198680028 198680028 198680063 198680063 ...\n",
      " $ window       : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 3 3 3 3 1 1 1 1 2 2 ...\n",
      " $ analysis     : Ord.factor w/ 2 levels \"MPRA model\"<\"meta analysis\": 2 2 2 2 1 1 1 1 1 1 ...\n",
      "\n",
      "max_log2FC_0\n",
      " num 0.78\n",
      "\n",
      "min_log2FC_0\n",
      " num -0.59\n",
      "\n",
      "abs_max_log2FC_0\n",
      " num 0.78\n",
      "\n",
      "abs_min_log2FC_0\n",
      " num 0.59\n",
      "\n",
      "labels.Log2FC:\t-0.78 0 0.78\n",
      "DEF_2\n",
      "'data.frame':\t24 obs. of  16 variables:\n",
      " $ SNP          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ comparison   : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" ...\n",
      " $ Cell_Type    : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 4 3 1 2 1 3 2 4 2 3 ...\n",
      " $ log2FC       : num  -0.06 -0.13 0 0.26 0.15 -0.03 0.78 0.01 0.74 -0.09 ...\n",
      " $ log2Skew     : num  -0.0395 -0.035 -0.0508 -0.062 0.0603 ...\n",
      " $ log2FC_Padj  : num  0 2.47 0 15.97 0 ...\n",
      " $ log2Skew_Padj: num  0.9 1.0065 2.3746 1.2994 0.0604 ...\n",
      " $ VAR          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ chr          : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\" ...\n",
      " $ start        : int  198679881 198679881 198679881 198679881 198679881 198679881 198679881 198679881 198679916 198679916 ...\n",
      " $ end          : int  198680149 198680149 198680149 198680149 198680028 198680028 198680028 198680028 198680063 198680063 ...\n",
      " $ window       : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 3 3 3 3 1 1 1 1 2 2 ...\n",
      " $ analysis     : Ord.factor w/ 2 levels \"MPRA model\"<\"meta analysis\": 2 2 2 2 1 1 1 1 1 1 ...\n",
      " $ COORD        : num  3 3 3 3 1 1 1 1 2 2 ...\n",
      " $ ymin         : num  3 3 3 3 1 1 1 1 2 2 ...\n",
      " $ ymax         : num  3.5 3.5 3.5 3.5 1.5 1.5 1.5 1.5 2.5 2.5 ...\n",
      "\n",
      "ymin_0\n",
      "Min. 1st Qu. Median Mean 3rd Qu. Max.\n",
      "1 2 3 3 4 5\n",
      "ymax_0\n",
      "Min. 1st Qu. Median Mean 3rd Qu. Max.\n",
      "1.5 2.5 3.5 3.5 4.5 5.5\n",
      "graph_Log2FC plot\n",
      "max_log2Skew_0\n",
      " num 0.17\n",
      "\n",
      "min_log2Skew_0\n",
      " num -0.22\n",
      "\n",
      "abs_max_log2Skew_0\n",
      " num 0.17\n",
      "\n",
      "abs_min_log2Skew_0\n",
      " num 0.22\n",
      "\n",
      "labels.log2Skew:\t-0.22 0 0.11 0.22\n",
      "DEF_2\n",
      "'data.frame':\t24 obs. of  16 variables:\n",
      " $ SNP          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ comparison   : chr  \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" \"ref_vs_alt\" ...\n",
      " $ Cell_Type    : Ord.factor w/ 4 levels \"K562\"<\"CHRF\"<..: 4 3 1 2 1 3 2 4 2 3 ...\n",
      " $ log2FC       : num  -0.06 -0.13 0 0.26 0.15 -0.03 0.78 0.01 0.74 -0.09 ...\n",
      " $ log2Skew     : num  -0.04 -0.04 -0.05 -0.06 0.06 0.03 -0.01 -0.04 0.17 0.03 ...\n",
      " $ log2FC_Padj  : num  0 2.47 0 15.97 0 ...\n",
      " $ log2Skew_Padj: num  0.9 1.0065 2.3746 1.2994 0.0604 ...\n",
      " $ VAR          : chr  \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" \"chr1_198680015_G_A\" ...\n",
      " $ chr          : chr  \"chr1\" \"chr1\" \"chr1\" \"chr1\" ...\n",
      " $ start        : int  198679881 198679881 198679881 198679881 198679881 198679881 198679881 198679881 198679916 198679916 ...\n",
      " $ end          : int  198680149 198680149 198680149 198680149 198680028 198680028 198680028 198680028 198680063 198680063 ...\n",
      " $ window       : Ord.factor w/ 5 levels \"TILE_1\"<\"TILE_2\"<..: 3 3 3 3 1 1 1 1 2 2 ...\n",
      " $ analysis     : Ord.factor w/ 2 levels \"MPRA model\"<\"meta analysis\": 2 2 2 2 1 1 1 1 1 1 ...\n",
      " $ COORD        : num  3 3 3 3 1 1 1 1 2 2 ...\n",
      " $ ymin         : num  3 3 3 3 1 1 1 1 2 2 ...\n",
      " $ ymax         : num  3.5 3.5 3.5 3.5 1.5 1.5 1.5 1.5 2.5 2.5 ...\n",
      "\n",
      "ymin_0\n",
      "Min. 1st Qu. Median Mean 3rd Qu. Max.\n",
      "1 2 3 3 4 5\n",
      "ymax_0\n",
      "Min. 1st Qu. Median Mean 3rd Qu. Max.\n",
      "1.5 2.5 3.5 3.5 4.5 5.5\n",
      "log2Skew plot\n"
     ]
    }
   ],
   "source": [
    " DEBUG<-0\n",
    "  \n",
    "  for(i in START:length(VARS))\n",
    "  {\n",
    "    VAR_sel<-VARS[i]\n",
    "      \n",
    "    dataframe_subset_sel<-dataframe_subset[which(dataframe_subset$VAR == VAR_sel),]   \n",
    "      \n",
    "    \n",
    "    rs_sel<-unique(dataframe_subset_sel$rs)\n",
    "    \n",
    "    \n",
    "\n",
    "      cat(\"-------------------------------------------------------------->\\t\")\n",
    "      cat(sprintf(as.character(i)))\n",
    "      cat(\"\\t\")\n",
    "      cat(sprintf(as.character(rs_sel)))\n",
    "      cat(\"\\t\")\n",
    "      cat(sprintf(as.character(VAR_sel)))\n",
    "      cat(\"\\n\")\n",
    "\n",
    "    if(DEBUG == 1)\n",
    "    {\n",
    "      cat(\"dataframe_subset_sel_0\\n\")\n",
    "      cat(str(dataframe_subset_sel))\n",
    "      cat(\"\\n\")\n",
    "    } \n",
    "\n",
    "    path_graphs_rs<-paste(path_graphs,paste(rs_sel,VAR_sel, sep='__'),'/',sep='')\n",
    "\n",
    "    if (file.exists(path_graphs_rs)){\n",
    "\n",
    "\n",
    "    }else{\n",
    "\n",
    "      dir.create(file.path(path_graphs_rs))\n",
    "\n",
    "    }#path_graphs_rs\n",
    "\n",
    "      ################################### extract MPRA results to plot ###################################################################################################\n",
    "\n",
    "      Element_selection_sub<-unique(MPRA_tiles$ID[which(MPRA_tiles$VAR%in%dataframe_subset_sel$VAR)])\n",
    "\n",
    "      if(DEBUG == 1)\n",
    "      {\n",
    "        cat(\"Element_selection_sub_0\\n\")\n",
    "        cat(str(Element_selection_sub))\n",
    "        cat(\"\\n\")\n",
    "\n",
    "      }\n",
    "\n",
    "      MPRA_tiles_sub<-MPRA_tiles[which(MPRA_tiles$ID%in%Element_selection_sub &\n",
    "                                      !is.na(MPRA_tiles$VAR)),]\n",
    "\n",
    "      if(DEBUG == 1)\n",
    "      {\n",
    "        cat(\"MPRA_tiles_sub_0\\n\")\n",
    "        cat(str(MPRA_tiles_sub))\n",
    "        cat(\"\\n\")\n",
    "\n",
    "      }\n",
    "\n",
    "     \n",
    "\n",
    "\n",
    "        if(dim(MPRA_tiles_sub)[1] >0){\n",
    "\n",
    "          MPRA_VAR_array<-unique(MPRA_tiles_sub$VAR)\n",
    "\n",
    "          if(DEBUG == 1)\n",
    "          {\n",
    "            cat(\"MPRA_VAR_array_0\\n\")\n",
    "            cat(str(MPRA_VAR_array))\n",
    "            cat(\"\\n\")\n",
    "            \n",
    "          }\n",
    "\n",
    "          chr_sel<-unique(MPRA_tiles_sub$chr)\n",
    "          start_sel<-min(MPRA_tiles_sub$start)-10\n",
    "          end_sel<-max(MPRA_tiles_sub$end)+10\n",
    "          \n",
    "          \n",
    "          for(k in 1:length(MPRA_VAR_array))\n",
    "          {\n",
    "            MPRA_VAR_array_sel<-MPRA_VAR_array[k]\n",
    "            \n",
    "            cat(\"--------MPRA_VAR_array_sel---------------->\\t\")\n",
    "            cat(sprintf(as.character(k)))\n",
    "            cat(\"\\t\") \n",
    "            cat(sprintf(as.character(chr_sel)))\n",
    "            cat(\"\\t\")\n",
    "            cat(sprintf(as.character(start_sel)))\n",
    "            cat(\"\\t\")\n",
    "            cat(sprintf(as.character(end_sel)))\n",
    "            cat(\"\\t\")\n",
    "            cat(sprintf(as.character(MPRA_VAR_array_sel)))\n",
    "            cat(\"\\n\")\n",
    "            \n",
    "\n",
    "            FLAG_diplotype<-length(grep(\"__\",MPRA_VAR_array_sel))\n",
    "\n",
    "            \n",
    "              cat(\"FLAG_diplotype_0\\n\")\n",
    "              cat(sprintf(as.character(FLAG_diplotype)))\n",
    "              cat(\"\\n\")\n",
    "           \n",
    "\n",
    "              if(FLAG_diplotype ==0){\n",
    "\n",
    "                  pos_array<-MPRA_VAR_array_sel\n",
    "         \n",
    "            \n",
    "                  \n",
    "              }else{\n",
    "\n",
    "                  pos_array<-unlist(strsplit(MPRA_VAR_array_sel, split=\"__\"))           \n",
    "                  \n",
    "\n",
    "             }#FLAG_diplotype ==0\n",
    "                       \n",
    "            \n",
    "            \n",
    "              cat(\"pos_array_0\\n\")\n",
    "              cat(sprintf(as.character(pos_array)))\n",
    "              cat(\"\\n\")\n",
    "            \n",
    "              \n",
    "            pos_array<-gsub(\"^chr[^_]+_\",\"\",pos_array)\n",
    "            \n",
    "            if(DEBUG == 1)\n",
    "            {\n",
    "              cat(\"pos_array_1\\n\")\n",
    "              cat(sprintf(as.character(pos_array)))\n",
    "              cat(\"\\n\")\n",
    "            }\n",
    "            \n",
    "            pos_array<-as.integer(gsub(\"_.+$\",\"\",pos_array))\n",
    "            \n",
    "           \n",
    "          cat(\"pos_array_2\\n\")\n",
    "          cat(sprintf(as.character(pos_array)))\n",
    "          cat(\"\\n\")\n",
    "            \n",
    "            \n",
    "            \n",
    "                        \n",
    "            MPRA_tiles_sub_VAR_sel<-MPRA_tiles_sub[which(MPRA_tiles_sub$VAR == MPRA_VAR_array_sel),]\n",
    "            \n",
    "            if(DEBUG == 1)\n",
    "            {\n",
    "              cat(\"MPRA_tiles_sub_VAR_sel_0\\n\")\n",
    "              cat(str(MPRA_tiles_sub_VAR_sel))\n",
    "              cat(\"\\n\")\n",
    "              \n",
    "            }\n",
    "            \n",
    "            \n",
    "            \n",
    "            if(dim(MPRA_tiles_sub_VAR_sel)[1] >0)\n",
    "            { \n",
    "\n",
    "              #### minimal REP df -------------------------\n",
    "\n",
    "              REP<-unique(MPRA_tiles_sub_VAR_sel[,which(colnames(MPRA_tiles_sub_VAR_sel)%in%c(\"SNP\",\"comparison\",\"window\",\"VAR\",\"chr\",\"start\",\"end\",\"Cell_Type\",\"log2FC\",\"logPadj_BF\",\"log2Skew\",\"Skew_logPadj\"))])\n",
    "\n",
    "              REP$analysis<-'MPRA model'\n",
    "\n",
    "                 if(DEBUG == 1)\n",
    "                    {\n",
    "                      cat(\"REP_0\\n\")\n",
    "                      cat(str(REP))\n",
    "                      cat(\"\\n\")\n",
    "                      \n",
    "                    }\n",
    "\n",
    "                colnames(REP)[which(colnames(REP) == 'log2FC')]<-'log2FC'\n",
    "                colnames(REP)[which(colnames(REP) == 'logPadj_BF')]<-'log2FC_Padj'\n",
    "                colnames(REP)[which(colnames(REP) == 'log2Skew')]<-'log2Skew'\n",
    "                colnames(REP)[which(colnames(REP) == 'Skew_logPadj')]<-'log2Skew_Padj'\n",
    "\n",
    "                 if(DEBUG == 1)\n",
    "                    {\n",
    "                      cat(\"REP_1\\n\")\n",
    "                      cat(str(REP))\n",
    "                      cat(\"\\n\")\n",
    "                      \n",
    "                    }\n",
    "\n",
    "              #### Extract metanalysis results-----\n",
    "\n",
    "             MPRA_post_meta_analysis_sel <-unique(MPRA_post_meta_analysis[which(MPRA_post_meta_analysis$VAR == MPRA_VAR_array_sel),])             \n",
    "\n",
    "\n",
    "                if(DEBUG == 1)\n",
    "                {\n",
    "                  cat(\"----------------------------------->MPRA_post_meta_analysis_sel_0\\n\")\n",
    "                  cat(str(MPRA_post_meta_analysis_sel))\n",
    "                  cat(\"\\n\")                  \n",
    "                }\n",
    "\n",
    "                if(dim(MPRA_post_meta_analysis_sel)[1] >0){\n",
    "\n",
    "                    REP_meta<-unique(MPRA_post_meta_analysis_sel[,which(colnames(MPRA_post_meta_analysis_sel)%in%c(\"SNP\",\"comparison\",\"VAR\",\"chr\",\"start\",\"end\",\"Cell_Type\",\"log2FC_meta\",\"lof2FC_meta_padj\",\"log2Skew_meta\",\"lof2Skew_meta_padj\"))])\n",
    "\n",
    "                    REP_meta$chr<-chr_sel\n",
    "                    REP_meta$start<-as.integer(start_sel+10)\n",
    "                    REP_meta$end<-as.integer(end_sel-10)\n",
    "\n",
    "                    REP_meta$window<-\"TILE_3\"\n",
    "\n",
    "                    if(DEBUG == 1)\n",
    "                    {\n",
    "                      cat(\"REP_meta_0\\n\")\n",
    "                      cat(str(REP_meta))\n",
    "                      cat(\"\\n\")                  \n",
    "                    }\n",
    "\n",
    "                    REP_meta$window<-factor(REP_meta$window,\n",
    "                            levels=c(\"TILE_1\",\"TILE_2\",\"TILE_3\",\"TILE_4\",\"TILE_5\"),\n",
    "                            ordered=T)\n",
    "\n",
    "                     REP_meta$analysis<-'meta analysis'\n",
    "\n",
    "                     colnames(REP_meta)[which(colnames(REP_meta) == 'log2FC_meta')]<-'log2FC'\n",
    "                    colnames(REP_meta)[which(colnames(REP_meta) == 'lof2FC_meta_padj')]<-'log2FC_Padj'\n",
    "                    colnames(REP_meta)[which(colnames(REP_meta) == 'log2Skew_meta')]<-'log2Skew'\n",
    "                    colnames(REP_meta)[which(colnames(REP_meta) == 'lof2Skew_meta_padj')]<-'log2Skew_Padj'\n",
    "\n",
    "                    \n",
    "                     if(DEBUG == 1)\n",
    "                    {\n",
    "                      cat(\"REP_meta_1\\n\")\n",
    "                      cat(str(REP_meta))\n",
    "                      cat(\"\\n\")                  \n",
    "                    }\n",
    "\n",
    "                    \n",
    "                }else{\n",
    "\n",
    "                    REP_meta<-data.frame()\n",
    "                    \n",
    "                }#dim(MPRA_post_meta_analysis_sel)[1] >0\n",
    "\n",
    "                #### rbind analysis and meta-analysis -----------------\n",
    "\n",
    "                DEF<-rbind(REP_meta,REP)\n",
    "\n",
    "                  if(DEBUG == 1)\n",
    "                    {\n",
    "                      cat(\"DEF_0\\n\")\n",
    "                      cat(str(DEF))\n",
    "                      cat(\"\\n\")                  \n",
    "                    }\n",
    "\n",
    "                DEF$analysis<-factor(DEF$analysis,\n",
    "                            levels=c('MPRA model','meta analysis'),\n",
    "                            ordered=T)\n",
    "\n",
    "\n",
    "                if(DEBUG == 1)\n",
    "                    {\n",
    "                      cat(\"DEF_1\\n\")\n",
    "                      cat(str(DEF))\n",
    "                      cat(\"\\n\")                  \n",
    "                    }\n",
    "\n",
    "\n",
    "                \n",
    "\n",
    "                \n",
    "              ############### log2FC plot\n",
    "              \n",
    "              DEF$log2FC<-round(DEF$log2FC,2)\n",
    "              \n",
    "              A<-round(summary(DEF$log2FC[!is.na(DEF$log2FC)]),4)\n",
    "              \n",
    "              max_log2FC<-max(A)\n",
    "              min_log2FC<-min(A)\n",
    "              \n",
    "              abs_max_log2FC<-abs(max_log2FC)\n",
    "              abs_min_log2FC<-abs(min_log2FC)\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"max_log2FC_0\\n\")\n",
    "                cat(str(max_log2FC))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"min_log2FC_0\\n\")\n",
    "                cat(str(min_log2FC))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"abs_max_log2FC_0\\n\")\n",
    "                cat(str(abs_max_log2FC))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"abs_min_log2FC_0\\n\")\n",
    "                cat(str(abs_min_log2FC))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "              \n",
    "              if(abs_max_log2FC > abs_min_log2FC)\n",
    "              {\n",
    "                step<-abs(max_log2FC - -1*max_log2FC)/4\n",
    "                breaks.Log2FC<-sort(unique(c(0,seq(from= -1*abs_max_log2FC, to=abs_max_log2FC,by=step))))\n",
    "                breaks.Log2FC<-sort(unique(c(0,-1*abs_max_log2FC, abs_max_log2FC)))\n",
    "                labels.Log2FC<-as.character(round(breaks.Log2FC,2))\n",
    "                \n",
    "              }else{\n",
    "                \n",
    "                step<-abs(min_log2FC - -1*min_log2FC)/4\n",
    "                breaks.Log2FC<-sort(unique(c(0,seq(from= -1*abs_min_log2FC, to=abs_min_log2FC,by=step))))\n",
    "                breaks.Log2FC<-sort(unique(c(0,-1*abs_min_log2FC,abs_min_log2FC,by=step)))\n",
    "                labels.Log2FC<-as.character(round(breaks.Log2FC,2))\n",
    "                \n",
    "              }#abs_max_log2FC > abs_min_log2FC\n",
    "              \n",
    "              \n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"labels.Log2FC:\\t\")\n",
    "                cat(sprintf(as.character(labels.Log2FC)))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                \n",
    "                \n",
    "                # quit(status = 1)\n",
    "              }\n",
    "              \n",
    "              DEF$COORD<-as.numeric(DEF$window)\n",
    "              DEF$ymin<-DEF$COORD\n",
    "              DEF$ymax<-DEF$COORD+0.5\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"DEF_2\\n\")\n",
    "                cat(str(DEF))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"ymin_0\\n\")\n",
    "                cat(sprintf(as.character(names(summary(DEF$ymin)))))\n",
    "                cat(\"\\n\")\n",
    "                cat(sprintf(as.character(summary(DEF$ymin))))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"ymax_0\\n\")\n",
    "                cat(sprintf(as.character(names(summary(DEF$ymax)))))\n",
    "                cat(\"\\n\")\n",
    "                cat(sprintf(as.character(summary(DEF$ymax))))\n",
    "                cat(\"\\n\")\n",
    "              }       \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              graph_Log2FC<-ggplot()+\n",
    "                scale_x_continuous(name=\"genomic coordinate\",breaks=c(start_sel,end_sel),\n",
    "                                   labels=as.character(c(start_sel,end_sel)),\n",
    "                                   limits=c(start_sel,end_sel))+\n",
    "                scale_y_continuous(name=NULL)+                 \n",
    "                geom_rect(data=DEF,\n",
    "                          aes(ymin=ymin,\n",
    "                              ymax=ymax,\n",
    "                              xmin=start,\n",
    "                              xmax=end,\n",
    "                              fill=log2FC), color='black',\n",
    "                          size=0.01)+\n",
    "                scale_fill_gradient2(\n",
    "                  midpoint = 0,\n",
    "                  low = \"gray\",\n",
    "                  mid = \"white\",\n",
    "                  high = \"green\",\n",
    "                  breaks=breaks.Log2FC,labels=labels.Log2FC,\n",
    "                  limits=c(breaks.Log2FC[1],breaks.Log2FC[length(breaks.Log2FC)]),\n",
    "                  name=paste('log2FC',sep=\"\\n\"),na.value = \"gray\")+\n",
    "                new_scale(\"fill\")+\n",
    "                geom_rect(data=DEF[which(DEF$log2FC_Padj >= 1.3),],\n",
    "                          aes(ymin=ymin,\n",
    "                              ymax=ymax,\n",
    "                              xmin=start,\n",
    "                              xmax=end),\n",
    "                          fill=NA,\n",
    "                          color='black',size=0.5)\n",
    "              \n",
    "              \n",
    "              graph_Log2FC<-graph_Log2FC+\n",
    "                facet_grid( Cell_Type+analysis ~ ., scales='free_x', space='free_x', switch=\"y\", drop=FALSE)+\n",
    "                theme_cowplot(font_size = 2)+\n",
    "                theme( strip.background = element_blank(),\n",
    "                       strip.placement = \"outside\",\n",
    "                       strip.text = element_text(size=6),\n",
    "                       panel.spacing = unit(0.2, \"lines\"),\n",
    "                       panel.background=element_rect(fill=\"white\"),\n",
    "                       panel.border=element_rect(colour=\"white\",size=0,5),\n",
    "                       panel.grid.major = element_blank(),\n",
    "                       panel.grid.minor = element_blank())+\n",
    "                geom_vline(xintercept=pos_array, color=\"black\", linetype='dashed',size=0.25)+\n",
    "                theme_classic()+\n",
    "                theme(axis.title.y=element_blank(),\n",
    "                      axis.title.x=element_text(size=8, color=\"black\", family=\"sans\"),\n",
    "                      axis.text.y=element_blank(),\n",
    "                      axis.text.x=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                      axis.line.x = element_line(size = 0.2),\n",
    "                      axis.ticks.x = element_line(size = 0.2),\n",
    "                      axis.ticks.y = element_blank(),\n",
    "                      axis.line.y = element_line(size = 0.2))+\n",
    "                theme(legend.title = element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                      legend.text = element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                      legend.position=\"bottom\")+\n",
    "                ggeasy::easy_center_title()\n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              setwd(path_graphs_rs)\n",
    "              \n",
    "              svgname<-paste(paste(\"MPRA\",'Log2FC', MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "              makesvg = TRUE\n",
    "              \n",
    "              if (makesvg == TRUE)\n",
    "              {\n",
    "                ggsave(svgname, plot= graph_Log2FC,\n",
    "                       device=\"svg\",height=4, width=2)\n",
    "              }\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"graph_Log2FC plot\\n\")\n",
    "                \n",
    "              }\n",
    "              \n",
    "              \n",
    "              ############### log2Skew plot ----------------------\n",
    "              \n",
    "              DEF$log2Skew<-round(DEF$log2Skew,2)\n",
    "              \n",
    "              A<-round(summary(DEF$log2Skew[!is.na(DEF$log2Skew)]),4)\n",
    "              \n",
    "              max_log2Skew<-max(A)\n",
    "              min_log2Skew<-min(A)\n",
    "              \n",
    "              abs_max_log2Skew<-abs(max_log2Skew)\n",
    "              abs_min_log2Skew<-abs(min_log2Skew)\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"max_log2Skew_0\\n\")\n",
    "                cat(str(max_log2Skew))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"min_log2Skew_0\\n\")\n",
    "                cat(str(min_log2Skew))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"abs_max_log2Skew_0\\n\")\n",
    "                cat(str(abs_max_log2Skew))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"abs_min_log2Skew_0\\n\")\n",
    "                cat(str(abs_min_log2Skew))\n",
    "                cat(\"\\n\")\n",
    "              }\n",
    "              \n",
    "              if(abs_max_log2Skew > abs_min_log2Skew)\n",
    "              {\n",
    "                step<-abs(max_log2Skew - -1*max_log2Skew)/4\n",
    "                breaks.log2Skew<-sort(unique(c(0,seq(from= -1*abs_max_log2Skew, to=abs_max_log2Skew,by=step))))\n",
    "                breaks.log2Skew<-sort(unique(c(0,-1*abs_max_log2Skew, abs_max_log2Skew)))\n",
    "                labels.log2Skew<-as.character(round(breaks.log2Skew,2))\n",
    "                \n",
    "              }else{\n",
    "                \n",
    "                step<-abs(min_log2Skew - -1*min_log2Skew)/4\n",
    "                breaks.log2Skew<-sort(unique(c(0,seq(from= -1*abs_min_log2Skew, to=abs_min_log2Skew,by=step))))\n",
    "                breaks.log2Skew<-sort(unique(c(0,-1*abs_min_log2Skew,abs_min_log2Skew,by=step)))\n",
    "                labels.log2Skew<-as.character(round(breaks.log2Skew,2))\n",
    "                \n",
    "              }#abs_max_log2Skew > abs_min_log2Skew\n",
    "              \n",
    "              \n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"labels.log2Skew:\\t\")\n",
    "                cat(sprintf(as.character(labels.log2Skew)))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                \n",
    "                \n",
    "                # quit(status = 1)\n",
    "              }\n",
    "              \n",
    "              DEF$COORD<-as.numeric(DEF$window)\n",
    "              DEF$ymin<-DEF$COORD\n",
    "              DEF$ymax<-DEF$COORD+0.5\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"DEF_2\\n\")\n",
    "                cat(str(DEF))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"ymin_0\\n\")\n",
    "                cat(sprintf(as.character(names(summary(DEF$ymin)))))\n",
    "                cat(\"\\n\")\n",
    "                cat(sprintf(as.character(summary(DEF$ymin))))\n",
    "                cat(\"\\n\")\n",
    "                \n",
    "                cat(\"ymax_0\\n\")\n",
    "                cat(sprintf(as.character(names(summary(DEF$ymax)))))\n",
    "                cat(\"\\n\")\n",
    "                cat(sprintf(as.character(summary(DEF$ymax))))\n",
    "                cat(\"\\n\")\n",
    "              }       \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              graph_log2Skew<-ggplot()+\n",
    "                scale_x_continuous(name=\"genomic coordinate\",breaks=c(start_sel,end_sel),\n",
    "                                   labels=as.character(c(start_sel,end_sel)),\n",
    "                                   limits=c(start_sel,end_sel))+\n",
    "                scale_y_continuous(name=NULL)+                 \n",
    "                geom_rect(data=DEF,\n",
    "                          aes(ymin=ymin,\n",
    "                              ymax=ymax,\n",
    "                              xmin=start,\n",
    "                              xmax=end,\n",
    "                              fill=log2Skew), color='black',\n",
    "                          size=0.01)+\n",
    "                scale_fill_gradient2(\n",
    "                  midpoint = 0,\n",
    "                  low = \"blue\",\n",
    "                  mid = \"white\",\n",
    "                  high = \"red\",\n",
    "                  breaks=breaks.log2Skew,labels=labels.log2Skew,\n",
    "                  limits=c(breaks.log2Skew[1],breaks.log2Skew[length(breaks.log2Skew)]),\n",
    "                  name=paste('log2Skew',sep=\"\\n\"),na.value = \"gray\")+\n",
    "                new_scale(\"fill\")+\n",
    "                geom_rect(data=DEF[which(DEF$log2Skew_Padj >= 1.3),],\n",
    "                          aes(ymin=ymin,\n",
    "                              ymax=ymax,\n",
    "                              xmin=start,\n",
    "                              xmax=end),\n",
    "                          fill=NA,\n",
    "                          color='black',size=0.5)\n",
    "              \n",
    "              \n",
    "              graph_log2Skew<-graph_log2Skew+\n",
    "                facet_grid( Cell_Type+analysis ~ ., scales='free_x', space='free_x', switch=\"y\", drop=FALSE)+\n",
    "                theme_cowplot(font_size = 2)+\n",
    "                theme( strip.background = element_blank(),\n",
    "                       strip.placement = \"outside\",\n",
    "                       strip.text = element_text(size=6),\n",
    "                       panel.spacing = unit(0.2, \"lines\"),\n",
    "                       panel.background=element_rect(fill=\"white\"),\n",
    "                       panel.border=element_rect(colour=\"white\",size=0,5),\n",
    "                       panel.grid.major = element_blank(),\n",
    "                       panel.grid.minor = element_blank())+\n",
    "                geom_vline(xintercept=pos_array, color=\"black\", linetype='dashed',size=0.25)+\n",
    "                theme_classic()+\n",
    "                theme(axis.title.y=element_blank(),\n",
    "                      axis.title.x=element_text(size=8, color=\"black\", family=\"sans\"),\n",
    "                      axis.text.y=element_blank(),\n",
    "                      axis.text.x=element_text(angle=0,size=6, color=\"black\", family=\"sans\"),\n",
    "                      axis.line.x = element_line(size = 0.2),\n",
    "                      axis.ticks.x = element_line(size = 0.2),\n",
    "                      axis.ticks.y = element_blank(),\n",
    "                      axis.line.y = element_line(size = 0.2))+\n",
    "                theme(legend.title = element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                      legend.text = element_text(size=6, color=\"black\", family=\"sans\"),\n",
    "                      legend.position=\"bottom\")+\n",
    "                ggeasy::easy_center_title()\n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              setwd(path_graphs_rs)\n",
    "              \n",
    "              svgname<-paste(paste(\"MPRA\",'log2Skew', MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "              makesvg = TRUE\n",
    "              \n",
    "              if (makesvg == TRUE)\n",
    "              {\n",
    "                ggsave(svgname, plot= graph_log2Skew,\n",
    "                       device=\"svg\",height=4, width=2)\n",
    "              }\n",
    "              \n",
    "              if(DEBUG == 1)\n",
    "              {\n",
    "                cat(\"log2Skew plot\\n\")\n",
    "                \n",
    "              }\n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "              \n",
    "            }else{\n",
    "              \n",
    "              graph_Log2FC<-NA\n",
    "              graph_log2Skew<-NA\n",
    "              \n",
    "            }#dim(MPRA_tiles_sub_VAR_sel)[1] >0         \n",
    "            \n",
    "            \n",
    "            \n",
    "            graph_DEF<-plot_grid(graph_Log2FC,graph_log2Skew,\n",
    "                                 nrow = 1,\n",
    "                                 ncol = 2,                                   \n",
    "                                 rel_widths=c(0.5,0.5))\n",
    "            \n",
    "            graph_DEF_with_title <- ggdraw() +\n",
    "              draw_plot(graph_DEF, x = 0, y = 0, width = 1, height = 0.92) + # Draw your combined plots\n",
    "              draw_label(MPRA_VAR_array_sel,\n",
    "                         x = 0.02, y = 0.95, # Adjust x, y for position (0,0 is bottom-left, 1,1 is top-right)\n",
    "                         size = 8,         # Font size for the title (adjust as needed)\n",
    "                         fontfamily = \"sans\", # Sans-serif font\n",
    "                         fontface = \"bold\",   # Make it bold for emphasis\n",
    "                         vjust = 0.5, \n",
    "                         hjust = 0) # Adjust vertical and horizontal justification\n",
    "            \n",
    "            \n",
    "            \n",
    "            \n",
    "            setwd(path_graphs_rs)\n",
    "            \n",
    "            svgname<-paste(paste(\"MPRA\",'DEF', MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "            makesvg = TRUE\n",
    "            \n",
    "            if (makesvg == TRUE)\n",
    "            {\n",
    "              ggsave(svgname, plot= graph_DEF_with_title,\n",
    "                     device=\"svg\",\n",
    "                     height=5, width=5)\n",
    "              \n",
    "            }\n",
    "            \n",
    "            \n",
    "            \n",
    "            \n",
    "            path_figures = paste(\"/group/soranzo/manuel.tardaguila/Rebuild_sit_plots/test/Figure_graphs/\",sep='')\n",
    "            \n",
    "            if(file.exists(path_figures)){\n",
    "              \n",
    "            }else{\n",
    "              dir.create(path_figures)\n",
    "            }\n",
    "\n",
    "              makesvg = FALSE\n",
    "            \n",
    "            \n",
    "            if(rs_sel == 'rs139141690'){\n",
    "              \n",
    "              svgname<-paste(paste(\"Figure_4_\",'panel_A', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "              makesvg = TRUE\n",
    "              \n",
    "              \n",
    "            }else{\n",
    "              \n",
    "              if(rs_sel == 'rs12733073'){\n",
    "                \n",
    "                \n",
    "                svgname<-paste(paste(\"Figure_S4_\",'panel_A', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                makesvg = TRUE\n",
    "                \n",
    "              }else{\n",
    "                \n",
    "                if(rs_sel == 'rs569022307'){\n",
    "                  \n",
    "                  svgname<-paste(paste(\"Figure_S4_\",'panel_B', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                    makesvg = TRUE\n",
    "                  \n",
    "                }else{\n",
    "                  \n",
    "                  if(rs_sel == 'rs150649461'){\n",
    "                    \n",
    "                    svgname<-paste(paste(\"Figure_S4_\",'panel_C', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                      makesvg = TRUE\n",
    "                    \n",
    "                    \n",
    "                  }else{\n",
    "                    \n",
    "                    if(rs_sel == 'rs150221884'){\n",
    "                      \n",
    "                      svgname<-paste(paste(\"Figure_S4_\",'panel_D', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                        makesvg = TRUE\n",
    "                      \n",
    "                      \n",
    "                    }else{\n",
    "                      \n",
    "                      if(rs_sel == 'rs543594419'){\n",
    "                        \n",
    "                        svgname<-paste(paste(\"Figure_S4_\",'panel_E', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                          makesvg = TRUE\n",
    "                        \n",
    "                        \n",
    "                      }else{\n",
    "                        \n",
    "                        if(rs_sel == 'rs572476245'){\n",
    "                          \n",
    "                          svgname<-paste(paste(\"Figure_S4_\",'panel_E', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                            makesvg = TRUE\n",
    "                          \n",
    "                          \n",
    "                        }else{\n",
    "                          \n",
    "                          if(rs_sel == 'rs114050631'){\n",
    "                            \n",
    "                            svgname<-paste(paste(\"Figure_S2_\",'panel_A', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                              makesvg = TRUE\n",
    "                            \n",
    "                            \n",
    "                          }else{\n",
    "                            \n",
    "                            if(rs_sel == 'rs150813342'){\n",
    "                              \n",
    "                              svgname<-paste(paste(\"Figure_S2_\",'panel_B_1', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                makesvg = TRUE\n",
    "                              \n",
    "                              \n",
    "                            }else{\n",
    "                              \n",
    "                              if(rs_sel == 'rs187715179'){\n",
    "                                \n",
    "                                svgname<-paste(paste(\"Figure_S2_\",'panel_B_2', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                  makesvg = TRUE\n",
    "\n",
    "                                  }else{                                     \n",
    "\n",
    "\n",
    "                                  if(rs_sel == 'rs182089148'){\n",
    "\n",
    "                                  \n",
    "                                     svgname<-paste(paste(\"Figure_S4_\",'panel_UNKWN', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                     makesvg = TRUE\n",
    "                                   \n",
    "                                \n",
    "                                    }else{\n",
    "\n",
    "                                       if(rs_sel == 'rs112401631'){\n",
    "\n",
    "\n",
    "                                           svgname<-paste(paste(\"Figure_S4_\",'panel_UNKWN', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                           makesvg = TRUE\n",
    "\n",
    "                                           }else{\n",
    "\n",
    "                                               if(rs_sel == 'rs17758695'){\n",
    "    \n",
    "    \n",
    "                                               svgname<-paste(paste(\"Figure_S4_\",'panel_UNKWN', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                               makesvg = TRUE\n",
    "    \n",
    "                                               }else{\n",
    "\n",
    "\n",
    "                                                   if(rs_sel == 'rs149489081'){\n",
    "    \n",
    "    \n",
    "                                                       svgname<-paste(paste(\"Figure_S4_\",'panel_UNKWN', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                                       makesvg = TRUE\n",
    "            \n",
    "                                                       }else{\n",
    "\n",
    "                                                       if(rs_sel == 'rs35592432'){\n",
    "    \n",
    "    \n",
    "                                                           svgname<-paste(paste(\"Figure_S4_\",'panel_UNKWN', 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                                           makesvg = TRUE\n",
    "                \n",
    "                                                           }else{\n",
    "\n",
    "                                                            svgname<-paste(paste(\"Other_locus\",rs_sel, 'MPRA',MPRA_VAR_array_sel,rs_sel, sep='_'),\".svg\",sep='')\n",
    "                                                            makesvg = FALSE\n",
    "                                                           \n",
    "                                                        }#rs_sel == 'rs35592432'\n",
    "                                                       \n",
    "                                                    }#rs_sel == 'rs149489081'\n",
    "\n",
    "                                                }#rs_sel == 'rs17758695'                                                                                \n",
    "\n",
    "                                           }#rs_sel == 'rs112401631'            \n",
    "  \n",
    "                                      }#  rs_sel == 'rs182089148'                                                          \n",
    "                              }#if(rs_sel == 'rs187715179'\n",
    "                              \n",
    "                            }#if(rs_sel == 'rs150813342')    \n",
    "                            \n",
    "                          }#rs_sel == 'rs114050631'                                                  \n",
    "                          \n",
    "                          \n",
    "                        }#rs_sel == 'rs572476245'\n",
    "                        \n",
    "                      }#rs_sel == 'rs543594419'\n",
    "                      \n",
    "                    }#rs_sel == 'rs150221884'\n",
    "                    \n",
    "                  }#rs_sel == 'rs150649461'\n",
    "                  \n",
    "                  \n",
    "                }#rs_sel == 'rs569022307'\n",
    "                \n",
    "              }#rs_sel == 'rs12733073'\n",
    "              \n",
    "            }#rs_sel == 'chr7_101499930_G_A'\n",
    "            \n",
    "            \n",
    "            \n",
    "            \n",
    "            setwd(path_figures)\n",
    "            \n",
    "            if (makesvg == TRUE)\n",
    "            {\n",
    "              ggsave(svgname, plot= graph_DEF_with_title,\n",
    "                     device=\"svg\",\n",
    "                     height=5, width=5)\n",
    "              \n",
    "            }\n",
    "            \n",
    "            ## break\n",
    "          }# k in 1:length(MPRA_VAR_array)  \n",
    "        }else{\n",
    "          \n",
    "          cat(\"-----------------------------------------------------> NO MPRA results\\n\")\n",
    "        }#dim(MPRA_VAR_sub)[1] >0\n",
    "\n",
    "      ## break\n",
    "      \n",
    "}#i in START:length(VARS)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "73d375aa-e132-4e99-8ea9-9337dd6cc37d",
   "metadata": {},
   "source": [
    "# AFTER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "163b4651-8640-4623-a767-3475b71c88ce",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13ca09d3-0de9-45f5-a0f1-dd7cefba28ef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3906bc68-4f5a-4125-b924-8f54aa9c6784",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d0e2229-002a-46ad-8288-58a8e48f3d24",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4de7f9f2-5593-436b-a9c7-db65175977a6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6b86a27e-208a-4b72-a147-fc8c1343cd84",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81255fc5-9493-456d-b90d-44655e81714d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55bd6659-3220-4f5a-ae77-f9b852f06044",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e53ff2df-ce7b-496c-ab72-d17f46aa6a37",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "general_purpose",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
